// ng-metro4/esm2015/lib/base/animation/animation.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
let AnimationDirective = class AnimationDirective {
    constructor(element) {
        this.element = element;
        this.oldClasses = [];
        this.jElement = $(this.element.nativeElement);
    }
    createElement() {
        const newClasses = [`ani-${this.hover ? 'hover-' : ''}${this.animation}`];
        this.oldClasses.forEach(c => {
            this.jElement.removeClass(c);
        });
        newClasses.forEach(c => {
            this.jElement.addClass(c);
        });
        this.oldClasses = newClasses;
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
};
AnimationDirective.ctorParameters = () => [
    { type: ElementRef }
];
__decorate([
    Input('m4-animation'),
    __metadata("design:type", String)
], AnimationDirective.prototype, "animation", void 0);
__decorate([
    Input('animation-hover'),
    __metadata("design:type", Boolean)
], AnimationDirective.prototype, "hover", void 0);
AnimationDirective = __decorate([
    Directive({
        selector: '[m4-animation]'
    }),
    __metadata("design:paramtypes", [ElementRef])
], AnimationDirective);
export { AnimationDirective };
