// ng-metro4/esm5/lib/form/radio/radio.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ObjectHelper } from '../../helper/object-helper';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
var RadioComponent = /** @class */ (function (_super) {
    __extends(RadioComponent, _super);
    function RadioComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RadioComponent_1 = RadioComponent;
    RadioComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.clonedElement.attr('name', _this.name);
            _this.radio = _this.clonedElement.radio().data('radio');
            _this.clonedElement.one('blur', function () {
                _this.touchCallback();
            });
            _this.clonedElement.on('change', function () {
                _this.changeValue(_this.value);
            });
            complete();
        });
    };
    RadioComponent.prototype.disable = function (disabled) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (disabled) {
                _this.clonedElement.parent().addClass('disabled');
                _this.clonedElement.attr('disabled', 'disabled');
            }
            else {
                _this.clonedElement.parent().removeClass('disabled');
                _this.clonedElement.removeAttr('disabled');
            }
        });
    };
    RadioComponent.prototype.newValue = function () {
        if (!this.radio) {
            return;
        }
        if (ObjectHelper.compare(this.innerValue, this.value)) {
            this.clonedElement.prop('checked', true);
        }
        else {
            this.clonedElement.prop('checked', false);
        }
    };
    RadioComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (_this.clonedElement) {
                var target_1 = _this.clonedElement.parent();
                oldClasses.forEach(function (cls) {
                    target_1.removeClass(cls);
                });
                newClasses.forEach(function (cls) {
                    target_1.addClass(cls);
                });
            }
        });
    };
    var RadioComponent_1;
    __decorate([
        Input('name'),
        __metadata("design:type", String)
    ], RadioComponent.prototype, "name", void 0);
    __decorate([
        Input('value'),
        __metadata("design:type", Object)
    ], RadioComponent.prototype, "value", void 0);
    __decorate([
        Input('style'),
        __metadata("design:type", Number)
    ], RadioComponent.prototype, "style", void 0);
    __decorate([
        Input('caption'),
        __metadata("design:type", String)
    ], RadioComponent.prototype, "caption", void 0);
    __decorate([
        Input('caption-position'),
        __metadata("design:type", String)
    ], RadioComponent.prototype, "captionPosition", void 0);
    __decorate([
        Input('cls-radio'),
        __metadata("design:type", String)
    ], RadioComponent.prototype, "clsRadio", void 0);
    __decorate([
        Input('cls-caption'),
        __metadata("design:type", String)
    ], RadioComponent.prototype, "clsCaption", void 0);
    __decorate([
        Input('cls-check'),
        __metadata("design:type", String)
    ], RadioComponent.prototype, "clsCheck", void 0);
    __decorate([
        ViewChild('input', { static: true }),
        __metadata("design:type", ElementRef)
    ], RadioComponent.prototype, "input", void 0);
    RadioComponent = RadioComponent_1 = __decorate([
        Component({
            selector: 'm4-radio',
            template: "<input #input\r\n\r\n\r\n       [attr.data-style]=\"style\"\r\n       [attr.data-caption]=\"caption\"\r\n       [attr.data-caption-position]=\"captionPosition\"\r\n\r\n       [attr.data-cls-radio]=\"clsRadio\"\r\n       [attr.data-cls-caption]=\"clsCaption\"\r\n       [attr.data-cls-check]=\"clsCheck\">\r\n",
            providers: [DefaultValueAccessor.get(RadioComponent_1), TypeAlias.get(RadioComponent_1)],
            changeDetection: ChangeDetectionStrategy.Default,
            encapsulation: ViewEncapsulation.None,
            styles: [".radio.primary input[type=radio]:checked~.check{background-color:#0366d6;border-color:#0366d6}.radio.primary .check{border-color:#0366d6}.radio.secondary input[type=radio]:checked~.check{background-color:#607d8b;border-color:#607d8b}.radio.secondary .check{border-color:#607d8b}.radio.success input[type=radio]:checked~.check{background-color:#60a917;border-color:#60a917}.radio.success .check{border-color:#60a917}.radio.alert input[type=radio]:checked~.check{background-color:#ce352c;border-color:#ce352c}.radio.alert .check{border-color:#ce352c}.radio.warning input[type=radio]:checked~.check{background-color:#ff9447;border-color:#ff9447}.radio.warning .check{border-color:#ff9447}.radio.yellow input[type=radio]:checked~.check{background-color:#ffe484;border-color:#ffe484}.radio.yellow .check{border-color:#ffe484}.radio.info input[type=radio]:checked~.check{background-color:#5ebdec;border-color:#5ebdec}.radio.info .check{border-color:#5ebdec}.radio.dark input[type=radio]:checked~.check{background-color:#505050;border-color:#505050}.radio.dark .check{border-color:#505050}.radio.light input[type=radio]:checked~.check{background-color:#f8f8f8;border-color:#f8f8f8}.radio.light .check{border-color:#f8f8f8}"]
        })
    ], RadioComponent);
    return RadioComponent;
}(ControlBase));
export { RadioComponent };
