// ng-metro4/esm5/lib/controls/gravatar/gravatar.component.js
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
var GravatarComponent = /** @class */ (function () {
    function GravatarComponent() {
        this.size = 80;
    }
    GravatarComponent.prototype.ngOnInit = function () {
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], GravatarComponent.prototype, "email", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], GravatarComponent.prototype, "size", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], GravatarComponent.prototype, "default", void 0);
    GravatarComponent = __decorate([
        Component({
            selector: 'm4-gravatar',
            template: "<img data-role=\"gravatar\"\r\n     [attr.data-email]=\"email\"\r\n     [attr.data-size]=\"size\"\r\n     [attr.data-default]=\"default\">\r\n",
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        }),
        __metadata("design:paramtypes", [])
    ], GravatarComponent);
    return GravatarComponent;
}());
export { GravatarComponent };
