// ng-metro4/esm5/lib/form/calendar/calendar.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import * as moment from 'moment';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
var _moment = moment;
var CalendarComponent = /** @class */ (function (_super) {
    __extends(CalendarComponent, _super);
    function CalendarComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CalendarComponent_1 = CalendarComponent;
    CalendarComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.object.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.calendar = _this.clonedElement.calendar().data('calendar');
            asapScheduler.schedule(function () {
                _this.clonedElement.find('.calendar-content .day, .calendar-content span, .calendar-footer button').on('click', function () {
                    _this.touchCallback();
                    _this.clonedElement.find('.calendar-content .day, .calendar-content span, .calendar-footer button').unbind('click');
                });
            });
            var selectDays = function (selection) {
                var dates = selection.map(function (s) { return _moment(s); });
                if (_this.multiSelect) {
                    _this.changeValue(dates);
                }
                else {
                    _this.changeValue(dates.length > 0 ? dates[0] : null);
                }
            };
            _this.calendar.options.onDayClick = function (selection) {
                if (_this.pickerMode) {
                    selectDays(selection);
                }
            };
            _this.calendar.options.onDone = function (selection) {
                selectDays(selection);
            };
            complete();
        });
    };
    CalendarComponent.prototype.disable = function (disabled) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (disabled) {
                _this.clonedElement.addClass('disabled');
            }
            else {
                _this.clonedElement.removeClass('disabled');
            }
        });
    };
    CalendarComponent.prototype.newValue = function () {
        var _this = this;
        if (!this.calendar) {
            return;
        }
        this.calendar.selected = [];
        asapScheduler.schedule(function () {
            if (_this.innerValue) {
                if (_this.multiSelect) {
                    _this.calendar.setPreset(_this.innerValue.map(function (v) { return v.format('MM/DD/YYYY'); }).join(','));
                }
                else {
                    _this.calendar.setPreset(_this.innerValue.format('MM/DD/YYYY'));
                }
            }
            else {
                _this.calendar.setPreset('');
            }
        });
    };
    CalendarComponent.prototype.convertMomentArray = function (arr) {
        return arr ? arr.map(function (v) { return v.format('MM/DD/YYYY'); }).join(',') : null;
    };
    CalendarComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (_this.clonedElement) {
                var target_1 = _this.clonedElement;
                oldClasses.forEach(function (cls) {
                    target_1.removeClass(cls);
                });
                newClasses.forEach(function (cls) {
                    target_1.addClass(cls);
                });
            }
        });
    };
    var CalendarComponent_1;
    __decorate([
        Input('years-before'),
        __metadata("design:type", Number)
    ], CalendarComponent.prototype, "yearsBefore", void 0);
    __decorate([
        Input('years-after'),
        __metadata("design:type", Number)
    ], CalendarComponent.prototype, "yearsAfter", void 0);
    __decorate([
        Input('show'),
        __metadata("design:type", Object)
    ], CalendarComponent.prototype, "show", void 0);
    __decorate([
        Input('picker-mode'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "pickerMode", void 0);
    __decorate([
        Input('locale'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "locale", void 0);
    __decorate([
        Input('week-start'),
        __metadata("design:type", Number)
    ], CalendarComponent.prototype, "weekStart", void 0);
    __decorate([
        Input('outside'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "outside", void 0);
    __decorate([
        Input('buttons'),
        __metadata("design:type", Array)
    ], CalendarComponent.prototype, "buttons", void 0);
    __decorate([
        Input('ripple'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "ripple", void 0);
    __decorate([
        Input('ripple-color'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "rippleColor", void 0);
    __decorate([
        Input('exclude'),
        __metadata("design:type", Array)
    ], CalendarComponent.prototype, "exclude", void 0);
    __decorate([
        Input('include'),
        __metadata("design:type", Array)
    ], CalendarComponent.prototype, "include", void 0);
    __decorate([
        Input('min-date'),
        __metadata("design:type", Object)
    ], CalendarComponent.prototype, "minDate", void 0);
    __decorate([
        Input('max-date'),
        __metadata("design:type", Object)
    ], CalendarComponent.prototype, "maxDate", void 0);
    __decorate([
        Input('week-day-click'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "weekDayClick", void 0);
    __decorate([
        Input('multi-select'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "multiSelect", void 0);
    __decorate([
        Input('show-header'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "showHeader", void 0);
    __decorate([
        Input('show-footer'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "showFooter", void 0);
    __decorate([
        Input('wide-point'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "widePoint", void 0);
    __decorate([
        Input('wide'),
        __metadata("design:type", Boolean)
    ], CalendarComponent.prototype, "wide", void 0);
    __decorate([
        Input('cls-today'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsToday", void 0);
    __decorate([
        Input('cls-selected'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsSelected", void 0);
    __decorate([
        Input('cls-exclude'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsExclude", void 0);
    __decorate([
        Input('cls-cancel-button'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsCancelButton", void 0);
    __decorate([
        Input('cls-today-button'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsTodayButton", void 0);
    __decorate([
        Input('cls-clear-button'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsClearButton", void 0);
    __decorate([
        Input('cls-done-button'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsDoneButton", void 0);
    __decorate([
        Input('cls-calendar'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsCalendar", void 0);
    __decorate([
        Input('cls-calendar-header'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsCalendarHeader", void 0);
    __decorate([
        Input('cls-calendar-content'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsCalendarContent", void 0);
    __decorate([
        Input('cls-calendar-footer'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsCalendarFooter", void 0);
    __decorate([
        Input('cls-calendar-months'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsCalendarMonths", void 0);
    __decorate([
        Input('cls-calendar-years'),
        __metadata("design:type", String)
    ], CalendarComponent.prototype, "clsCalendarYears", void 0);
    __decorate([
        ViewChild('object', { static: true }),
        __metadata("design:type", ElementRef)
    ], CalendarComponent.prototype, "object", void 0);
    CalendarComponent = CalendarComponent_1 = __decorate([
        Component({
            selector: 'm4-calendar',
            template: "<div #object\r\n\r\n\r\n     [attr.data-years-before]=\"yearsBefore\"\r\n     [attr.data-years-after]=\"yearsAfter\"\r\n     [attr.data-show]=\"show?.format('MM/DD/YYYY')\"\r\n     [attr.data-picker-mode]=\"pickerMode && !multiSelect\"\r\n     [attr.data-locale]=\"locale\"\r\n     [attr.data-week-start]=\"weekStart\"\r\n     [attr.data-outside]=\"outside\"\r\n     [attr.data-buttons]=\"buttons\"\r\n     [attr.data-ripple]=\"ripple\"\r\n     [attr.data-ripple-color]=\"rippleColor\"\r\n     [attr.data-exclude]=\"convertMomentArray(exclude)\"\r\n     [attr.data-special]=\"convertMomentArray(include)\"\r\n     [attr.data-min-date]=\"minDate?.format('MM/DD/YYYY')\"\r\n     [attr.data-max-date]=\"maxDate?.format('MM/DD/YYYY')\"\r\n     [attr.data-week-day-click]=\"weekDayClick\"\r\n     [attr.data-multi-select]=\"multiSelect\"\r\n     [attr.data-show-header]=\"showHeader\"\r\n     [attr.data-show-footer]=\"showFooter\"\r\n     [attr.data-wide-point]=\"widePoint\"\r\n     [attr.data-wide]=\"wide\"\r\n\r\n     [attr.data-cls-today]=\"clsToday\"\r\n     [attr.data-cls-selected]=\"clsSelected\"\r\n     [attr.data-cls-exclude]=\"clsExclude\"\r\n     [attr.data-cls-cancel-button]=\"clsCancelButton\"\r\n     [attr.data-cls-today-button]=\"clsTodayButton\"\r\n     [attr.data-cls-clear-button]=\"clsClearButton\"\r\n     [attr.data-cls-done-button]=\"clsDoneButton\"\r\n     [attr.data-cls-calendar]=\"clsCalendar\"\r\n     [attr.data-cls-calendar-header]=\"clsCalendarHeader\"\r\n     [attr.data-cls-calendar-content]=\"clsCalendarContent\"\r\n     [attr.data-cls-calendar-footer]=\"clsCalendarFooter\"\r\n     [attr.data-cls-calendar-months]=\"clsCalendarMonths\"\r\n     [attr.data-cls-calendar-years]=\"clsCalendarYears\">\r\n</div>\r\n",
            providers: [DefaultValueAccessor.get(CalendarComponent_1), TypeAlias.get(CalendarComponent_1)],
            encapsulation: ViewEncapsulation.None,
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [".calendar.primary{border-color:#0366d6}.calendar.primary .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #0366d6!important}.calendar.primary .calendar-content .day.selected::after{border-top-color:#0366d6}.calendar.primary .calendar-content .day.today,.calendar.primary .calendar-header{background-color:#0372ef}.calendar.secondary{border-color:#607d8b}.calendar.secondary .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #607d8b!important}.calendar.secondary .calendar-content .day.selected::after{border-top-color:#607d8b}.calendar.secondary .calendar-content .day.today,.calendar.secondary .calendar-header{background-color:#6b8a99}.calendar.success{border-color:#60a917}.calendar.success .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #60a917!important}.calendar.success .calendar-content .day.selected::after{border-top-color:#60a917}.calendar.success .calendar-content .day.today,.calendar.success .calendar-header{background-color:#6dbf1a}.calendar.alert{border-color:#ce352c}.calendar.alert .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #ce352c!important}.calendar.alert .calendar-content .day.selected::after{border-top-color:#ce352c}.calendar.alert .calendar-content .day.today,.calendar.alert .calendar-header{background-color:#d6463e}.calendar.warning{border-color:#ff9447}.calendar.warning .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #ff9447!important}.calendar.warning .calendar-content .day.selected::after{border-top-color:#ff9447}.calendar.warning .calendar-content .day.today,.calendar.warning .calendar-header{background-color:#ffa361}.calendar.yellow{border-color:#ffe484}.calendar.yellow .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #ffe484!important}.calendar.yellow .calendar-content .day.selected::after{border-top-color:#ffe484}.calendar.yellow .calendar-content .day.today,.calendar.yellow .calendar-header{background-color:#ffea9e}.calendar.info{border-color:#5ebdec}.calendar.info .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #5ebdec!important}.calendar.info .calendar-content .day.selected::after{border-top-color:#5ebdec}.calendar.info .calendar-content .day.today,.calendar.info .calendar-header{background-color:#75c7ee}.calendar.dark{border-color:#505050}.calendar.dark .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #505050!important}.calendar.dark .calendar-content .day.selected::after{border-top-color:#505050}.calendar.dark .calendar-content .day.today,.calendar.dark .calendar-header{background-color:#5d5d5d}.calendar.light{border-color:#f8f8f8}.calendar.light .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #f8f8f8!important}.calendar.light .calendar-content .day.selected::after{border-top-color:#f8f8f8}.calendar.light .calendar-content .day.today,.calendar.light .calendar-header{background-color:#fff}.calendar.disabled .calendar-content{color:#e4e4e4!important}.calendar.disabled .calendar-footer button{color:#e4e4e4!important;opacity:.65}"]
        })
    ], CalendarComponent);
    return CalendarComponent;
}(ControlBase));
export { CalendarComponent };
