// ng-metro4/esm2015/lib/controls/progress/progress.component.js
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
let ProgressComponent = class ProgressComponent {
    constructor() { }
    createControl() {
        asapScheduler.schedule(() => {
            const originalElement = $(this.progress.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.progressObj = this.clonedElement.progress().data('progress');
        });
    }
    ngOnInit() {
        this.createControl();
    }
    ngOnChanges(changes) {
        if (Object.keys(changes).filter(c => c !== 'value' && c !== 'buffer').length > 0) {
            this.createControl();
        }
        else {
            if (this.progressObj) {
                this.progressObj.val(this.value);
                this.progressObj.buff(this.buffer);
            }
        }
    }
};
__decorate([
    Input(),
    __metadata("design:type", Number)
], ProgressComponent.prototype, "value", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], ProgressComponent.prototype, "buffer", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ProgressComponent.prototype, "small", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ProgressComponent.prototype, "type", void 0);
__decorate([
    Input('cls-back'),
    __metadata("design:type", String)
], ProgressComponent.prototype, "clsBack", void 0);
__decorate([
    Input('cls-bar'),
    __metadata("design:type", String)
], ProgressComponent.prototype, "clsBar", void 0);
__decorate([
    Input('cls-buffer'),
    __metadata("design:type", String)
], ProgressComponent.prototype, "clsBuffer", void 0);
__decorate([
    ViewChild('progress', { static: true }),
    __metadata("design:type", ElementRef)
], ProgressComponent.prototype, "progress", void 0);
ProgressComponent = __decorate([
    Component({
        selector: 'm4-progress',
        template: "<div #progress\r\n     [attr.data-value]=\"value\"\r\n     [attr.data-buffer]=\"buffer\"\r\n     [attr.data-small]=\"small\"\r\n     [attr.data-type]=\"type\"\r\n     [attr.data-cls-back]=\"clsBack\"\r\n     [attr.data-cls-bar]=\"clsBar\"\r\n     [attr.data-cls-buffer]=\"clsBuffer\"\r\n></div>\r\n",
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    }),
    __metadata("design:paramtypes", [])
], ProgressComponent);
export { ProgressComponent };
