// ng-metro4/esm2015/lib/base/ripple/ripple.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
import { AttributeHelper } from '../../helper/attribute-helper';
import { asapScheduler } from 'rxjs';
let RippleDirective = class RippleDirective {
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
    }
    createElement() {
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-ripple-target', this.rippleTarget);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-ripple-color', this.rippleColor);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-ripple-alpha', this.rippleAlpha);
        asapScheduler.schedule(() => {
            if (!this.rippleObj) {
                this.rippleObj = $(this.element.nativeElement).ripple().data('ripple');
            }
            else {
                this.rippleObj.options.rippleColor = this.rippleColor;
                this.rippleObj.options.rippleTarget = this.rippleTarget;
                this.rippleObj.options.rippleAlpha = this.rippleAlpha;
            }
        });
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
};
RippleDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
__decorate([
    Input('ripple-target'),
    __metadata("design:type", String)
], RippleDirective.prototype, "rippleTarget", void 0);
__decorate([
    Input('ripple-color'),
    __metadata("design:type", String)
], RippleDirective.prototype, "rippleColor", void 0);
__decorate([
    Input('ripple-alpha'),
    __metadata("design:type", Number)
], RippleDirective.prototype, "rippleAlpha", void 0);
RippleDirective = __decorate([
    Directive({
        selector: '[m4-ripple]'
    }),
    __metadata("design:paramtypes", [ElementRef, Renderer2])
], RippleDirective);
export { RippleDirective };
