// ng-metro4/esm2015/lib/form/calendar/calendar.component.js
var CalendarComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import * as moment from 'moment';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
const _moment = moment;
let CalendarComponent = CalendarComponent_1 = class CalendarComponent extends ControlBase {
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.object.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.calendar = this.clonedElement.calendar().data('calendar');
            asapScheduler.schedule(() => {
                this.clonedElement.find('.calendar-content .day, .calendar-content span, .calendar-footer button').on('click', () => {
                    this.touchCallback();
                    this.clonedElement.find('.calendar-content .day, .calendar-content span, .calendar-footer button').unbind('click');
                });
            });
            const selectDays = (selection) => {
                const dates = selection.map(s => _moment(s));
                if (this.multiSelect) {
                    this.changeValue(dates);
                }
                else {
                    this.changeValue(dates.length > 0 ? dates[0] : null);
                }
            };
            this.calendar.options.onDayClick = (selection) => {
                if (this.pickerMode) {
                    selectDays(selection);
                }
            };
            this.calendar.options.onDone = (selection) => {
                selectDays(selection);
            };
            complete();
        });
    }
    disable(disabled) {
        asapScheduler.schedule(() => {
            if (disabled) {
                this.clonedElement.addClass('disabled');
            }
            else {
                this.clonedElement.removeClass('disabled');
            }
        });
    }
    newValue() {
        if (!this.calendar) {
            return;
        }
        this.calendar.selected = [];
        asapScheduler.schedule(() => {
            if (this.innerValue) {
                if (this.multiSelect) {
                    this.calendar.setPreset(this.innerValue.map(v => v.format('MM/DD/YYYY')).join(','));
                }
                else {
                    this.calendar.setPreset(this.innerValue.format('MM/DD/YYYY'));
                }
            }
            else {
                this.calendar.setPreset('');
            }
        });
    }
    convertMomentArray(arr) {
        return arr ? arr.map(v => v.format('MM/DD/YYYY')).join(',') : null;
    }
    newClassValue(newClasses, oldClasses) {
        asapScheduler.schedule(() => {
            if (this.clonedElement) {
                const target = this.clonedElement;
                oldClasses.forEach((cls) => {
                    target.removeClass(cls);
                });
                newClasses.forEach((cls) => {
                    target.addClass(cls);
                });
            }
        });
    }
};
__decorate([
    Input('years-before'),
    __metadata("design:type", Number)
], CalendarComponent.prototype, "yearsBefore", void 0);
__decorate([
    Input('years-after'),
    __metadata("design:type", Number)
], CalendarComponent.prototype, "yearsAfter", void 0);
__decorate([
    Input('show'),
    __metadata("design:type", Object)
], CalendarComponent.prototype, "show", void 0);
__decorate([
    Input('picker-mode'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "pickerMode", void 0);
__decorate([
    Input('locale'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "locale", void 0);
__decorate([
    Input('week-start'),
    __metadata("design:type", Number)
], CalendarComponent.prototype, "weekStart", void 0);
__decorate([
    Input('outside'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "outside", void 0);
__decorate([
    Input('buttons'),
    __metadata("design:type", Array)
], CalendarComponent.prototype, "buttons", void 0);
__decorate([
    Input('ripple'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "ripple", void 0);
__decorate([
    Input('ripple-color'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "rippleColor", void 0);
__decorate([
    Input('exclude'),
    __metadata("design:type", Array)
], CalendarComponent.prototype, "exclude", void 0);
__decorate([
    Input('include'),
    __metadata("design:type", Array)
], CalendarComponent.prototype, "include", void 0);
__decorate([
    Input('min-date'),
    __metadata("design:type", Object)
], CalendarComponent.prototype, "minDate", void 0);
__decorate([
    Input('max-date'),
    __metadata("design:type", Object)
], CalendarComponent.prototype, "maxDate", void 0);
__decorate([
    Input('week-day-click'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "weekDayClick", void 0);
__decorate([
    Input('multi-select'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "multiSelect", void 0);
__decorate([
    Input('show-header'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "showHeader", void 0);
__decorate([
    Input('show-footer'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "showFooter", void 0);
__decorate([
    Input('wide-point'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "widePoint", void 0);
__decorate([
    Input('wide'),
    __metadata("design:type", Boolean)
], CalendarComponent.prototype, "wide", void 0);
__decorate([
    Input('cls-today'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsToday", void 0);
__decorate([
    Input('cls-selected'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsSelected", void 0);
__decorate([
    Input('cls-exclude'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsExclude", void 0);
__decorate([
    Input('cls-cancel-button'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsCancelButton", void 0);
__decorate([
    Input('cls-today-button'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsTodayButton", void 0);
__decorate([
    Input('cls-clear-button'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsClearButton", void 0);
__decorate([
    Input('cls-done-button'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsDoneButton", void 0);
__decorate([
    Input('cls-calendar'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsCalendar", void 0);
__decorate([
    Input('cls-calendar-header'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsCalendarHeader", void 0);
__decorate([
    Input('cls-calendar-content'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsCalendarContent", void 0);
__decorate([
    Input('cls-calendar-footer'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsCalendarFooter", void 0);
__decorate([
    Input('cls-calendar-months'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsCalendarMonths", void 0);
__decorate([
    Input('cls-calendar-years'),
    __metadata("design:type", String)
], CalendarComponent.prototype, "clsCalendarYears", void 0);
__decorate([
    ViewChild('object', { static: true }),
    __metadata("design:type", ElementRef)
], CalendarComponent.prototype, "object", void 0);
CalendarComponent = CalendarComponent_1 = __decorate([
    Component({
        selector: 'm4-calendar',
        template: "<div #object\r\n\r\n\r\n     [attr.data-years-before]=\"yearsBefore\"\r\n     [attr.data-years-after]=\"yearsAfter\"\r\n     [attr.data-show]=\"show?.format('MM/DD/YYYY')\"\r\n     [attr.data-picker-mode]=\"pickerMode && !multiSelect\"\r\n     [attr.data-locale]=\"locale\"\r\n     [attr.data-week-start]=\"weekStart\"\r\n     [attr.data-outside]=\"outside\"\r\n     [attr.data-buttons]=\"buttons\"\r\n     [attr.data-ripple]=\"ripple\"\r\n     [attr.data-ripple-color]=\"rippleColor\"\r\n     [attr.data-exclude]=\"convertMomentArray(exclude)\"\r\n     [attr.data-special]=\"convertMomentArray(include)\"\r\n     [attr.data-min-date]=\"minDate?.format('MM/DD/YYYY')\"\r\n     [attr.data-max-date]=\"maxDate?.format('MM/DD/YYYY')\"\r\n     [attr.data-week-day-click]=\"weekDayClick\"\r\n     [attr.data-multi-select]=\"multiSelect\"\r\n     [attr.data-show-header]=\"showHeader\"\r\n     [attr.data-show-footer]=\"showFooter\"\r\n     [attr.data-wide-point]=\"widePoint\"\r\n     [attr.data-wide]=\"wide\"\r\n\r\n     [attr.data-cls-today]=\"clsToday\"\r\n     [attr.data-cls-selected]=\"clsSelected\"\r\n     [attr.data-cls-exclude]=\"clsExclude\"\r\n     [attr.data-cls-cancel-button]=\"clsCancelButton\"\r\n     [attr.data-cls-today-button]=\"clsTodayButton\"\r\n     [attr.data-cls-clear-button]=\"clsClearButton\"\r\n     [attr.data-cls-done-button]=\"clsDoneButton\"\r\n     [attr.data-cls-calendar]=\"clsCalendar\"\r\n     [attr.data-cls-calendar-header]=\"clsCalendarHeader\"\r\n     [attr.data-cls-calendar-content]=\"clsCalendarContent\"\r\n     [attr.data-cls-calendar-footer]=\"clsCalendarFooter\"\r\n     [attr.data-cls-calendar-months]=\"clsCalendarMonths\"\r\n     [attr.data-cls-calendar-years]=\"clsCalendarYears\">\r\n</div>\r\n",
        providers: [DefaultValueAccessor.get(CalendarComponent_1), TypeAlias.get(CalendarComponent_1)],
        encapsulation: ViewEncapsulation.None,
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [".calendar.primary{border-color:#0366d6}.calendar.primary .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #0366d6!important}.calendar.primary .calendar-content .day.selected::after{border-top-color:#0366d6}.calendar.primary .calendar-content .day.today,.calendar.primary .calendar-header{background-color:#0372ef}.calendar.secondary{border-color:#607d8b}.calendar.secondary .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #607d8b!important}.calendar.secondary .calendar-content .day.selected::after{border-top-color:#607d8b}.calendar.secondary .calendar-content .day.today,.calendar.secondary .calendar-header{background-color:#6b8a99}.calendar.success{border-color:#60a917}.calendar.success .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #60a917!important}.calendar.success .calendar-content .day.selected::after{border-top-color:#60a917}.calendar.success .calendar-content .day.today,.calendar.success .calendar-header{background-color:#6dbf1a}.calendar.alert{border-color:#ce352c}.calendar.alert .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #ce352c!important}.calendar.alert .calendar-content .day.selected::after{border-top-color:#ce352c}.calendar.alert .calendar-content .day.today,.calendar.alert .calendar-header{background-color:#d6463e}.calendar.warning{border-color:#ff9447}.calendar.warning .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #ff9447!important}.calendar.warning .calendar-content .day.selected::after{border-top-color:#ff9447}.calendar.warning .calendar-content .day.today,.calendar.warning .calendar-header{background-color:#ffa361}.calendar.yellow{border-color:#ffe484}.calendar.yellow .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #ffe484!important}.calendar.yellow .calendar-content .day.selected::after{border-top-color:#ffe484}.calendar.yellow .calendar-content .day.today,.calendar.yellow .calendar-header{background-color:#ffea9e}.calendar.info{border-color:#5ebdec}.calendar.info .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #5ebdec!important}.calendar.info .calendar-content .day.selected::after{border-top-color:#5ebdec}.calendar.info .calendar-content .day.today,.calendar.info .calendar-header{background-color:#75c7ee}.calendar.dark{border-color:#505050}.calendar.dark .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #505050!important}.calendar.dark .calendar-content .day.selected::after{border-top-color:#505050}.calendar.dark .calendar-content .day.today,.calendar.dark .calendar-header{background-color:#5d5d5d}.calendar.light{border-color:#f8f8f8}.calendar.light .calendar-content .day.selected{box-shadow:inset 0 0 0 1px #f8f8f8!important}.calendar.light .calendar-content .day.selected::after{border-top-color:#f8f8f8}.calendar.light .calendar-content .day.today,.calendar.light .calendar-header{background-color:#fff}.calendar.disabled .calendar-content{color:#e4e4e4!important}.calendar.disabled .calendar-footer button{color:#e4e4e4!important;opacity:.65}"]
    })
], CalendarComponent);
export { CalendarComponent };
