// ng-metro4/esm2015/lib/base/button/button.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, HostBinding, HostListener, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
let ButtonDirective = class ButtonDirective {
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.oldClasses = [];
        this.jElement = $(this.element.nativeElement);
    }
    createElement() {
        if (!this.type) {
            this.type = 'button';
        }
        const buttonClass = this.specialBtn === 'command' ? 'command-button' :
            this.specialBtn === 'image' ? 'image-button' :
                this.specialBtn === 'shortcut' ? 'shortcut' :
                    this.specialBtn === 'ribbon' ? 'ribbon-button' :
                        this.specialBtn === 'action' || this.specialBtn === 'multi-action' ? 'action-button' : 'button';
        const newClasses = [buttonClass];
        if (this.btnStyle) {
            newClasses.push(this.btnStyle);
        }
        if (this.outline) {
            newClasses.push('outline');
        }
        if (this.size) {
            newClasses.push(this.size);
        }
        if (this.rounded) {
            newClasses.push('rounded');
        }
        if (this.shape) {
            newClasses.push(this.shape);
        }
        if (this.shadow) {
            newClasses.push('drop-shadow');
        }
        if (this.flat) {
            newClasses.push('flat-button');
        }
        this.oldClasses.forEach(c => {
            this.jElement.removeClass(c);
        });
        newClasses.forEach(c => {
            this.jElement.addClass(c);
        });
        this.oldClasses = newClasses;
    }
    clickEvent() {
        if (this.specialBtn === 'multi-action') {
            this.jElement.toggleClass('active');
        }
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
};
ButtonDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
__decorate([
    Input('btn-style'),
    __metadata("design:type", String)
], ButtonDirective.prototype, "btnStyle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonDirective.prototype, "outline", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ButtonDirective.prototype, "size", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonDirective.prototype, "rounded", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ButtonDirective.prototype, "shape", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonDirective.prototype, "shadow", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonDirective.prototype, "flat", void 0);
__decorate([
    Input('special-btn'),
    __metadata("design:type", String)
], ButtonDirective.prototype, "specialBtn", void 0);
__decorate([
    Input(), HostBinding('type'),
    __metadata("design:type", Object)
], ButtonDirective.prototype, "type", void 0);
__decorate([
    HostListener('click'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], ButtonDirective.prototype, "clickEvent", null);
ButtonDirective = __decorate([
    Directive({
        selector: 'button[m4-button]',
    }),
    __metadata("design:paramtypes", [ElementRef, Renderer2])
], ButtonDirective);
export { ButtonDirective };
