// ng-metro4/esm2015/lib/form/checkbox-group/checkbox-group.component.js
var CheckboxGroupComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ContentChildren, ElementRef, forwardRef, Input, QueryList, ViewChildren } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { CheckboxComponent } from '../checkbox/checkbox.component';
import { ArrayHelper } from '../../helper/array-helper';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
let CheckboxGroupComponent = CheckboxGroupComponent_1 = class CheckboxGroupComponent extends ControlBase {
    constructor(element, cdRef) {
        super(element, cdRef);
        this.readonly = false;
    }
    createControl() {
        return new Promise((complete) => {
            asapScheduler.schedule(() => {
                this.checkboxes = !!this.options ? this.checkboxesView : this.checkboxesContent;
                const checkboxCreations = this.checkboxes.map((item) => {
                    return new Promise((checkboxComplete) => {
                        item.updateProperty('readonly', this.readonly);
                        item.registerOnChange((v) => {
                            this.computeInnerValue();
                        });
                        item.registerOnTouched(() => {
                            this.touchCallback();
                        });
                        asapScheduler.schedule(() => {
                            item.createControl().then(() => {
                                checkboxComplete();
                            });
                        });
                    });
                });
                Promise.all(checkboxCreations).then(() => {
                    this.callNewValue();
                    complete();
                });
            });
        });
    }
    computeInnerValue() {
        const values = this.checkboxes.filter(item => item.innerValue === true && item.value).map(item => item.value);
        this.changeValue(values);
    }
    disable(disabled) {
        this.checkboxes.forEach((item) => {
            asapScheduler.schedule(() => {
                item.disable(disabled);
            });
        });
    }
    newValue() {
        if (!this.checkboxes) {
            return;
        }
        this.checkboxes.forEach((item) => {
            item.writeValue(this.innerValue && ArrayHelper.contains(this.innerValue, item.value));
        });
    }
    newClassValue(newClasses, oldClasses) {
        this.checkboxes.forEach((item) => {
            asapScheduler.schedule(() => {
                item.newClassValue(newClasses, oldClasses);
            }, 1);
        });
    }
};
CheckboxGroupComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: ChangeDetectorRef }
];
__decorate([
    Input(),
    __metadata("design:type", Object)
], CheckboxGroupComponent.prototype, "readonly", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], CheckboxGroupComponent.prototype, "options", void 0);
__decorate([
    ContentChildren(forwardRef(() => CheckboxComponent), { descendants: true }),
    __metadata("design:type", QueryList)
], CheckboxGroupComponent.prototype, "checkboxesContent", void 0);
__decorate([
    ViewChildren(forwardRef(() => CheckboxComponent)),
    __metadata("design:type", QueryList)
], CheckboxGroupComponent.prototype, "checkboxesView", void 0);
CheckboxGroupComponent = CheckboxGroupComponent_1 = __decorate([
    Component({
        selector: 'm4-checkbox-group',
        template: "<ng-container *ngIf=\"options; else noOptions\">\r\n  <m4-checkbox *ngFor=\"let option of options | objectKeys\" [value]=\"options[option]\" [caption]=\"option\"></m4-checkbox>\r\n</ng-container>\r\n<ng-template #noOptions>\r\n  <ng-content></ng-content>\r\n</ng-template>\r\n",
        providers: [DefaultValueAccessor.get(CheckboxGroupComponent_1), TypeAlias.get(CheckboxGroupComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    }),
    __metadata("design:paramtypes", [ElementRef, ChangeDetectorRef])
], CheckboxGroupComponent);
export { CheckboxGroupComponent };
