// ng-metro4/esm5/lib/base/animation/animation.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
var AnimationDirective = /** @class */ (function () {
    function AnimationDirective(element) {
        this.element = element;
        this.oldClasses = [];
        this.jElement = $(this.element.nativeElement);
    }
    AnimationDirective.prototype.createElement = function () {
        var _this = this;
        var newClasses = ["ani-" + (this.hover ? 'hover-' : '') + this.animation];
        this.oldClasses.forEach(function (c) {
            _this.jElement.removeClass(c);
        });
        newClasses.forEach(function (c) {
            _this.jElement.addClass(c);
        });
        this.oldClasses = newClasses;
    };
    AnimationDirective.prototype.ngOnInit = function () {
        this.createElement();
    };
    AnimationDirective.prototype.ngOnChanges = function (changes) {
        this.createElement();
    };
    AnimationDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    __decorate([
        Input('m4-animation'),
        __metadata("design:type", String)
    ], AnimationDirective.prototype, "animation", void 0);
    __decorate([
        Input('animation-hover'),
        __metadata("design:type", Boolean)
    ], AnimationDirective.prototype, "hover", void 0);
    AnimationDirective = __decorate([
        Directive({
            selector: '[m4-animation]'
        }),
        __metadata("design:paramtypes", [ElementRef])
    ], AnimationDirective);
    return AnimationDirective;
}());
export { AnimationDirective };
