// ng-metro4/esm2015/lib/base/let/let.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
let LetDirective = class LetDirective {
    constructor(vcRef, templateRef) {
        this.vcRef = vcRef;
        this.templateRef = templateRef;
        this.context = {};
    }
    set m4Let(context) {
        this.context.$implicit = this.context.ngVar = context;
        this.updateView();
    }
    updateView() {
        this.vcRef.clear();
        this.vcRef.createEmbeddedView(this.templateRef, this.context);
    }
};
LetDirective.ctorParameters = () => [
    { type: ViewContainerRef },
    { type: TemplateRef }
];
__decorate([
    Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], LetDirective.prototype, "m4Let", null);
LetDirective = __decorate([
    Directive({
        selector: '[m4Let]',
        exportAs: 'let'
    }),
    __metadata("design:paramtypes", [ViewContainerRef, TemplateRef])
], LetDirective);
export { LetDirective };
