// ng-metro4/esm2015/lib/form/radio-group/radio-group.component.js
var RadioGroupComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ContentChildren, ElementRef, forwardRef, Input, QueryList, ViewChildren } from '@angular/core';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ControlBase } from '../control-base';
import { RadioComponent } from '../radio/radio.component';
import { StringHelper } from '../../helper/string-helper';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
let RadioGroupComponent = RadioGroupComponent_1 = class RadioGroupComponent extends ControlBase {
    constructor(element, cdRef) {
        super(element, cdRef);
        this.name = StringHelper.guid();
    }
    createControl() {
        return new Promise((complete) => {
            asapScheduler.schedule(() => {
                this.radios = !!this.options ? this.radiosView : this.radiosContent;
                const radioCreations = this.radios.map((item) => {
                    return new Promise((radioComplete) => {
                        item.name = this.name;
                        item.registerOnChange((v) => {
                            this.changeValue(v);
                            this.newValue();
                        });
                        item.registerOnTouched(() => {
                            this.touchCallback();
                        });
                        asapScheduler.schedule(() => {
                            item.createControl().then(() => {
                                radioComplete();
                            });
                        }, 1);
                    });
                });
                Promise.all(radioCreations).then(() => {
                    this.callNewValue();
                    complete();
                });
            });
        });
    }
    disable(disabled) {
        this.radios.forEach((item) => {
            asapScheduler.schedule(() => {
                item.disable(disabled);
            }, 1);
        });
    }
    newValue() {
        if (!this.radios) {
            return;
        }
        this.radios.forEach((item) => {
            item.writeValue(this.innerValue);
        });
    }
    newClassValue(newClasses, oldClasses) {
        this.radios.forEach((item) => {
            asapScheduler.schedule(() => {
                item.newClassValue(newClasses, oldClasses);
            }, 1);
        });
    }
};
RadioGroupComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: ChangeDetectorRef }
];
__decorate([
    Input(),
    __metadata("design:type", Object)
], RadioGroupComponent.prototype, "options", void 0);
__decorate([
    ContentChildren(forwardRef(() => RadioComponent), { descendants: true }),
    __metadata("design:type", QueryList)
], RadioGroupComponent.prototype, "radiosContent", void 0);
__decorate([
    ViewChildren(forwardRef(() => RadioComponent)),
    __metadata("design:type", QueryList)
], RadioGroupComponent.prototype, "radiosView", void 0);
RadioGroupComponent = RadioGroupComponent_1 = __decorate([
    Component({
        selector: 'm4-radio-group',
        template: "<ng-container *ngIf=\"options; else noOptions\">\r\n  <m4-radio *ngFor=\"let option of options | objectKeys\" [value]=\"options[option]\"\r\n            [caption]=\"option\" [name]=\"name\"></m4-radio>\r\n</ng-container>\r\n<ng-template #noOptions>\r\n  <ng-content></ng-content>\r\n</ng-template>\r\n",
        providers: [DefaultValueAccessor.get(RadioGroupComponent_1), TypeAlias.get(RadioGroupComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    }),
    __metadata("design:paramtypes", [ElementRef, ChangeDetectorRef])
], RadioGroupComponent);
export { RadioGroupComponent };
