// ng-metro4/esm2015/lib/form/date-picker/date-picker.component.js
var DatePickerComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import * as moment from 'moment';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { ObjectHelper } from '../../helper/object-helper';
const _moment = moment;
let DatePickerComponent = DatePickerComponent_1 = class DatePickerComponent extends ControlBase {
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.parent().remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.datePicker = this.clonedElement.datepicker().data('datepicker');
            this.clonedElement.parent().find('.date-wrapper').one('click', () => {
                this.touchCallback();
            });
            this.datePicker.options.onSet = (val, elem_val) => {
                this.changeValue(_moment(elem_val, 'YYYY-MM-DD'));
            };
            complete();
        });
    }
    disable(disabled) {
        if (disabled) {
            this.clonedElement.parent().addClass('disabled');
        }
        else {
            this.clonedElement.parent().removeClass('disabled');
        }
    }
    newValue() {
        if (!this.datePicker) {
            return;
        }
        this.datePicker.val(this.innerValue ? this.innerValue.format('YYYY-MM-DD') : '');
    }
    newClassValue(newClasses, oldClasses) {
        if (this.clonedElement) {
            const target = this.clonedElement.parent();
            oldClasses.forEach((cls) => {
                target.removeClass(cls);
            });
            newClasses.forEach((cls) => {
                target.addClass(cls);
            });
        }
    }
};
__decorate([
    Input('month'),
    __metadata("design:type", Boolean)
], DatePickerComponent.prototype, "month", void 0);
__decorate([
    Input('day'),
    __metadata("design:type", Boolean)
], DatePickerComponent.prototype, "day", void 0);
__decorate([
    Input('year'),
    __metadata("design:type", Boolean)
], DatePickerComponent.prototype, "year", void 0);
__decorate([
    Input('locale'),
    __metadata("design:type", String)
], DatePickerComponent.prototype, "locale", void 0);
__decorate([
    Input('min-year'),
    __metadata("design:type", Number)
], DatePickerComponent.prototype, "minYear", void 0);
__decorate([
    Input('max-year'),
    __metadata("design:type", Number)
], DatePickerComponent.prototype, "maxYear", void 0);
__decorate([
    Input('distance'),
    __metadata("design:type", Number)
], DatePickerComponent.prototype, "distance", void 0);
__decorate([
    Input('cls-picker'),
    __metadata("design:type", String)
], DatePickerComponent.prototype, "clsPicker", void 0);
__decorate([
    Input('cls-start'),
    __metadata("design:type", String)
], DatePickerComponent.prototype, "clsStart", void 0);
__decorate([
    Input('cls-month'),
    __metadata("design:type", String)
], DatePickerComponent.prototype, "clsMonth", void 0);
__decorate([
    Input('cls-day'),
    __metadata("design:type", String)
], DatePickerComponent.prototype, "clsDay", void 0);
__decorate([
    Input('cls-year'),
    __metadata("design:type", String)
], DatePickerComponent.prototype, "clsYear", void 0);
__decorate([
    ViewChild('input', { static: true }),
    __metadata("design:type", ElementRef)
], DatePickerComponent.prototype, "input", void 0);
DatePickerComponent = DatePickerComponent_1 = __decorate([
    Component({
        selector: 'm4-date-picker',
        template: "<input type=\"text\" #input\r\n\r\n\r\n       [attr.data-month]=\"month\"\r\n       [attr.data-day]=\"day\"\r\n       [attr.data-year]=\"year\"\r\n       [attr.data-locale]=\"locale\"\r\n       [attr.data-min-year]=\"minYear\"\r\n       [attr.data-max-year]=\"maxYear\"\r\n       [attr.data-distance]=\"distance\"\r\n\r\n       [attr.data-cls-picker]=\"clsPicker\"\r\n       [attr.data-cls-start]=\"clsStart\"\r\n       [attr.data-cls-month]=\"clsMonth\"\r\n       [attr.data-cls-day]=\"clsDay\"\r\n       [attr.data-cls-year]=\"clsYear\">\r\n",
        providers: [DefaultValueAccessor.get(DatePickerComponent_1), TypeAlias.get(DatePickerComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        encapsulation: ViewEncapsulation.None,
        styles: [".date-picker.primary .date-wrapper{border-color:#0366d6}.date-picker.secondary .date-wrapper{border-color:#607d8b}.date-picker.success .date-wrapper{border-color:#60a917}.date-picker.alert .date-wrapper{border-color:#ce352c}.date-picker.warning .date-wrapper{border-color:#ff9447}.date-picker.yellow .date-wrapper{border-color:#ffe484}.date-picker.info .date-wrapper{border-color:#5ebdec}.date-picker.dark .date-wrapper{border-color:#505050}.date-picker.light .date-wrapper{border-color:#f8f8f8}"]
    })
], DatePickerComponent);
export { DatePickerComponent };
