// ng-metro4/esm5/lib/form/select/select.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
import { StringHelper } from '../../helper/string-helper';
var SelectComponent = /** @class */ (function (_super) {
    __extends(SelectComponent, _super);
    function SelectComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.multiple = false;
        _this.duration = 0;
        return _this;
    }
    SelectComponent_1 = SelectComponent;
    SelectComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.children().appendTo(originalElement);
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            originalElement.children().appendTo(_this.clonedElement);
            _this.select = _this.clonedElement.select().data('select');
            _this.clonedElement.parent().on('mousedown', function (ev) {
                if (ev.button === 0) {
                    _this.touchCallback();
                    _this.clonedElement.parent().off('mousedown');
                }
            });
            _this.select.options.onChange = function (val) {
                if (_this.options instanceof Array) {
                    var allOptions_1 = [];
                    _this.options.forEach(function (option) {
                        if (!!option.options) {
                            option.options.forEach(function (subOption) {
                                allOptions_1.push(subOption);
                            });
                        }
                        else {
                            allOptions_1.push(option);
                        }
                    });
                    val = val.map(function (key) { return allOptions_1.find(function (option) { return StringHelper.createHash(option.value) === key; }); }).filter(function (v) { return !!v; }).map(function (v) { return v.value; });
                }
                if (_this.multiple) {
                    _this.changeValue(val.slice(0));
                }
                else {
                    _this.changeValue(val[0]);
                }
            };
            if (_this.options && !(_this.options instanceof Array)) {
                _this.select.data(_this.options);
            }
            complete();
        });
    };
    SelectComponent.prototype.disable = function (disabled) {
        if (disabled) {
            this.select.disable();
        }
        else {
            this.select.enable();
        }
    };
    SelectComponent.prototype.newValue = function () {
        if (!this.select) {
            return;
        }
        var selectValue = this.multiple ? this.innerValue : [this.innerValue];
        if (this.options instanceof Array) {
            selectValue = selectValue.map(function (v) { return StringHelper.createHash(v); });
        }
        if (this.multiple) {
            this.select.reset();
        }
        this.select.val(selectValue);
    };
    SelectComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (_this.clonedElement) {
                var target_1 = _this.clonedElement.parent();
                oldClasses.forEach(function (cls) {
                    target_1.removeClass(cls);
                });
                newClasses.forEach(function (cls) {
                    target_1.addClass(cls);
                });
            }
        }, 1);
    };
    var SelectComponent_1;
    __decorate([
        Input('options'),
        __metadata("design:type", Object)
    ], SelectComponent.prototype, "options", void 0);
    __decorate([
        Input('multiple'),
        __metadata("design:type", Object)
    ], SelectComponent.prototype, "multiple", void 0);
    __decorate([
        Input('duration'),
        __metadata("design:type", Object)
    ], SelectComponent.prototype, "duration", void 0);
    __decorate([
        Input('prepend'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "prepend", void 0);
    __decorate([
        Input('append'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "append", void 0);
    __decorate([
        Input('drop-height'),
        __metadata("design:type", Number)
    ], SelectComponent.prototype, "dropHeight", void 0);
    __decorate([
        Input('filter'),
        __metadata("design:type", Boolean)
    ], SelectComponent.prototype, "filter", void 0);
    __decorate([
        Input('filter-placeholder'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "filterPlaceholder", void 0);
    __decorate([
        Input('cls-select'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsSelect", void 0);
    __decorate([
        Input('cls-prepend'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsPrepend", void 0);
    __decorate([
        Input('cls-append'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsAppend", void 0);
    __decorate([
        Input('cls-option'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsOption", void 0);
    __decorate([
        Input('cls-option-group'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsOptionGroup", void 0);
    __decorate([
        Input('cls-drop-list'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsDropList", void 0);
    __decorate([
        Input('cls-selected-item'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsSelectedItem", void 0);
    __decorate([
        Input('cls-selected-item-remover'),
        __metadata("design:type", String)
    ], SelectComponent.prototype, "clsSelectedItemRemover", void 0);
    __decorate([
        ViewChild('input', { static: true }),
        __metadata("design:type", ElementRef)
    ], SelectComponent.prototype, "input", void 0);
    SelectComponent = SelectComponent_1 = __decorate([
        Component({
            selector: 'm4-select',
            template: "<select #input [multiple]=\"multiple\"\r\n\r\n\r\n        [attr.data-prepend]=\"prepend\"\r\n        [attr.data-duration]=\"duration\"\r\n        [attr.data-append]=\"append\"\r\n        [attr.data-drop-height]=\"dropHeight\"\r\n        [attr.data-filter]=\"filter\"\r\n        [attr.data-filter-placeholder]=\"filterPlaceholder\"\r\n\r\n        [attr.data-cls-select]=\"clsSelect\"\r\n        [attr.data-cls-prepend]=\"clsPrepend\"\r\n        [attr.data-cls-append]=\"clsAppend\"\r\n        [attr.data-cls-option]=\"clsOption\"\r\n        [attr.data-cls-option-group]=\"clsOptionGroup\"\r\n        [attr.data-cls-drop-list]=\"clsDropList\"\r\n        [attr.data-cls-selected-item]=\"clsSelectedItem\"\r\n        [attr.data-cls-selected-item-remover]=\"clsSelectedItemRemover\">\r\n\r\n  <ng-content *ngIf=\"!options\"></ng-content>\r\n\r\n  <ng-container *ngIf=\"options | isArray\">\r\n    <ng-container *ngFor=\"let option of options\">\r\n      <option *ngIf=\"!option.options\" [value]=\"option.value | createHash\" [attr.data-template]=\"option.dataTemplate\">\r\n        {{option.title}}\r\n      </option>\r\n\r\n      <optgroup *ngIf=\"option.options\" [label]=\"option.groupName\">\r\n        <option *ngFor=\"let childOption of option.options\" [value]=\"childOption.value | createHash\" [attr.data-template]=\"childOption.dataTemplate\">\r\n          {{childOption.title}}\r\n        </option>\r\n      </optgroup>\r\n    </ng-container>\r\n  </ng-container>\r\n</select>\r\n",
            providers: [DefaultValueAccessor.get(SelectComponent_1), TypeAlias.get(SelectComponent_1)],
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        })
    ], SelectComponent);
    return SelectComponent;
}(ControlBase));
export { SelectComponent };
