// ng-metro4/esm5/lib/controls/donut/donut.component.js
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
var DonutComponent = /** @class */ (function () {
    function DonutComponent() {
    }
    DonutComponent.prototype.createControl = function () {
        var _this = this;
        asapScheduler.schedule(function () {
            var originalElement = $(_this.donut.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.donutObj = _this.clonedElement.donut().data('donut');
        });
    };
    DonutComponent.prototype.ngOnInit = function () {
        this.createControl();
    };
    DonutComponent.prototype.ngOnChanges = function (changes) {
        if (Object.keys(changes).filter(function (c) { return c !== 'value'; }).length > 0) {
            this.createControl();
        }
        else {
            if (this.donutObj) {
                this.donutObj.val(this.value);
            }
        }
    };
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], DonutComponent.prototype, "value", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], DonutComponent.prototype, "size", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], DonutComponent.prototype, "radius", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], DonutComponent.prototype, "hole", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], DonutComponent.prototype, "background", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], DonutComponent.prototype, "color", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], DonutComponent.prototype, "stroke", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], DonutComponent.prototype, "fill", void 0);
    __decorate([
        Input('font-size'),
        __metadata("design:type", Number)
    ], DonutComponent.prototype, "fontSize", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], DonutComponent.prototype, "total", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], DonutComponent.prototype, "caption", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], DonutComponent.prototype, "animate", void 0);
    __decorate([
        Input('show-value'),
        __metadata("design:type", Boolean)
    ], DonutComponent.prototype, "showValue", void 0);
    __decorate([
        ViewChild('donut', { static: true }),
        __metadata("design:type", ElementRef)
    ], DonutComponent.prototype, "donut", void 0);
    DonutComponent = __decorate([
        Component({
            selector: 'm4-donut',
            template: "<div #donut\r\n     [attr.data-value]=\"value\"\r\n     [attr.data-size]=\"size\"\r\n     [attr.data-radius]=\"radius\"\r\n     [attr.data-hole]=\"hole\"\r\n     [attr.data-background]=\"background\"\r\n     [attr.data-color]=\"color\"\r\n     [attr.data-stroke]=\"stroke\"\r\n     [attr.data-fill]=\"fill\"\r\n     [attr.data-font-size]=\"fontSize\"\r\n     [attr.data-total]=\"total\"\r\n     [attr.data-cap]=\"caption\"\r\n     [attr.data-animate]=\"animate\"\r\n     [attr.data-show-value]=\"showValue\"></div>\r\n",
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        }),
        __metadata("design:paramtypes", [])
    ], DonutComponent);
    return DonutComponent;
}());
export { DonutComponent };
