// ng-metro4/esm5/lib/form/calendar-picker/calendar-picker.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ControlBase } from '../control-base';
import * as moment from 'moment';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
var _moment = moment;
var CalendarPickerComponent = /** @class */ (function (_super) {
    __extends(CalendarPickerComponent, _super);
    function CalendarPickerComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CalendarPickerComponent_1 = CalendarPickerComponent;
    CalendarPickerComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.calendarPicker = _this.clonedElement.calendarpicker().data('calendarpicker');
            _this.clonedElement.one('blur', function () {
                _this.touchCallback();
            });
            _this.calendarPicker.options.onChange = function () {
                asapScheduler.schedule(function () {
                    var val = _this.calendarPicker.value;
                    _this.changeValue(val ? _moment(val.toLocaleDateString('en'), 'MM/DD/YYYY') : null);
                });
            };
            complete();
        });
    };
    CalendarPickerComponent.prototype.disable = function (disabled) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (disabled) {
                _this.calendarPicker.disable();
            }
            else {
                _this.calendarPicker.enable();
            }
        });
    };
    CalendarPickerComponent.prototype.newValue = function () {
        var _this = this;
        if (!this.calendarPicker) {
            return;
        }
        asapScheduler.schedule(function () {
            _this.calendarPicker.val(_this.innerValue ? _this.innerValue.format('MM/DD/YYYY') : '01/01/0000');
        });
    };
    CalendarPickerComponent.prototype.convertMomentArray = function (arr) {
        return arr ? arr.map(function (v) { return v.format('MM/DD/YYYY'); }).join(',') : null;
    };
    CalendarPickerComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (_this.clonedElement) {
                var target_1 = _this.clonedElement.parent();
                oldClasses.forEach(function (cls) {
                    target_1.removeClass(cls);
                });
                newClasses.forEach(function (cls) {
                    target_1.addClass(cls);
                });
            }
        });
    };
    var CalendarPickerComponent_1;
    __decorate([
        Input('calendar-wide'),
        __metadata("design:type", Boolean)
    ], CalendarPickerComponent.prototype, "calendarWide", void 0);
    __decorate([
        Input('calendar-wide-point'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "calendarWidePoint", void 0);
    __decorate([
        Input('dialog-mode'),
        __metadata("design:type", Boolean)
    ], CalendarPickerComponent.prototype, "dialogMode", void 0);
    __decorate([
        Input('dialog-point'),
        __metadata("design:type", Number)
    ], CalendarPickerComponent.prototype, "dialogPoint", void 0);
    __decorate([
        Input('dialog-overlay'),
        __metadata("design:type", Boolean)
    ], CalendarPickerComponent.prototype, "dialogOverlay", void 0);
    __decorate([
        Input('overlay-color'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "overlayColor", void 0);
    __decorate([
        Input('overlay-alpha'),
        __metadata("design:type", Number)
    ], CalendarPickerComponent.prototype, "overlayAlpha", void 0);
    __decorate([
        Input('locale'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "locale", void 0);
    __decorate([
        Input('size'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "size", void 0);
    __decorate([
        Input('format'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "format", void 0);
    __decorate([
        Input('clear-button'),
        __metadata("design:type", Boolean)
    ], CalendarPickerComponent.prototype, "clearButton", void 0);
    __decorate([
        Input('clear-button-icon'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clearButtonIcon", void 0);
    __decorate([
        Input('calendar-button-icon'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "calendarButtonIcon", void 0);
    __decorate([
        Input('years-before'),
        __metadata("design:type", Number)
    ], CalendarPickerComponent.prototype, "yearsBefore", void 0);
    __decorate([
        Input('years-after'),
        __metadata("design:type", Number)
    ], CalendarPickerComponent.prototype, "yearsAfter", void 0);
    __decorate([
        Input('week-start'),
        __metadata("design:type", Number)
    ], CalendarPickerComponent.prototype, "weekStart", void 0);
    __decorate([
        Input('outside'),
        __metadata("design:type", Boolean)
    ], CalendarPickerComponent.prototype, "outside", void 0);
    __decorate([
        Input('ripple'),
        __metadata("design:type", Boolean)
    ], CalendarPickerComponent.prototype, "ripple", void 0);
    __decorate([
        Input('ripple-color'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "rippleColor", void 0);
    __decorate([
        Input('exclude'),
        __metadata("design:type", Array)
    ], CalendarPickerComponent.prototype, "exclude", void 0);
    __decorate([
        Input('include'),
        __metadata("design:type", Array)
    ], CalendarPickerComponent.prototype, "include", void 0);
    __decorate([
        Input('min-date'),
        __metadata("design:type", Object)
    ], CalendarPickerComponent.prototype, "minDate", void 0);
    __decorate([
        Input('max-date'),
        __metadata("design:type", Object)
    ], CalendarPickerComponent.prototype, "maxDate", void 0);
    __decorate([
        Input('show-header'),
        __metadata("design:type", Boolean)
    ], CalendarPickerComponent.prototype, "showHeader", void 0);
    __decorate([
        Input('cls-picker'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsPicker", void 0);
    __decorate([
        Input('cls-today'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsToday", void 0);
    __decorate([
        Input('cls-selected'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsSelected", void 0);
    __decorate([
        Input('cls-exclude'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsExclude", void 0);
    __decorate([
        Input('cls-calendar'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsCalendar", void 0);
    __decorate([
        Input('cls-calendar-header'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsCalendarHeader", void 0);
    __decorate([
        Input('cls-calendar-content'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsCalendarContent", void 0);
    __decorate([
        Input('cls-calendar-months'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsCalendarMonths", void 0);
    __decorate([
        Input('cls-calendar-years'),
        __metadata("design:type", String)
    ], CalendarPickerComponent.prototype, "clsCalendarYears", void 0);
    __decorate([
        ViewChild('input', { static: true }),
        __metadata("design:type", ElementRef)
    ], CalendarPickerComponent.prototype, "input", void 0);
    CalendarPickerComponent = CalendarPickerComponent_1 = __decorate([
        Component({
            selector: 'm4-calendar-picker',
            template: "<input type=\"text\" #input\r\n\r\n       data-input-format=\"%m/%d/%Y\"\r\n\r\n       [attr.data-calendar-wide]=\"calendarWide\"\r\n       [attr.data-calendar-wide-point]=\"calendarWidePoint\"\r\n       [attr.data-dialog-mode]=\"dialogMode\"\r\n       [attr.data-dialog-point]=\"dialogPoint\"\r\n       [attr.data-dialog-overlay]=\"dialogOverlay\"\r\n       [attr.data-overlay-color]=\"overlayColor\"\r\n       [attr.data-overlay-alpha]=\"overlayAlpha\"\r\n       [attr.data-locale]=\"locale\"\r\n       [attr.data-size]=\"size\"\r\n       [attr.data-format]=\"format\"\r\n       [attr.data-clear-button]=\"clearButton\"\r\n       [attr.data-clear-button-icon]=\"clearButtonIcon\"\r\n       [attr.data-calendar-button-icon]=\"calendarButtonIcon\"\r\n       [attr.data-years-before]=\"yearsBefore\"\r\n       [attr.data-years-after]=\"yearsAfter\"\r\n       [attr.data-week-start]=\"weekStart\"\r\n       [attr.data-outside]=\"outside\"\r\n       [attr.data-ripple]=\"ripple\"\r\n       [attr.data-ripple-color]=\"rippleColor\"\r\n       [attr.data-exclude]=\"convertMomentArray(exclude)\"\r\n       [attr.data-special]=\"convertMomentArray(include)\"\r\n       [attr.data-min-date]=\"minDate?.format('MM/DD/YYYY')\"\r\n       [attr.data-max-date]=\"maxDate?.format('MM/DD/YYYY')\"\r\n       [attr.data-show-header]=\"showHeader\"\r\n\r\n       [attr.data-cls-picker]=\"clsPicker\"\r\n       [attr.data-cls-today]=\"clsToday\"\r\n       [attr.data-cls-selected]=\"clsSelected\"\r\n       [attr.data-cls-exclude]=\"clsExclude\"\r\n       [attr.data-cls-calendar]=\"clsCalendar\"\r\n       [attr.data-cls-calendar-header]=\"clsCalendarHeader\"\r\n       [attr.data-cls-calendar-content]=\"clsCalendarContent\"\r\n       [attr.data-cls-calendar-months]=\"clsCalendarMonths\"\r\n       [attr.data-cls-calendar-years]=\"clsCalendarYears\">\r\n",
            providers: [DefaultValueAccessor.get(CalendarPickerComponent_1), TypeAlias.get(CalendarPickerComponent_1)],
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        })
    ], CalendarPickerComponent);
    return CalendarPickerComponent;
}(ControlBase));
export { CalendarPickerComponent };
