// ng-metro4/esm5/lib/base/ripple/ripple.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
import { AttributeHelper } from '../../helper/attribute-helper';
import { asapScheduler } from 'rxjs';
var RippleDirective = /** @class */ (function () {
    function RippleDirective(element, renderer) {
        this.element = element;
        this.renderer = renderer;
    }
    RippleDirective.prototype.createElement = function () {
        var _this = this;
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-ripple-target', this.rippleTarget);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-ripple-color', this.rippleColor);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-ripple-alpha', this.rippleAlpha);
        asapScheduler.schedule(function () {
            if (!_this.rippleObj) {
                _this.rippleObj = $(_this.element.nativeElement).ripple().data('ripple');
            }
            else {
                _this.rippleObj.options.rippleColor = _this.rippleColor;
                _this.rippleObj.options.rippleTarget = _this.rippleTarget;
                _this.rippleObj.options.rippleAlpha = _this.rippleAlpha;
            }
        });
    };
    RippleDirective.prototype.ngOnInit = function () {
        this.createElement();
    };
    RippleDirective.prototype.ngOnChanges = function (changes) {
        this.createElement();
    };
    RippleDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    __decorate([
        Input('ripple-target'),
        __metadata("design:type", String)
    ], RippleDirective.prototype, "rippleTarget", void 0);
    __decorate([
        Input('ripple-color'),
        __metadata("design:type", String)
    ], RippleDirective.prototype, "rippleColor", void 0);
    __decorate([
        Input('ripple-alpha'),
        __metadata("design:type", Number)
    ], RippleDirective.prototype, "rippleAlpha", void 0);
    RippleDirective = __decorate([
        Directive({
            selector: '[m4-ripple]'
        }),
        __metadata("design:paramtypes", [ElementRef, Renderer2])
    ], RippleDirective);
    return RippleDirective;
}());
export { RippleDirective };
