// ng-metro4/esm5/lib/base/loading/loading-display.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnDestroy, TemplateRef, ViewContainerRef } from '@angular/core';
import { LoadingDirective } from './loading.directive';
var LoadingDisplayDirective = /** @class */ (function () {
    function LoadingDisplayDirective(element, templateRef, viewContainer, loadingDirective) {
        var _this = this;
        this.element = element;
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
        this.loadingDirective = loadingDirective;
        this.subscription = this.loadingDirective.stateChange.subscribe(function (loading) {
            _this.loadingState = loading;
            _this.updateView();
        });
    }
    Object.defineProperty(LoadingDisplayDirective.prototype, "loading", {
        set: function (val) {
            this.conditionValue = val;
            this.updateView();
        },
        enumerable: true,
        configurable: true
    });
    LoadingDisplayDirective.prototype.updateView = function () {
        this.viewContainer.clear();
        if (this.loadingState || this.conditionValue) {
            this.viewContainer.createEmbeddedView(this.templateRef);
        }
    };
    LoadingDisplayDirective.prototype.ngOnDestroy = function () {
        if (this.subscription) {
            this.subscription.unsubscribe();
        }
    };
    LoadingDisplayDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: LoadingDirective }
    ]; };
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], LoadingDisplayDirective.prototype, "loading", null);
    LoadingDisplayDirective = __decorate([
        Directive({
            selector: '[loading]'
        }),
        __metadata("design:paramtypes", [ElementRef,
            TemplateRef,
            ViewContainerRef,
            LoadingDirective])
    ], LoadingDisplayDirective);
    return LoadingDisplayDirective;
}());
export { LoadingDisplayDirective };
