// ng-metro4/esm5/lib/information/dialog/dialog.component.js
import { __decorate, __metadata } from "tslib";
import { AfterViewInit, ChangeDetectionStrategy, Component, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, Output, SimpleChanges, ViewChild } from '@angular/core';
import { Subject } from 'rxjs';
import { finalize, take } from 'rxjs/operators';
var DialogComponent = /** @class */ (function () {
    function DialogComponent(element) {
        this.element = element;
        this.isOpen = false;
        this.closeEvent = new EventEmitter();
        this.closeSubject$ = new Subject();
    }
    DialogComponent.prototype.open = function () {
        var _this = this;
        this.dialogObj.open();
        return this.closeSubject$.pipe(take(1), finalize(function () { return _this.close(); }));
    };
    DialogComponent.prototype.close = function () {
        if (this.dialogObj) {
            this.dialogObj.close();
        }
    };
    DialogComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        var dialogOptions = {
            width: this.width,
            show: this.isOpen,
            overlay: this.overlay,
            overlayClickClose: this.overlayClickClose
        };
        if (this.overlayColor) {
            dialogOptions.overlayColor = this.overlayColor;
        }
        if (this.overlayAlpha) {
            dialogOptions.overlayAlpha = this.overlayAlpha;
        }
        this.dialogObj = $(this.dialog.nativeElement).dialog(dialogOptions).data('dialog');
        this.dialogObj.options.onClose = function () {
            _this.closeEvent.emit();
            _this.closeSubject$.next(_this.data);
        };
        this.observeClassValue();
    };
    DialogComponent.prototype.observeClassValue = function () {
        var _this = this;
        var classValueCallback = function () {
            var classValue = _this.element.nativeElement.getAttribute('class');
            _this.dialogObj.element.attr('class', (classValue ? classValue + ' ' : '') + "dialog");
        };
        this.classObserver = new MutationObserver(classValueCallback);
        this.classObserver.observe(this.element.nativeElement, {
            attributeFilter: ['class'],
            attributes: true
        });
        classValueCallback();
    };
    DialogComponent.prototype.ngOnChanges = function (changes) {
        if (this.dialogObj) {
            if (changes['isOpen']) {
                if (this.isOpen) {
                    this.open();
                }
                else {
                    this.close();
                }
            }
            if (changes['width']) {
                this.dialogObj.element.css('width', this.width + "px");
            }
            if (changes['overlay'] || changes['overlayColor'] || changes['overlayAlpha'] || changes['overlayClickClose']) {
                this.dialogObj.options.overlay = this.overlay;
                this.dialogObj.options.overlayColor = this.overlayColor;
                this.dialogObj.options.overlayAlpha = this.overlayAlpha;
                this.dialogObj.options.overlayClickClose = this.overlayClickClose;
            }
        }
    };
    DialogComponent.prototype.ngOnDestroy = function () {
        if (this.dialogObj) {
            this.close();
            this.dialogObj.element.remove();
        }
        if (this.classObserver) {
            this.classObserver.disconnect();
        }
    };
    DialogComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    __decorate([
        Input('open'),
        __metadata("design:type", Object)
    ], DialogComponent.prototype, "isOpen", void 0);
    __decorate([
        Input('width'),
        __metadata("design:type", String)
    ], DialogComponent.prototype, "width", void 0);
    __decorate([
        Input('overlay'),
        __metadata("design:type", Boolean)
    ], DialogComponent.prototype, "overlay", void 0);
    __decorate([
        Input('overlay-color'),
        __metadata("design:type", String)
    ], DialogComponent.prototype, "overlayColor", void 0);
    __decorate([
        Input('overlay-alpha'),
        __metadata("design:type", String)
    ], DialogComponent.prototype, "overlayAlpha", void 0);
    __decorate([
        Input('overlay-click-close'),
        __metadata("design:type", Boolean)
    ], DialogComponent.prototype, "overlayClickClose", void 0);
    __decorate([
        Input('data'),
        __metadata("design:type", Object)
    ], DialogComponent.prototype, "data", void 0);
    __decorate([
        ViewChild('dialog', { static: true }),
        __metadata("design:type", ElementRef)
    ], DialogComponent.prototype, "dialog", void 0);
    __decorate([
        Output('close-event'),
        __metadata("design:type", Object)
    ], DialogComponent.prototype, "closeEvent", void 0);
    DialogComponent = __decorate([
        Component({
            selector: 'm4-dialog',
            template: "<div #dialog>\r\n  <div class=\"dialog-title\">\r\n    <ng-content select=\"[dialog-title]\"></ng-content>\r\n  </div>\r\n  <div class=\"dialog-content\">\r\n    <ng-content select=\"[dialog-content]\"></ng-content>\r\n  </div>\r\n  <div class=\"dialog-actions\">\r\n    <ng-content select=\"[dialog-actions]\"></ng-content>\r\n  </div>\r\n</div>\r\n",
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        }),
        __metadata("design:paramtypes", [ElementRef])
    ], DialogComponent);
    return DialogComponent;
}());
export { DialogComponent };
