// ng-metro4/esm5/lib/base/let/let.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
var LetDirective = /** @class */ (function () {
    function LetDirective(vcRef, templateRef) {
        this.vcRef = vcRef;
        this.templateRef = templateRef;
        this.context = {};
    }
    Object.defineProperty(LetDirective.prototype, "m4Let", {
        set: function (context) {
            this.context.$implicit = this.context.ngVar = context;
            this.updateView();
        },
        enumerable: true,
        configurable: true
    });
    LetDirective.prototype.updateView = function () {
        this.vcRef.clear();
        this.vcRef.createEmbeddedView(this.templateRef, this.context);
    };
    LetDirective.ctorParameters = function () { return [
        { type: ViewContainerRef },
        { type: TemplateRef }
    ]; };
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], LetDirective.prototype, "m4Let", null);
    LetDirective = __decorate([
        Directive({
            selector: '[m4Let]',
            exportAs: 'let'
        }),
        __metadata("design:paramtypes", [ViewContainerRef, TemplateRef])
    ], LetDirective);
    return LetDirective;
}());
export { LetDirective };
