// ng-metro4/esm5/lib/controls/panel/panel.component.js
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
var PanelComponent = /** @class */ (function () {
    function PanelComponent() {
    }
    PanelComponent.prototype.createElement = function () {
        var _this = this;
        asapScheduler.schedule(function () {
            var originalElement = $(_this.panel.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.children().appendTo(originalElement);
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            originalElement.children().appendTo(_this.clonedElement);
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.panelObj = _this.clonedElement.panel({
                customButtons: _this.customButtons
            }).data('panel');
        });
    };
    PanelComponent.prototype.ngOnInit = function () {
        this.createElement();
    };
    PanelComponent.prototype.ngOnChanges = function (changes) {
        this.createElement();
    };
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], PanelComponent.prototype, "width", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], PanelComponent.prototype, "height", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "title", void 0);
    __decorate([
        Input('title-icon'),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "titleIcon", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], PanelComponent.prototype, "collapsible", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], PanelComponent.prototype, "collapsed", void 0);
    __decorate([
        Input('custom-buttons'),
        __metadata("design:type", Array)
    ], PanelComponent.prototype, "customButtons", void 0);
    __decorate([
        Input('cls-panel'),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "clsPanel", void 0);
    __decorate([
        Input('cls-title'),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "clsTitle", void 0);
    __decorate([
        Input('cls-title-caption'),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "clsTitleCaption", void 0);
    __decorate([
        Input('cls-title-icon'),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "clsTitleIcon", void 0);
    __decorate([
        Input('cls-content'),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "clsContent", void 0);
    __decorate([
        Input('cls-collapse-toggle'),
        __metadata("design:type", String)
    ], PanelComponent.prototype, "clsCollapseToggle", void 0);
    __decorate([
        ViewChild('panel', { static: true }),
        __metadata("design:type", ElementRef)
    ], PanelComponent.prototype, "panel", void 0);
    PanelComponent = __decorate([
        Component({
            selector: 'm4-panel',
            template: "<div #panel\r\n     [attr.data-width]=\"width\"\r\n     [attr.data-height]=\"height\"\r\n     [attr.data-title-caption]=\"title\"\r\n     [attr.data-title-icon]=\"titleIcon\"\r\n     [attr.data-collapsible]=\"collapsible\"\r\n     [attr.data-collapsed]=\"collapsed\"\r\n\r\n     [attr.data-cls-panel]=\"clsPanel\"\r\n     [attr.data-cls-title]=\"clsTitle\"\r\n     [attr.data-cls-title-caption]=\"clsTitleCaption\"\r\n     [attr.data-cls-title-icon]=\"clsTitleIcon\"\r\n     [attr.data-cls-content]=\"clsContent\"\r\n     [attr.data-cls-collapse-toggle]=\"clsCollapseToggle\">\r\n  <ng-content></ng-content>\r\n</div>\r\n",
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        }),
        __metadata("design:paramtypes", [])
    ], PanelComponent);
    return PanelComponent;
}());
export { PanelComponent };
