// ng-metro4/esm2015/lib/controls/popover/popover.component.js
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ContentChild, ElementRef, ViewChild } from '@angular/core';
import { PopoverDirective } from './popover.directive';
import { asapScheduler } from 'rxjs';
let PopoverComponent = class PopoverComponent {
    ngAfterViewInit() {
        asapScheduler.schedule(() => {
            this.popover.popoverObj.options.onPopoverShow = () => {
                if (this.popover.popoverObj.popover) {
                    $(this.content.nativeElement).children().appendTo(this.popover.popoverObj.popover.find('.popover-content'));
                    this.popover.popoverObj.size = Metro.utils.hiddenElementSize(this.popover.popoverObj.popover);
                    this.popover.popoverObj.setPosition();
                }
            };
            this.popover.popoverObj.options.onPopoverHide = () => {
                asapScheduler.schedule(() => {
                    if (this.popover.popoverObj && this.popover.popoverObj.popover) {
                        this.popover.popoverObj.popover.find('.popover-content').children().appendTo($(this.content.nativeElement));
                    }
                }, 300);
            };
        }, 1);
    }
};
__decorate([
    ContentChild(PopoverDirective, { static: true }),
    __metadata("design:type", PopoverDirective)
], PopoverComponent.prototype, "popover", void 0);
__decorate([
    ViewChild('popoverContent', { static: true }),
    __metadata("design:type", ElementRef)
], PopoverComponent.prototype, "content", void 0);
PopoverComponent = __decorate([
    Component({
        selector: 'm4-popover',
        template: "<ng-content></ng-content>\r\n\r\n<div class=\"m4-popover-content\" #popoverContent>\r\n  <ng-content select=\"[content]\"></ng-content>\r\n</div>\r\n",
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [".m4-popover-content{display:none}"]
    })
], PopoverComponent);
export { PopoverComponent };
