// ng-metro4/esm2015/lib/controls/popover/popover.directive.js
import { __decorate, __metadata, __param } from "tslib";
import { Directive, ElementRef, Inject, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import { AttributeHelper } from '../../helper/attribute-helper';
import { asapScheduler } from 'rxjs';
let PopoverDirective = class PopoverDirective {
    constructor(element, renderer, document) {
        this.element = element;
        this.renderer = renderer;
        this.document = document;
        this.popoverPosition = 'top';
    }
    createElement() {
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-text', this.popoverText);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-position', this.popoverPosition);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-trigger', this.popoverTrigger);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-hide', this.popoverHide);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-cls-popover', this.clsPopover);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-close-button', this.closeBtn);
        asapScheduler.schedule(() => {
            if (!this.popoverObj) {
                this.popoverObj = $(this.element.nativeElement).popover().data('popover');
            }
            else {
                this.popoverObj.options.popoverText = this.popoverText;
                this.popoverObj.options.popoverPosition = this.popoverPosition;
                this.popoverObj.options.popoverTrigger = this.popoverTrigger;
                this.popoverObj.options.popoverHide = this.popoverHide;
                this.popoverObj.options.closeButton = this.closeBtn;
                this.popoverObj.options.clsPopover = this.clsPopover;
            }
        });
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
};
PopoverDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
__decorate([
    Input('m4-popover'),
    __metadata("design:type", String)
], PopoverDirective.prototype, "popoverText", void 0);
__decorate([
    Input('popover-position'),
    __metadata("design:type", String)
], PopoverDirective.prototype, "popoverPosition", void 0);
__decorate([
    Input('popover-trigger'),
    __metadata("design:type", String)
], PopoverDirective.prototype, "popoverTrigger", void 0);
__decorate([
    Input('popover-hide'),
    __metadata("design:type", Number)
], PopoverDirective.prototype, "popoverHide", void 0);
__decorate([
    Input('popover-close-button'),
    __metadata("design:type", Boolean)
], PopoverDirective.prototype, "closeBtn", void 0);
__decorate([
    Input('cls-popover'),
    __metadata("design:type", Number)
], PopoverDirective.prototype, "clsPopover", void 0);
PopoverDirective = __decorate([
    Directive({
        selector: '[m4-popover]'
    }),
    __param(2, Inject(DOCUMENT)),
    __metadata("design:paramtypes", [ElementRef, Renderer2, Object])
], PopoverDirective);
export { PopoverDirective };
