// ng-metro4/esm2015/lib/controls/draggable/draggable.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, Renderer2, SimpleChanges } from '@angular/core';
import { asapScheduler } from 'rxjs';
let DraggableDirective = class DraggableDirective {
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.dragStart = new EventEmitter();
        this.dragStop = new EventEmitter();
        this.dragMove = new EventEmitter();
    }
    createElement() {
        asapScheduler.schedule(() => {
            if (this.draggableData) {
                if (this.dragArea) {
                    this.draggableData.elem.remove();
                }
                this.draggableData.destroy();
            }
            const draggableOptions = {
                onDragStart: (position) => {
                    this.dragStart.emit(position);
                },
                onDragStop: (position) => {
                    this.dragStop.emit(position);
                },
                onDragMove: (position) => {
                    this.dragMove.emit(position);
                }
            };
            if (this.dragElement) {
                draggableOptions.dragElement = this.dragElement;
            }
            if (this.dragArea) {
                draggableOptions.dragArea = this.dragArea;
            }
            this.draggableData = $(this.element.nativeElement).draggable(draggableOptions).data('draggable');
        });
    }
    setPosition(position) {
        if (this.draggableData) {
            $(this.draggableData.elem).css({
                left: position.x + 'px',
                top: position.y + 'px'
            });
        }
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
    ngOnDestroy() {
        $(this.element.nativeElement).remove();
    }
};
DraggableDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
__decorate([
    Input('m4-draggable'),
    __metadata("design:type", String)
], DraggableDirective.prototype, "dragElement", void 0);
__decorate([
    Input('drag-area'),
    __metadata("design:type", String)
], DraggableDirective.prototype, "dragArea", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], DraggableDirective.prototype, "dragStart", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], DraggableDirective.prototype, "dragStop", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], DraggableDirective.prototype, "dragMove", void 0);
DraggableDirective = __decorate([
    Directive({
        selector: '[m4-draggable]'
    }),
    __metadata("design:paramtypes", [ElementRef, Renderer2])
], DraggableDirective);
export { DraggableDirective };
