// ng-metro4/esm2015/lib/form/rating/rating.component.js
var RatingComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
let RatingComponent = RatingComponent_1 = class RatingComponent extends ControlBase {
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.parent().remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.rating = this.clonedElement.rating().data('rating');
            this.clonedElement.parent().find('ul li').one('click', () => {
                this.touchCallback();
            });
            this.clonedElement.on('change', () => {
                const newValue = this.clonedElement.val();
                const valueParsed = +newValue;
                if (!Number.isNaN(valueParsed)) {
                    this.changeValue(valueParsed);
                }
                else {
                    this.changeValue(newValue);
                }
            });
            complete();
        });
    }
    disable(disabled) {
        if (disabled) {
            this.clonedElement.parent().parent().addClass('disabled');
        }
        else {
            this.clonedElement.parent().parent().removeClass('disabled');
        }
    }
    newValue() {
        if (!this.rating) {
            return;
        }
        const stars = this.clonedElement.parent().find('ul li');
        stars.removeClass('on');
        if (this.values) {
            const index = this.values.indexOf(this.innerValue);
            if (index !== -1) {
                for (let i = 0; i <= index; i++) {
                    $(stars.get(i)).addClass('on');
                }
            }
        }
        else {
            this.rating.val(this.innerValue);
        }
    }
    newClassValue(newClasses, oldClasses) {
        asapScheduler.schedule(() => {
            if (this.clonedElement) {
                const target = this.clonedElement.parent();
                oldClasses.forEach((cls) => {
                    target.removeClass(cls);
                });
                newClasses.forEach((cls) => {
                    target.addClass(cls);
                });
            }
        });
    }
};
__decorate([
    Input('stars'),
    __metadata("design:type", Number)
], RatingComponent.prototype, "stars", void 0);
__decorate([
    Input('round-func'),
    __metadata("design:type", String)
], RatingComponent.prototype, "roundFunc", void 0);
__decorate([
    Input('values'),
    __metadata("design:type", Array)
], RatingComponent.prototype, "values", void 0);
__decorate([
    Input('message'),
    __metadata("design:type", String)
], RatingComponent.prototype, "message", void 0);
__decorate([
    Input('star-color'),
    __metadata("design:type", String)
], RatingComponent.prototype, "starColor", void 0);
__decorate([
    Input('stared-color'),
    __metadata("design:type", String)
], RatingComponent.prototype, "staredColor", void 0);
__decorate([
    Input('static'),
    __metadata("design:type", Boolean)
], RatingComponent.prototype, "static", void 0);
__decorate([
    Input('cls-rating'),
    __metadata("design:type", String)
], RatingComponent.prototype, "clsRating", void 0);
__decorate([
    Input('cls-stars'),
    __metadata("design:type", String)
], RatingComponent.prototype, "clsStars", void 0);
__decorate([
    Input('cls-result'),
    __metadata("design:type", String)
], RatingComponent.prototype, "clsResult", void 0);
__decorate([
    ViewChild('input', { static: true }),
    __metadata("design:type", ElementRef)
], RatingComponent.prototype, "input", void 0);
RatingComponent = RatingComponent_1 = __decorate([
    Component({
        selector: 'm4-rating',
        template: "<input type=\"text\" #input\r\n\r\n\r\n       [attr.data-stars]=\"values ? values.length : stars\"\r\n       [attr.data-round-func]=\"roundFunc\"\r\n       [attr.data-values]=\"values?.join(',')\"\r\n       [attr.data-message]=\"message\"\r\n       [attr.data-star-color]=\"starColor\"\r\n       [attr.data-stared-color]=\"staredColor\"\r\n       [attr.data-static]=\"static\"\r\n\r\n       [attr.data-cls-rating]=\"clsRating\"\r\n       [attr.data-cls-stars]=\"clsStars\"\r\n       [attr.data-cls-result]=\"clsResult\">\r\n",
        encapsulation: ViewEncapsulation.None,
        providers: [DefaultValueAccessor.get(RatingComponent_1), TypeAlias.get(RatingComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [".rating.primary .stars li{color:#3969a0}.rating.primary .stars li.on{color:#0366d6}.rating.secondary .stars li{color:#767676}.rating.secondary .stars li.on{color:#607d8b}.rating.success .stars li{color:#607947}.rating.success .stars li.on{color:#60a917}.rating.alert .stars li{color:#906d6b}.rating.alert .stars li.on{color:#ce352c}.rating.warning .stars li{color:#d19c75}.rating.warning .stars li.on{color:#ff9447}.rating.yellow .stars li{color:#e0d3a3}.rating.yellow .stars li.on{color:#ffe484}.rating.info .stars li{color:#8baebf}.rating.info .stars li.on{color:#5ebdec}.rating.dark .stars li,.rating.dark .stars li.on{color:#505050}.rating.light .stars li,.rating.light .stars li.on{color:#f8f8f8}.disabled .rating .stars li.on{color:#e4e4e4!important}"]
    })
], RatingComponent);
export { RatingComponent };
