// ng-metro4/esm5/lib/helper/lists.js
import { __assign } from "tslib";
export var observerClassExceptions = ['ng-valid', 'ng-dirty', 'ng-touched', 'ng-untouched', 'ng-pristine', 'ng-invalid'];
export var iconDictionary = {
    '3d-rotation': { 'category': 'application', 'name': '3d-rotation' },
    'ac_unit': { 'category': 'application', 'name': 'ac_unit' },
    'accessible': { 'category': 'application', 'name': 'accessible' },
    'adb': { 'category': 'application', 'name': 'adb' },
    'add': { 'category': 'application', 'name': 'add' },
    'add-photo': { 'category': 'application', 'name': 'add-photo' },
    'add-shopping-cart': { 'category': 'application', 'name': 'add-shopping-cart' },
    'air': { 'category': 'application', 'name': 'air' },
    'alarm': { 'category': 'application', 'name': 'alarm' },
    'alarm-on': { 'category': 'application', 'name': 'alarm-on' },
    'amex': { 'category': 'application', 'name': 'amex' },
    'anchor': { 'category': 'application', 'name': 'anchor' },
    'apps': { 'category': 'application', 'name': 'apps' },
    'arrow-drop-down': { 'category': 'application', 'name': 'arrow-drop-down' },
    'arrow-drop-up': { 'category': 'application', 'name': 'arrow-drop-up' },
    'assignment': { 'category': 'application', 'name': 'assignment' },
    'attachment': { 'category': 'application', 'name': 'attachment' },
    'backspace': { 'category': 'application', 'name': 'backspace' },
    'backspace1': { 'category': 'application', 'name': 'backspace1' },
    'balance-scale': { 'category': 'application', 'name': 'balance-scale' },
    'barbell': { 'category': 'application', 'name': 'barbell' },
    'barcode': { 'category': 'application', 'name': 'barcode' },
    'beenhere': { 'category': 'application', 'name': 'beenhere' },
    'bell': { 'category': 'application', 'name': 'bell' },
    'bin': { 'category': 'application', 'name': 'bin' },
    'bitcoin': { 'category': 'application', 'name': 'bitcoin' },
    'blocked': { 'category': 'application', 'name': 'blocked' },
    'bold': { 'category': 'application', 'name': 'bold' },
    'book-reference': { 'category': 'application', 'name': 'book-reference' },
    'bookmark': { 'category': 'application', 'name': 'bookmark' },
    'bookmarks': { 'category': 'application', 'name': 'bookmarks' },
    'books': { 'category': 'application', 'name': 'books' },
    'brightness': { 'category': 'application', 'name': 'brightness' },
    'brightness-auto': { 'category': 'application', 'name': 'brightness-auto' },
    'broadcast': { 'category': 'application', 'name': 'broadcast' },
    'bubble': { 'category': 'application', 'name': 'bubble' },
    'bubbles': { 'category': 'application', 'name': 'bubbles' },
    'bug': { 'category': 'application', 'name': 'bug' },
    'cabinet': { 'category': 'application', 'name': 'cabinet' },
    'cake': { 'category': 'application', 'name': 'cake' },
    'calculator': { 'category': 'application', 'name': 'calculator' },
    'calculator2': { 'category': 'application', 'name': 'calculator2' },
    'calendar': { 'category': 'application', 'name': 'calendar' },
    'camera': { 'category': 'application', 'name': 'camera' },
    'camera-front': { 'category': 'application', 'name': 'camera-front' },
    'camera-rear': { 'category': 'application', 'name': 'camera-rear' },
    'cancel': { 'category': 'application', 'name': 'cancel' },
    'cart': { 'category': 'application', 'name': 'cart' },
    'cast': { 'category': 'application', 'name': 'cast' },
    'cast-connected': { 'category': 'application', 'name': 'cast-connected' },
    'cc-paypal': { 'category': 'application', 'name': 'cc-paypal' },
    'cell-no-internet': { 'category': 'application', 'name': 'cell-no-internet' },
    'cell-null': { 'category': 'application', 'name': 'cell-null' },
    'cell-off': { 'category': 'application', 'name': 'cell-off' },
    'cell-setting': { 'category': 'application', 'name': 'cell-setting' },
    'chat': { 'category': 'application', 'name': 'chat' },
    'chat-bubble': { 'category': 'application', 'name': 'chat-bubble' },
    'chat-bubble-outline': { 'category': 'application', 'name': 'chat-bubble-outline' },
    'checkmark': { 'category': 'application', 'name': 'checkmark' },
    'clipboard': { 'category': 'application', 'name': 'clipboard' },
    'cloud': { 'category': 'application', 'name': 'cloud' },
    'cloud-download': { 'category': 'application', 'name': 'cloud-download' },
    'cloud-upload': { 'category': 'application', 'name': 'cloud-upload' },
    'cny': { 'category': 'application', 'name': 'cny' },
    'cog': { 'category': 'application', 'name': 'cog' },
    'cogs': { 'category': 'application', 'name': 'cogs' },
    'coins': { 'category': 'application', 'name': 'coins' },
    'command': { 'category': 'application', 'name': 'command' },
    'comment': { 'category': 'application', 'name': 'comment' },
    'compass': { 'category': 'application', 'name': 'compass' },
    'compass2': { 'category': 'application', 'name': 'compass2' },
    'contacts-dialer': { 'category': 'application', 'name': 'contacts-dialer' },
    'contacts-mail': { 'category': 'application', 'name': 'contacts-mail' },
    'contrast': { 'category': 'application', 'name': 'contrast' },
    'copy': { 'category': 'application', 'name': 'copy' },
    'copyright': { 'category': 'application', 'name': 'copyright' },
    'create-new-folder': { 'category': 'application', 'name': 'create-new-folder' },
    'creative-commons': { 'category': 'application', 'name': 'creative-commons' },
    'credit-card': { 'category': 'application', 'name': 'credit-card' },
    'crop': { 'category': 'application', 'name': 'crop' },
    'cross': { 'category': 'application', 'name': 'cross' },
    'cut': { 'category': 'application', 'name': 'cut' },
    'dashboard': { 'category': 'application', 'name': 'dashboard' },
    'database': { 'category': 'application', 'name': 'database' },
    'developer_board': { 'category': 'application', 'name': 'developer_board' },
    'devices': { 'category': 'application', 'name': 'devices' },
    'dice': { 'category': 'application', 'name': 'dice' },
    'discout': { 'category': 'application', 'name': 'discout' },
    'discover': { 'category': 'application', 'name': 'discover' },
    'dollar': { 'category': 'application', 'name': 'dollar' },
    'dollar2': { 'category': 'application', 'name': 'dollar2' },
    'dollars': { 'category': 'application', 'name': 'dollars' },
    'done': { 'category': 'application', 'name': 'done' },
    'done_all': { 'category': 'application', 'name': 'done_all' },
    'download': { 'category': 'application', 'name': 'download' },
    'download2': { 'category': 'application', 'name': 'download2' },
    'drafts': { 'category': 'application', 'name': 'drafts' },
    'drive': { 'category': 'application', 'name': 'drive' },
    'drive2': { 'category': 'application', 'name': 'drive2' },
    'earth': { 'category': 'application', 'name': 'earth' },
    'earth2': { 'category': 'application', 'name': 'earth2' },
    'embed': { 'category': 'application', 'name': 'embed' },
    'embed2': { 'category': 'application', 'name': 'embed2' },
    'enlarge': { 'category': 'application', 'name': 'enlarge' },
    'enlarge2': { 'category': 'application', 'name': 'enlarge2' },
    'enter': { 'category': 'application', 'name': 'enter' },
    'envelop': { 'category': 'application', 'name': 'envelop' },
    'equalizer': { 'category': 'application', 'name': 'equalizer' },
    'equalizer-v': { 'category': 'application', 'name': 'equalizer-v' },
    'eur': { 'category': 'application', 'name': 'eur' },
    'ev-station': { 'category': 'application', 'name': 'ev-station' },
    'event-available': { 'category': 'application', 'name': 'event-available' },
    'event-busy': { 'category': 'application', 'name': 'event-busy' },
    'exit': { 'category': 'application', 'name': 'exit' },
    'expand-less': { 'category': 'application', 'name': 'expand-less' },
    'expand-more': { 'category': 'application', 'name': 'expand-more' },
    'eye': { 'category': 'application', 'name': 'eye' },
    'eyedropper': { 'category': 'application', 'name': 'eyedropper' },
    'favorite': { 'category': 'application', 'name': 'favorite' },
    'feed3': { 'category': 'application', 'name': 'feed3' },
    'female': { 'category': 'application', 'name': 'female' },
    'film': { 'category': 'application', 'name': 'film' },
    'filter': { 'category': 'application', 'name': 'filter' },
    'fingerprint': { 'category': 'application', 'name': 'fingerprint' },
    'fire': { 'category': 'application', 'name': 'fire' },
    'fire-extinguisher': { 'category': 'application', 'name': 'fire-extinguisher' },
    'flag': { 'category': 'application', 'name': 'flag' },
    'flash-auto': { 'category': 'application', 'name': 'flash-auto' },
    'flash-off': { 'category': 'application', 'name': 'flash-off' },
    'flash-on': { 'category': 'application', 'name': 'flash-on' },
    'floppy-disk': { 'category': 'application', 'name': 'floppy-disk' },
    'florist': { 'category': 'application', 'name': 'florist' },
    'flow-branch': { 'category': 'application', 'name': 'flow-branch' },
    'flow-cascade': { 'category': 'application', 'name': 'flow-cascade' },
    'flow-line': { 'category': 'application', 'name': 'flow-line' },
    'flow-parallel': { 'category': 'application', 'name': 'flow-parallel' },
    'flow-tree': { 'category': 'application', 'name': 'flow-tree' },
    'fonticons': { 'category': 'application', 'name': 'fonticons' },
    'gamepad': { 'category': 'application', 'name': 'gamepad' },
    'gas-station': { 'category': 'application', 'name': 'gas-station' },
    'gbp': { 'category': 'application', 'name': 'gbp' },
    'gift': { 'category': 'application', 'name': 'gift' },
    'goat': { 'category': 'application', 'name': 'goat' },
    'hammer': { 'category': 'application', 'name': 'hammer' },
    'hdr-off': { 'category': 'application', 'name': 'hdr-off' },
    'hdr-on': { 'category': 'application', 'name': 'hdr-on' },
    'headphones': { 'category': 'application', 'name': 'headphones' },
    'help': { 'category': 'application', 'name': 'help' },
    'history': { 'category': 'application', 'name': 'history' },
    'home': { 'category': 'application', 'name': 'home' },
    'hotel': { 'category': 'application', 'name': 'hotel' },
    'hour-glass': { 'category': 'application', 'name': 'hour-glass' },
    'http': { 'category': 'application', 'name': 'http' },
    'image': { 'category': 'application', 'name': 'image' },
    'images': { 'category': 'application', 'name': 'images' },
    'import-contacts': { 'category': 'application', 'name': 'import-contacts' },
    'import-export': { 'category': 'application', 'name': 'import-export' },
    'inbox': { 'category': 'application', 'name': 'inbox' },
    'indent-decrease': { 'category': 'application', 'name': 'indent-decrease' },
    'indent-increase': { 'category': 'application', 'name': 'indent-increase' },
    'info': { 'category': 'application', 'name': 'info' },
    'inr': { 'category': 'application', 'name': 'inr' },
    'insert-drive-file': { 'category': 'application', 'name': 'insert-drive-file' },
    'insert-template': { 'category': 'application', 'name': 'insert-template' },
    'install': { 'category': 'application', 'name': 'install' },
    'italic': { 'category': 'application', 'name': 'italic' },
    'justice': { 'category': 'application', 'name': 'justice' },
    'key': { 'category': 'application', 'name': 'key' },
    'keyboard': { 'category': 'application', 'name': 'keyboard' },
    'keyboard-return': { 'category': 'application', 'name': 'keyboard-return' },
    'keyboard-tab': { 'category': 'application', 'name': 'keyboard-tab' },
    'keyboard-voice': { 'category': 'application', 'name': 'keyboard-voice' },
    'krw': { 'category': 'application', 'name': 'krw' },
    'lamp': { 'category': 'application', 'name': 'lamp' },
    'language': { 'category': 'application', 'name': 'language' },
    'layers': { 'category': 'application', 'name': 'layers' },
    'layers-clear': { 'category': 'application', 'name': 'layers-clear' },
    'leanpub': { 'category': 'application', 'name': 'leanpub' },
    'library': { 'category': 'application', 'name': 'library' },
    'link': { 'category': 'application', 'name': 'link' },
    'list': { 'category': 'application', 'name': 'list' },
    'list-numbered': { 'category': 'application', 'name': 'list-numbered' },
    'list2': { 'category': 'application', 'name': 'list2' },
    'local-service': { 'category': 'application', 'name': 'local-service' },
    'location': { 'category': 'application', 'name': 'location' },
    'location-city': { 'category': 'application', 'name': 'location-city' },
    'location-off': { 'category': 'application', 'name': 'location-off' },
    'lock': { 'category': 'application', 'name': 'lock' },
    'lock-rotation': { 'category': 'application', 'name': 'lock-rotation' },
    'looks': { 'category': 'application', 'name': 'looks' },
    'ltr': { 'category': 'application', 'name': 'ltr' },
    'magic-wand': { 'category': 'application', 'name': 'magic-wand' },
    'mail': { 'category': 'application', 'name': 'mail' },
    'mail-read': { 'category': 'application', 'name': 'mail-read' },
    'male': { 'category': 'application', 'name': 'male' },
    'map': { 'category': 'application', 'name': 'map' },
    'map2': { 'category': 'application', 'name': 'map2' },
    'mars': { 'category': 'application', 'name': 'mars' },
    'mastercard': { 'category': 'application', 'name': 'mastercard' },
    'medal': { 'category': 'application', 'name': 'medal' },
    'medium': { 'category': 'application', 'name': 'medium' },
    'menu': { 'category': 'application', 'name': 'menu' },
    'meter': { 'category': 'application', 'name': 'meter' },
    'minus': { 'category': 'application', 'name': 'minus' },
    'money': { 'category': 'application', 'name': 'money' },
    'more-horiz': { 'category': 'application', 'name': 'more-horiz' },
    'more-vert': { 'category': 'application', 'name': 'more-vert' },
    'move-down': { 'category': 'application', 'name': 'move-down' },
    'move-up': { 'category': 'application', 'name': 'move-up' },
    'move_to_inbox': { 'category': 'application', 'name': 'move_to_inbox' },
    'multitrack-audio': { 'category': 'application', 'name': 'multitrack-audio' },
    'music': { 'category': 'application', 'name': 'music' },
    'navigation': { 'category': 'application', 'name': 'navigation' },
    'near-me': { 'category': 'application', 'name': 'near-me' },
    'network-cell': { 'category': 'application', 'name': 'network-cell' },
    'network-locked': { 'category': 'application', 'name': 'network-locked' },
    'network-wifi': { 'category': 'application', 'name': 'network-wifi' },
    'news': { 'category': 'application', 'name': 'news' },
    'not': { 'category': 'application', 'name': 'not' },
    'notification': { 'category': 'application', 'name': 'notification' },
    'open-book': { 'category': 'application', 'name': 'open-book' },
    'organization': { 'category': 'application', 'name': 'organization' },
    'page-break': { 'category': 'application', 'name': 'page-break' },
    'paint': { 'category': 'application', 'name': 'paint' },
    'palette': { 'category': 'application', 'name': 'palette' },
    'pan-tool': { 'category': 'application', 'name': 'pan-tool' },
    'paper-plane': { 'category': 'application', 'name': 'paper-plane' },
    'paragraph-center': { 'category': 'application', 'name': 'paragraph-center' },
    'paragraph-justify': { 'category': 'application', 'name': 'paragraph-justify' },
    'paragraph-left': { 'category': 'application', 'name': 'paragraph-left' },
    'paragraph-right': { 'category': 'application', 'name': 'paragraph-right' },
    'paste': { 'category': 'application', 'name': 'paste' },
    'paw': { 'category': 'application', 'name': 'paw' },
    'pencil': { 'category': 'application', 'name': 'pencil' },
    'phonelink-erase': { 'category': 'application', 'name': 'phonelink-erase' },
    'phonelink-lock': { 'category': 'application', 'name': 'phonelink-lock' },
    'phonelink-ring': { 'category': 'application', 'name': 'phonelink-ring' },
    'phonelink-setup': { 'category': 'application', 'name': 'phonelink-setup' },
    'photo-camera': { 'category': 'application', 'name': 'photo-camera' },
    'photo-camera-enhance': { 'category': 'application', 'name': 'photo-camera-enhance' },
    'piano': { 'category': 'application', 'name': 'piano' },
    'pilcrow': { 'category': 'application', 'name': 'pilcrow' },
    'pin': { 'category': 'application', 'name': 'pin' },
    'plus': { 'category': 'application', 'name': 'plus' },
    'power': { 'category': 'application', 'name': 'power' },
    'power-cord': { 'category': 'application', 'name': 'power-cord' },
    'present-all': { 'category': 'application', 'name': 'present-all' },
    'printer': { 'category': 'application', 'name': 'printer' },
    'profile': { 'category': 'application', 'name': 'profile' },
    'qa': { 'category': 'application', 'name': 'qa' },
    'qrcode': { 'category': 'application', 'name': 'qrcode' },
    'question': { 'category': 'application', 'name': 'question' },
    'quote': { 'category': 'application', 'name': 'quote' },
    'record_voice_over': { 'category': 'application', 'name': 'record_voice_over' },
    'redo': { 'category': 'application', 'name': 'redo' },
    'refresh': { 'category': 'application', 'name': 'refresh' },
    'registered': { 'category': 'application', 'name': 'registered' },
    'replay': { 'category': 'application', 'name': 'replay' },
    'reply': { 'category': 'application', 'name': 'reply' },
    'reply_all': { 'category': 'application', 'name': 'reply_all' },
    'restaurant': { 'category': 'application', 'name': 'restaurant' },
    'rocket': { 'category': 'application', 'name': 'rocket' },
    'room': { 'category': 'application', 'name': 'room' },
    'rotation': { 'category': 'application', 'name': 'rotation' },
    'rouble': { 'category': 'application', 'name': 'rouble' },
    'rtl': { 'category': 'application', 'name': 'rtl' },
    'school': { 'category': 'application', 'name': 'school' },
    'sd-card': { 'category': 'application', 'name': 'sd-card' },
    'search': { 'category': 'application', 'name': 'search' },
    'section': { 'category': 'application', 'name': 'section' },
    'security': { 'category': 'application', 'name': 'security' },
    'server': { 'category': 'application', 'name': 'server' },
    'settings-ethernet': { 'category': 'application', 'name': 'settings-ethernet' },
    'settings-power': { 'category': 'application', 'name': 'settings-power' },
    'settings-voice': { 'category': 'application', 'name': 'settings-voice' },
    'share': { 'category': 'application', 'name': 'share' },
    'shareable': { 'category': 'application', 'name': 'shareable' },
    'shift': { 'category': 'application', 'name': 'shift' },
    'shit': { 'category': 'application', 'name': 'shit' },
    'shop': { 'category': 'application', 'name': 'shop' },
    'shopping-basket': { 'category': 'application', 'name': 'shopping-basket' },
    'shopping-basket2': { 'category': 'application', 'name': 'shopping-basket2' },
    'shrink': { 'category': 'application', 'name': 'shrink' },
    'shrink2': { 'category': 'application', 'name': 'shrink2' },
    'sort-asc': { 'category': 'application', 'name': 'sort-asc' },
    'sort-desc': { 'category': 'application', 'name': 'sort-desc' },
    'spell-check': { 'category': 'application', 'name': 'spell-check' },
    'spoon-fork': { 'category': 'application', 'name': 'spoon-fork' },
    'squirrel': { 'category': 'application', 'name': 'squirrel' },
    'stack': { 'category': 'application', 'name': 'stack' },
    'stack2': { 'category': 'application', 'name': 'stack2' },
    'stack3': { 'category': 'application', 'name': 'stack3' },
    'star-empty': { 'category': 'application', 'name': 'star-empty' },
    'star-full': { 'category': 'application', 'name': 'star-full' },
    'star-half': { 'category': 'application', 'name': 'star-half' },
    'steps': { 'category': 'application', 'name': 'steps' },
    'strikethrough': { 'category': 'application', 'name': 'strikethrough' },
    'subscript': { 'category': 'application', 'name': 'subscript' },
    'suitcase': { 'category': 'application', 'name': 'suitcase' },
    'superscript': { 'category': 'application', 'name': 'superscript' },
    'switch': { 'category': 'application', 'name': 'switch' },
    'sync-disabled': { 'category': 'application', 'name': 'sync-disabled' },
    'sync-problem': { 'category': 'application', 'name': 'sync-problem' },
    'tab': { 'category': 'application', 'name': 'tab' },
    'table': { 'category': 'application', 'name': 'table' },
    'tag': { 'category': 'application', 'name': 'tag' },
    'tags': { 'category': 'application', 'name': 'tags' },
    'target': { 'category': 'application', 'name': 'target' },
    'thumbs-down': { 'category': 'application', 'name': 'thumbs-down' },
    'thumbs-up': { 'category': 'application', 'name': 'thumbs-up' },
    'tools': { 'category': 'application', 'name': 'tools' },
    'trademark': { 'category': 'application', 'name': 'trademark' },
    'traff': { 'category': 'application', 'name': 'traff' },
    'traffic-cone': { 'category': 'application', 'name': 'traffic-cone' },
    'tree': { 'category': 'application', 'name': 'tree' },
    'trophy': { 'category': 'application', 'name': 'trophy' },
    'try': { 'category': 'application', 'name': 'try' },
    'underline': { 'category': 'application', 'name': 'underline' },
    'undo': { 'category': 'application', 'name': 'undo' },
    'unfold-less': { 'category': 'application', 'name': 'unfold-less' },
    'unfold-more': { 'category': 'application', 'name': 'unfold-more' },
    'uninstall': { 'category': 'application', 'name': 'uninstall' },
    'unlink': { 'category': 'application', 'name': 'unlink' },
    'unlock': { 'category': 'application', 'name': 'unlock' },
    'upload': { 'category': 'application', 'name': 'upload' },
    'upload2': { 'category': 'application', 'name': 'upload2' },
    'usb': { 'category': 'application', 'name': 'usb' },
    'user': { 'category': 'application', 'name': 'user' },
    'user-check': { 'category': 'application', 'name': 'user-check' },
    'user-minus': { 'category': 'application', 'name': 'user-minus' },
    'user-plus': { 'category': 'application', 'name': 'user-plus' },
    'user-secret': { 'category': 'application', 'name': 'user-secret' },
    'users': { 'category': 'application', 'name': 'users' },
    'venus': { 'category': 'application', 'name': 'venus' },
    'verified': { 'category': 'application', 'name': 'verified' },
    'versions': { 'category': 'application', 'name': 'versions' },
    'vertical-align-bottom': { 'category': 'application', 'name': 'vertical-align-bottom' },
    'vertical-align-center': { 'category': 'application', 'name': 'vertical-align-center' },
    'vertical-align-top': { 'category': 'application', 'name': 'vertical-align-top' },
    'video-camera': { 'category': 'application', 'name': 'video-camera' },
    'visa': { 'category': 'application', 'name': 'visa' },
    'vpn-lock': { 'category': 'application', 'name': 'vpn-lock' },
    'vpn-publ': { 'category': 'application', 'name': 'vpn-publ' },
    'warning': { 'category': 'application', 'name': 'warning' },
    'watch': { 'category': 'application', 'name': 'watch' },
    'water': { 'category': 'application', 'name': 'water' },
    'widgets': { 'category': 'application', 'name': 'widgets' },
    'wifi-lock': { 'category': 'application', 'name': 'wifi-lock' },
    'wifi-off': { 'category': 'application', 'name': 'wifi-off' },
    'wrench': { 'category': 'application', 'name': 'wrench' },
    'zoom-in': { 'category': 'application', 'name': 'zoom-in' },
    'zoom-out': { 'category': 'application', 'name': 'zoom-out' },
    'arrow-down': { 'category': 'arrows', 'name': 'arrow-down' },
    'arrow-down-left': { 'category': 'arrows', 'name': 'arrow-down-left' },
    'arrow-down-right': { 'category': 'arrows', 'name': 'arrow-down-right' },
    'arrow-left': { 'category': 'arrows', 'name': 'arrow-left' },
    'arrow-right': { 'category': 'arrows', 'name': 'arrow-right' },
    'arrow-up': { 'category': 'arrows', 'name': 'arrow-up' },
    'arrow-up-left': { 'category': 'arrows', 'name': 'arrow-up-left' },
    'arrow-up-right': { 'category': 'arrows', 'name': 'arrow-up-right' },
    'chevron-left': { 'category': 'arrows', 'name': 'chevron-left' },
    'chevron-right': { 'category': 'arrows', 'name': 'chevron-right' },
    'chevron-thin-down': { 'category': 'arrows', 'name': 'chevron-thin-down' },
    'chevron-thin-left': { 'category': 'arrows', 'name': 'chevron-thin-left' },
    'chevron-thin-right': { 'category': 'arrows', 'name': 'chevron-thin-right' },
    'chevron-thin-up': { 'category': 'arrows', 'name': 'chevron-thin-up' },
    'amazon': { 'category': 'brand', 'name': 'amazon' },
    'android': { 'category': 'brand', 'name': 'android' },
    'apple': { 'category': 'brand', 'name': 'apple' },
    'blogger': { 'category': 'brand', 'name': 'blogger' },
    'chrome': { 'category': 'brand', 'name': 'chrome' },
    'codepen': { 'category': 'brand', 'name': 'codepen' },
    'creative-cloud': { 'category': 'brand', 'name': 'creative-cloud' },
    'css3': { 'category': 'brand', 'name': 'css3' },
    'delicious': { 'category': 'brand', 'name': 'delicious' },
    'deviantart': { 'category': 'brand', 'name': 'deviantart' },
    'digg': { 'category': 'brand', 'name': 'digg' },
    'dribbble': { 'category': 'brand', 'name': 'dribbble' },
    'dropbox': { 'category': 'brand', 'name': 'dropbox' },
    'evernote': { 'category': 'brand', 'name': 'evernote' },
    'facebook': { 'category': 'brand', 'name': 'facebook' },
    'facebook2': { 'category': 'brand', 'name': 'facebook2' },
    'finder': { 'category': 'brand', 'name': 'finder' },
    'firefox': { 'category': 'brand', 'name': 'firefox' },
    'flattr': { 'category': 'brand', 'name': 'flattr' },
    'foursquare': { 'category': 'brand', 'name': 'foursquare' },
    'git': { 'category': 'brand', 'name': 'git' },
    'github': { 'category': 'brand', 'name': 'github' },
    'gitlab': { 'category': 'brand', 'name': 'gitlab' },
    'google': { 'category': 'brand', 'name': 'google' },
    'google-plus': { 'category': 'brand', 'name': 'google-plus' },
    'google-plus2': { 'category': 'brand', 'name': 'google-plus2' },
    'google-wallet': { 'category': 'brand', 'name': 'google-wallet' },
    'grav': { 'category': 'brand', 'name': 'grav' },
    'html5': { 'category': 'brand', 'name': 'html5' },
    'IcoMoon': { 'category': 'brand', 'name': 'IcoMoon' },
    'ie': { 'category': 'brand', 'name': 'ie' },
    'imdb': { 'category': 'brand', 'name': 'imdb' },
    'instagram': { 'category': 'brand', 'name': 'instagram' },
    'joomla': { 'category': 'brand', 'name': 'joomla' },
    'jsfiddle': { 'category': 'brand', 'name': 'jsfiddle' },
    'lastfm': { 'category': 'brand', 'name': 'lastfm' },
    'linkedin': { 'category': 'brand', 'name': 'linkedin' },
    'linux': { 'category': 'brand', 'name': 'linux' },
    'meetup': { 'category': 'brand', 'name': 'meetup' },
    'odnoklassniki': { 'category': 'brand', 'name': 'odnoklassniki' },
    'onedrive': { 'category': 'brand', 'name': 'onedrive' },
    'opencart': { 'category': 'brand', 'name': 'opencart' },
    'openid': { 'category': 'brand', 'name': 'openid' },
    'opera': { 'category': 'brand', 'name': 'opera' },
    'paypal': { 'category': 'brand', 'name': 'paypal' },
    'picassa': { 'category': 'brand', 'name': 'picassa' },
    'pinterest': { 'category': 'brand', 'name': 'pinterest' },
    'reddit': { 'category': 'brand', 'name': 'reddit' },
    'safari': { 'category': 'brand', 'name': 'safari' },
    'skype': { 'category': 'brand', 'name': 'skype' },
    'soundcloud': { 'category': 'brand', 'name': 'soundcloud' },
    'stackoverflow': { 'category': 'brand', 'name': 'stackoverflow' },
    'steam': { 'category': 'brand', 'name': 'steam' },
    'steam2': { 'category': 'brand', 'name': 'steam2' },
    'stumbleupon': { 'category': 'brand', 'name': 'stumbleupon' },
    'swarm': { 'category': 'brand', 'name': 'swarm' },
    'themeisle': { 'category': 'brand', 'name': 'themeisle' },
    'tux': { 'category': 'brand', 'name': 'tux' },
    'twitch': { 'category': 'brand', 'name': 'twitch' },
    'twitter': { 'category': 'brand', 'name': 'twitter' },
    'vimeo': { 'category': 'brand', 'name': 'vimeo' },
    'vk': { 'category': 'brand', 'name': 'vk' },
    'wikipedia': { 'category': 'brand', 'name': 'wikipedia' },
    'windows': { 'category': 'brand', 'name': 'windows' },
    'wordpress': { 'category': 'brand', 'name': 'wordpress' },
    'yelp': { 'category': 'brand', 'name': 'yelp' },
    'youtube': { 'category': 'brand', 'name': 'youtube' },
    'youtube-play': { 'category': 'brand', 'name': 'youtube-play' },
    'youtube2': { 'category': 'brand', 'name': 'youtube2' },
    'chart-bars': { 'category': 'chart', 'name': 'chart-bars' },
    'chart-bars2': { 'category': 'chart', 'name': 'chart-bars2' },
    'chart-dots': { 'category': 'chart', 'name': 'chart-dots' },
    'chart-line': { 'category': 'chart', 'name': 'chart-line' },
    'chart-pie': { 'category': 'chart', 'name': 'chart-pie' },
    'display': { 'category': 'device', 'name': 'display' },
    'keyboard-hide': { 'category': 'device', 'name': 'keyboard-hide' },
    'laptop': { 'category': 'device', 'name': 'laptop' },
    'mobile': { 'category': 'device', 'name': 'mobile' },
    'tablet': { 'category': 'device', 'name': 'tablet' },
    'tablet-landscape': { 'category': 'device', 'name': 'tablet-landscape' },
    'description': { 'category': 'file', 'name': 'description' },
    'file-archive': { 'category': 'file', 'name': 'file-archive' },
    'file-audio': { 'category': 'file', 'name': 'file-audio' },
    'file-binary': { 'category': 'file', 'name': 'file-binary' },
    'file-code': { 'category': 'file', 'name': 'file-code' },
    'file-download': { 'category': 'file', 'name': 'file-download' },
    'file-empty': { 'category': 'file', 'name': 'file-empty' },
    'file-excel': { 'category': 'file', 'name': 'file-excel' },
    'file-image': { 'category': 'file', 'name': 'file-image' },
    'file-movie': { 'category': 'file', 'name': 'file-movie' },
    'file-music': { 'category': 'file', 'name': 'file-music' },
    'file-openoffice': { 'category': 'file', 'name': 'file-openoffice' },
    'file-pdf': { 'category': 'file', 'name': 'file-pdf' },
    'file-picture': { 'category': 'file', 'name': 'file-picture' },
    'file-play': { 'category': 'file', 'name': 'file-play' },
    'file-powerpoint': { 'category': 'file', 'name': 'file-powerpoint' },
    'file-text': { 'category': 'file', 'name': 'file-text' },
    'file-upload': { 'category': 'file', 'name': 'file-upload' },
    'file-video': { 'category': 'file', 'name': 'file-video' },
    'file-word': { 'category': 'file', 'name': 'file-word' },
    'file-zip': { 'category': 'file', 'name': 'file-zip' },
    'files-empty': { 'category': 'file', 'name': 'files-empty' },
    'folder': { 'category': 'file', 'name': 'folder' },
    'folder-download': { 'category': 'file', 'name': 'folder-download' },
    'folder-minus': { 'category': 'file', 'name': 'folder-minus' },
    'folder-open': { 'category': 'file', 'name': 'folder-open' },
    'folder-open2': { 'category': 'file', 'name': 'folder-open2' },
    'folder-plus': { 'category': 'file', 'name': 'folder-plus' },
    'folder-shared': { 'category': 'file', 'name': 'folder-shared' },
    'folder-special': { 'category': 'file', 'name': 'folder-special' },
    'folder-special2': { 'category': 'file', 'name': 'folder-special2' },
    'folder-upload': { 'category': 'file', 'name': 'folder-upload' },
    'folder2': { 'category': 'file', 'name': 'folder2' },
    'libreoffice': { 'category': 'file', 'name': 'libreoffice' },
    'note': { 'category': 'file', 'name': 'note' },
    'note-add': { 'category': 'file', 'name': 'note-add' },
    'ambulance': { 'category': 'medic', 'name': 'ambulance' },
    'heart': { 'category': 'medic', 'name': 'heart' },
    'heart-broken': { 'category': 'medic', 'name': 'heart-broken' },
    'heartbeat': { 'category': 'medic', 'name': 'heartbeat' },
    'injection': { 'category': 'medic', 'name': 'injection' },
    'lab': { 'category': 'medic', 'name': 'lab' },
    'medkit': { 'category': 'medic', 'name': 'medkit' },
    'microscope': { 'category': 'medic', 'name': 'microscope' },
    'stethoscope': { 'category': 'medic', 'name': 'stethoscope' },
    'thermometer2': { 'category': 'medic', 'name': 'thermometer2' },
    'user-md': { 'category': 'medic', 'name': 'user-md' },
    'battery-charge': { 'category': 'mobile', 'name': 'battery-charge' },
    'battery-empty': { 'category': 'mobile', 'name': 'battery-empty' },
    'battery-full': { 'category': 'mobile', 'name': 'battery-full' },
    'battery-one': { 'category': 'mobile', 'name': 'battery-one' },
    'battery-two': { 'category': 'mobile', 'name': 'battery-two' },
    'bluetooth': { 'category': 'mobile', 'name': 'bluetooth' },
    'bt-audio': { 'category': 'mobile', 'name': 'bt-audio' },
    'bt-connected': { 'category': 'mobile', 'name': 'bt-connected' },
    'bt-disabled': { 'category': 'mobile', 'name': 'bt-disabled' },
    'bt-searching': { 'category': 'mobile', 'name': 'bt-searching' },
    'bt-settings': { 'category': 'mobile', 'name': 'bt-settings' },
    'cell-on': { 'category': 'mobile', 'name': 'cell-on' },
    'dialer-sip': { 'category': 'mobile', 'name': 'dialer-sip' },
    'dialpad': { 'category': 'mobile', 'name': 'dialpad' },
    'feed': { 'category': 'mobile', 'name': 'feed' },
    'mic': { 'category': 'mobile', 'name': 'mic' },
    'my-location': { 'category': 'mobile', 'name': 'my-location' },
    'perm-phone-msg': { 'category': 'mobile', 'name': 'perm-phone-msg' },
    'phone': { 'category': 'mobile', 'name': 'phone' },
    'phone-bt': { 'category': 'mobile', 'name': 'phone-bt' },
    'phone-forwarded': { 'category': 'mobile', 'name': 'phone-forwarded' },
    'phone-in-talk': { 'category': 'mobile', 'name': 'phone-in-talk' },
    'phone-locked': { 'category': 'mobile', 'name': 'phone-locked' },
    'phone-missed': { 'category': 'mobile', 'name': 'phone-missed' },
    'phone-paused': { 'category': 'mobile', 'name': 'phone-paused' },
    'phonelink': { 'category': 'mobile', 'name': 'phonelink' },
    'phonelink-off': { 'category': 'mobile', 'name': 'phonelink-off' },
    'ring-volume': { 'category': 'mobile', 'name': 'ring-volume' },
    'settings-phone': { 'category': 'mobile', 'name': 'settings-phone' },
    'voicemail': { 'category': 'mobile', 'name': 'voicemail' },
    'wifi-connect': { 'category': 'mobile', 'name': 'wifi-connect' },
    'wifi-full': { 'category': 'mobile', 'name': 'wifi-full' },
    'wifi-low': { 'category': 'mobile', 'name': 'wifi-low' },
    'wifi-mid': { 'category': 'mobile', 'name': 'wifi-mid' },
    'backward': { 'category': 'player', 'name': 'backward' },
    'eject': { 'category': 'player', 'name': 'eject' },
    'equalizer2': { 'category': 'player', 'name': 'equalizer2' },
    'featured-play-list': { 'category': 'player', 'name': 'featured-play-list' },
    'featured-video': { 'category': 'player', 'name': 'featured-video' },
    'first': { 'category': 'player', 'name': 'first' },
    'forward': { 'category': 'player', 'name': 'forward' },
    'infinite': { 'category': 'player', 'name': 'infinite' },
    'last': { 'category': 'player', 'name': 'last' },
    'loop': { 'category': 'player', 'name': 'loop' },
    'loop2': { 'category': 'player', 'name': 'loop2' },
    'next': { 'category': 'player', 'name': 'next' },
    'pause': { 'category': 'player', 'name': 'pause' },
    'play': { 'category': 'player', 'name': 'play' },
    'previous': { 'category': 'player', 'name': 'previous' },
    'shuffle': { 'category': 'player', 'name': 'shuffle' },
    'stop': { 'category': 'player', 'name': 'stop' },
    'volume-high': { 'category': 'player', 'name': 'volume-high' },
    'volume-low': { 'category': 'player', 'name': 'volume-low' },
    'volume-medium': { 'category': 'player', 'name': 'volume-medium' },
    'volume-minus': { 'category': 'player', 'name': 'volume-minus' },
    'volume-mute': { 'category': 'player', 'name': 'volume-mute' },
    'volume-mute2': { 'category': 'player', 'name': 'volume-mute2' },
    'volume-plus': { 'category': 'player', 'name': 'volume-plus' },
    'spinner': { 'category': 'spinner', 'name': 'spinner' },
    'spinner2': { 'category': 'spinner', 'name': 'spinner2' },
    'spinner3': { 'category': 'spinner', 'name': 'spinner3' },
    'spinner4': { 'category': 'spinner', 'name': 'spinner4' },
    'spinner5': { 'category': 'spinner', 'name': 'spinner5' },
    'airplane': { 'category': 'vehicle', 'name': 'airplane' },
    'automobile': { 'category': 'vehicle', 'name': 'automobile' },
    'bicycle': { 'category': 'vehicle', 'name': 'bicycle' },
    'bus': { 'category': 'vehicle', 'name': 'bus' },
    'cab': { 'category': 'vehicle', 'name': 'cab' },
    'directions-bike': { 'category': 'vehicle', 'name': 'directions-bike' },
    'drive-eta': { 'category': 'vehicle', 'name': 'drive-eta' },
    'local-airport': { 'category': 'vehicle', 'name': 'local-airport' },
    'motorcycle': { 'category': 'vehicle', 'name': 'motorcycle' },
    'satellite': { 'category': 'vehicle', 'name': 'satellite' },
    'ship': { 'category': 'vehicle', 'name': 'ship' },
    'space-shuttle': { 'category': 'vehicle', 'name': 'space-shuttle' },
    'subway': { 'category': 'vehicle', 'name': 'subway' },
    'train': { 'category': 'vehicle', 'name': 'train' },
    'truck': { 'category': 'vehicle', 'name': 'truck' },
    'celsius': { 'category': 'weather', 'name': 'celsius' },
    'cloud2': { 'category': 'weather', 'name': 'cloud2' },
    'cloud3': { 'category': 'weather', 'name': 'cloud3' },
    'cloud4': { 'category': 'weather', 'name': 'cloud4' },
    'cloud5': { 'category': 'weather', 'name': 'cloud5' },
    'cloud6': { 'category': 'weather', 'name': 'cloud6' },
    'cloudy': { 'category': 'weather', 'name': 'cloudy' },
    'cloudy2': { 'category': 'weather', 'name': 'cloudy2' },
    'cloudy3': { 'category': 'weather', 'name': 'cloudy3' },
    'cloudy4': { 'category': 'weather', 'name': 'cloudy4' },
    'fahrenheit': { 'category': 'weather', 'name': 'fahrenheit' },
    'lightning': { 'category': 'weather', 'name': 'lightning' },
    'lightning2': { 'category': 'weather', 'name': 'lightning2' },
    'lightning3': { 'category': 'weather', 'name': 'lightning3' },
    'lightning4': { 'category': 'weather', 'name': 'lightning4' },
    'lightning5': { 'category': 'weather', 'name': 'lightning5' },
    'lines': { 'category': 'weather', 'name': 'lines' },
    'moon': { 'category': 'weather', 'name': 'moon' },
    'moon2': { 'category': 'weather', 'name': 'moon2' },
    'none': { 'category': 'weather', 'name': 'none' },
    'rainy': { 'category': 'weather', 'name': 'rainy' },
    'rainy2': { 'category': 'weather', 'name': 'rainy2' },
    'rainy3': { 'category': 'weather', 'name': 'rainy3' },
    'rainy4': { 'category': 'weather', 'name': 'rainy4' },
    'snowflake': { 'category': 'weather', 'name': 'snowflake' },
    'snowy': { 'category': 'weather', 'name': 'snowy' },
    'snowy2': { 'category': 'weather', 'name': 'snowy2' },
    'snowy3': { 'category': 'weather', 'name': 'snowy3' },
    'snowy4': { 'category': 'weather', 'name': 'snowy4' },
    'snowy5': { 'category': 'weather', 'name': 'snowy5' },
    'sun': { 'category': 'weather', 'name': 'sun' },
    'sun3': { 'category': 'weather', 'name': 'sun3' },
    'sun4': { 'category': 'weather', 'name': 'sun4' },
    'sunrise': { 'category': 'weather', 'name': 'sunrise' },
    'thermometer': { 'category': 'weather', 'name': 'thermometer' },
    'weather': { 'category': 'weather', 'name': 'weather' },
    'weather2': { 'category': 'weather', 'name': 'weather2' },
    'weather3': { 'category': 'weather', 'name': 'weather3' },
    'weather4': { 'category': 'weather', 'name': 'weather4' },
    'weather5': { 'category': 'weather', 'name': 'weather5' },
    'wind': { 'category': 'weather', 'name': 'wind' },
    'windy': { 'category': 'weather', 'name': 'windy' },
    'windy2': { 'category': 'weather', 'name': 'windy2' },
    'windy3': { 'category': 'weather', 'name': 'windy3' },
    'windy4': { 'category': 'weather', 'name': 'windy4' },
    'windy5': { 'category': 'weather', 'name': 'windy5' }
};
export var iconCategoryDictionary = {
    'application': 'application',
    'arrows': 'arrows',
    'brand': 'brand',
    'chart': 'chart',
    'device': 'device',
    'file': 'file',
    'medic': 'medic',
    'mobile': 'mobile',
    'player': 'player',
    'spinner': 'spinner',
    'vehicle': 'vehicle',
    'weather': 'weather'
};
export var colorDictionary = {
    'lightLime': 'lightLime',
    'lime': 'lime',
    'darkLime': 'darkLime',
    'lightGreen': 'lightGreen',
    'green': 'green',
    'darkGreen': 'darkGreen',
    'lightEmerald': 'lightEmerald',
    'emerald': 'emerald',
    'darkEmerald': 'darkEmerald',
    'lightBlue': 'lightBlue',
    'blue': 'blue',
    'darkBlue': 'darkBlue',
    'lightTeal': 'lightTeal',
    'teal': 'teal',
    'darkTeal': 'darkTeal',
    'lightCyan': 'lightCyan',
    'cyan': 'cyan',
    'darkCyan': 'darkCyan',
    'lightCobalt': 'lightCobalt',
    'cobalt': 'cobalt',
    'darkCobalt': 'darkCobalt',
    'lightIndigo': 'lightIndigo',
    'indigo': 'indigo',
    'darkIndigo': 'darkIndigo',
    'lightViolet': 'lightViolet',
    'violet': 'violet',
    'darkViolet': 'darkViolet',
    'lightPink': 'lightPink',
    'pink': 'pink',
    'darkPink': 'darkPink',
    'lightMagenta': 'lightMagenta',
    'magenta': 'magenta',
    'darkMagenta': 'darkMagenta',
    'lightCrimson': 'lightCrimson',
    'crimson': 'crimson',
    'darkCrimson': 'darkCrimson',
    'lightRed': 'lightRed',
    'red': 'red',
    'darkRed': 'darkRed',
    'lightOrange': 'lightOrange',
    'orange': 'orange',
    'darkOrange': 'darkOrange',
    'lightAmber': 'lightAmber',
    'amber': 'amber',
    'darkAmber': 'darkAmber',
    'lightYellow': 'lightYellow',
    'yellow': 'yellow',
    'darkYellow': 'darkYellow',
    'lightBrown': 'lightBrown',
    'brown': 'brown',
    'darkBrown': 'darkBrown',
    'lightOlive': 'lightOlive',
    'olive': 'olive',
    'darkOlive': 'darkOlive',
    'lightSteel': 'lightSteel',
    'steel': 'steel',
    'darkSteel': 'darkSteel',
    'lightMauve': 'lightMauve',
    'mauve': 'mauve',
    'darkMauve': 'darkMauve',
    'lightTaupe': 'lightTaupe',
    'taupe': 'taupe',
    'darkTaupe': 'darkTaupe',
    'lightGray': 'lightGray',
    'gray': 'gray',
    'darkGray': 'darkGray',
    'lightGrayBlue': 'lightGrayBlue',
    'grayBlue': 'grayBlue',
    'darkGrayBlue': 'darkGrayBlue',
    'black': 'black',
    'white': 'white',
    'dark': 'dark',
    'light': 'light',
    'grayWhite': 'grayWhite',
    'grayMouse': 'grayMouse'
};
export var animationDictionary = {
    'spin': 'spin',
    'pulse': 'pulse',
    'spanner': 'spanner',
    'ring': 'ring',
    'vertical': 'vertical',
    'horizontal': 'horizontal',
    'flash': 'flash',
    'bounce': 'bounce',
    'float': 'float',
    'heartbeat': 'heartbeat',
    'shake': 'shake',
    'shuttle': 'shuttle',
    'pass': 'pass',
    'ripple': 'ripple',
};
export var accentDictionary = {
    '': '',
    'primary': 'primary',
    'secondary': 'secondary',
    'success': 'success',
    'alert': 'alert',
    'warning': 'warning',
    'yellow': 'yellow',
    'info': 'info',
    'dark': 'dark',
    'light': 'light',
    'link': 'link'
};
export var inputDictionary = {
    'text': 'text',
    'number': 'number',
    'email': 'email',
    'password': 'password',
    'datetime-local': 'datetime-local',
    'color': 'color',
    'date': 'date',
    'month': 'month',
    'time': 'time',
    'week': 'week'
};
export var calendarButtonDictionary = {
    'cancel': 'cancel',
    'today': 'today',
    'clear': 'clear',
    'done': 'done',
};
export var widePointDictionary = {
    'fs': 'fs',
    'sm': 'sm',
    'md': 'md',
    'lg': 'lg',
    'xl': 'xl',
    'xxl': 'xxl'
};
export var positionHorizontalDictionary = {
    'left': 'left',
    'right': 'right'
};
export var spinnerButtonPositionDictionary = __assign(__assign({}, positionHorizontalDictionary), { 'default': 'default' });
export var positionVerticalDictionary = {
    'top': 'top',
    'bottom': 'bottom'
};
export var positionBaseDictionary = __assign(__assign({}, positionHorizontalDictionary), positionVerticalDictionary);
export var positionDictionary = __assign(__assign({}, positionBaseDictionary), { 'top-left': 'top-left', 'top-right': 'top-right', 'bottom-right': 'bottom-right', 'bottom-left': 'bottom-left' });
export var sizeDictionary = {
    '': '',
    'mini': 'mini',
    'small': 'small',
    'large': 'large'
};
export var gravatarDictionary = {
    'mm': 'mm',
    'identicon': 'identicon',
    'monsterid': 'monsterid',
    'wavatar': 'wavatar',
    'retro': 'retro',
    'robohash': 'robohash',
    'blank': 'blank'
};
export var activityDictionary = {
    'ring': 'ring',
    'metro': 'metro',
    'square': 'square',
    'cycle': 'cycle',
    'simple': 'simple'
};
export var activityStyleDictionary = {
    '': '',
    'dark': '',
    'color': ''
};
export var buttonShapeDictionary = {
    '': '',
    'square': 'square',
    'cycle': 'cycle'
};
export var buttonSpecialDictionary = {
    '': '',
    'command': 'command',
    'image': 'image',
    'shortcut': 'shortcut',
    'ribbon': 'ribbon',
    'multi-action': 'multi-action',
    'action': 'action',
};
export var popoverTriggerDictionary = {
    'click': 'click',
    'focus': 'focus',
    'hover': 'hover'
};
export var progressTypeDictionary = {
    '': '',
    'buffer': 'buffer',
    'load': 'load',
    'line': 'line'
};
export var roundTypeDictionary = {
    'round': 'round',
    'ceil': 'ceil',
    'floor': 'floor'
};
export var thinDictionary = {
    '': '',
    'thin': 'thin',
    'ultra-thin': 'ultra-thin'
};
export var easingDictionary = {
    'swing': 'swing',
    'easeInQuad': 'easeInQuad',
    'easeOutQuad': 'easeOutQuad',
    'easeInOutQuad': 'easeInOutQuad',
    'easeInCubic': 'easeInCubic',
    'easeOutCubic': 'easeOutCubic',
    'easeInOutCubic': 'easeInOutCubic',
    'easeInQuart': 'easeInQuart',
    'easeOutQuart': 'easeOutQuart',
    'easeInOutQuart': 'easeInOutQuart',
    'easeInQuint': 'easeInQuint',
    'easeOutQuint': 'easeOutQuint',
    'easeInOutQuint': 'easeInOutQuint',
    'easeInSine': 'easeInSine',
    'easeOutSine': 'easeOutSine',
    'easeInOutSine': 'easeInOutSine',
    'easeInExpo': 'easeInExpo',
    'easeOutExpo': 'easeOutExpo',
    'easeInOutExpo': 'easeInOutExpo',
    'easeInCirc': 'easeInCirc',
    'easeOutCirc': 'easeOutCirc',
    'easeInOutCirc': 'easeInOutCirc',
    'easeInElastic': 'easeInElastic',
    'easeOutElastic': 'easeOutElastic',
    'easeInOutElastic': 'easeInOutElastic',
    'easeInBack': 'easeInBack',
    'easeOutBack': 'easeOutBack',
    'easeInOutBack': 'easeInOutBack',
    'easeInBounce': 'easeInBounce',
    'easeOutBounce': 'easeOutBounce',
    'easeInOutBounce': 'easeInOutBounce'
};
export var fileReadModeDictionary = {
    '': '',
    'text': 'text',
    'arrayBuffer': 'arrayBuffer',
    'binaryString': 'binaryString',
    'dataUrl': 'dataUrl'
};
var Lists = /** @class */ (function () {
    function Lists() {
    }
    Lists.icons = function () {
        return Object.values(iconDictionary);
    };
    Lists.iconCategories = function () {
        return Object.values(iconCategoryDictionary);
    };
    Lists.colors = function () {
        return Object.values(colorDictionary);
    };
    Lists.animations = function () {
        return Object.values(animationDictionary);
    };
    Lists.accents = function () {
        return Object.values(accentDictionary);
    };
    Lists.widePoints = function () {
        return Object.values(widePointDictionary);
    };
    Lists.positions = function (all) {
        if (all === void 0) { all = true; }
        return Object.values(all ? positionDictionary : positionBaseDictionary);
    };
    Lists.sizes = function () {
        return Object.values(sizeDictionary);
    };
    Lists.gravatars = function () {
        return Object.values(gravatarDictionary);
    };
    Lists.activities = function () {
        return Object.values(activityDictionary);
    };
    Lists.activityStyles = function () {
        return Object.values(activityStyleDictionary);
    };
    Lists.buttonShapes = function () {
        return Object.values(buttonShapeDictionary);
    };
    Lists.buttonSpecials = function () {
        return Object.values(buttonSpecialDictionary);
    };
    Lists.popoverTriggers = function () {
        return Object.values(popoverTriggerDictionary);
    };
    Lists.progressTypes = function () {
        return Object.values(progressTypeDictionary);
    };
    Lists.roundTypes = function () {
        return Object.values(roundTypeDictionary);
    };
    Lists.thins = function () {
        return Object.values(thinDictionary);
    };
    Lists.easings = function () {
        return Object.values(easingDictionary);
    };
    return Lists;
}());
export { Lists };
