// ng-metro4/esm5/lib/form/rating/rating.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
var RatingComponent = /** @class */ (function (_super) {
    __extends(RatingComponent, _super);
    function RatingComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RatingComponent_1 = RatingComponent;
    RatingComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.rating = _this.clonedElement.rating().data('rating');
            _this.clonedElement.parent().find('ul li').one('click', function () {
                _this.touchCallback();
            });
            _this.clonedElement.on('change', function () {
                var newValue = _this.clonedElement.val();
                var valueParsed = +newValue;
                if (!Number.isNaN(valueParsed)) {
                    _this.changeValue(valueParsed);
                }
                else {
                    _this.changeValue(newValue);
                }
            });
            complete();
        });
    };
    RatingComponent.prototype.disable = function (disabled) {
        if (disabled) {
            this.clonedElement.parent().parent().addClass('disabled');
        }
        else {
            this.clonedElement.parent().parent().removeClass('disabled');
        }
    };
    RatingComponent.prototype.newValue = function () {
        if (!this.rating) {
            return;
        }
        var stars = this.clonedElement.parent().find('ul li');
        stars.removeClass('on');
        if (this.values) {
            var index = this.values.indexOf(this.innerValue);
            if (index !== -1) {
                for (var i = 0; i <= index; i++) {
                    $(stars.get(i)).addClass('on');
                }
            }
        }
        else {
            this.rating.val(this.innerValue);
        }
    };
    RatingComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        var _this = this;
        asapScheduler.schedule(function () {
            if (_this.clonedElement) {
                var target_1 = _this.clonedElement.parent();
                oldClasses.forEach(function (cls) {
                    target_1.removeClass(cls);
                });
                newClasses.forEach(function (cls) {
                    target_1.addClass(cls);
                });
            }
        });
    };
    var RatingComponent_1;
    __decorate([
        Input('stars'),
        __metadata("design:type", Number)
    ], RatingComponent.prototype, "stars", void 0);
    __decorate([
        Input('round-func'),
        __metadata("design:type", String)
    ], RatingComponent.prototype, "roundFunc", void 0);
    __decorate([
        Input('values'),
        __metadata("design:type", Array)
    ], RatingComponent.prototype, "values", void 0);
    __decorate([
        Input('message'),
        __metadata("design:type", String)
    ], RatingComponent.prototype, "message", void 0);
    __decorate([
        Input('star-color'),
        __metadata("design:type", String)
    ], RatingComponent.prototype, "starColor", void 0);
    __decorate([
        Input('stared-color'),
        __metadata("design:type", String)
    ], RatingComponent.prototype, "staredColor", void 0);
    __decorate([
        Input('static'),
        __metadata("design:type", Boolean)
    ], RatingComponent.prototype, "static", void 0);
    __decorate([
        Input('cls-rating'),
        __metadata("design:type", String)
    ], RatingComponent.prototype, "clsRating", void 0);
    __decorate([
        Input('cls-stars'),
        __metadata("design:type", String)
    ], RatingComponent.prototype, "clsStars", void 0);
    __decorate([
        Input('cls-result'),
        __metadata("design:type", String)
    ], RatingComponent.prototype, "clsResult", void 0);
    __decorate([
        ViewChild('input', { static: true }),
        __metadata("design:type", ElementRef)
    ], RatingComponent.prototype, "input", void 0);
    RatingComponent = RatingComponent_1 = __decorate([
        Component({
            selector: 'm4-rating',
            template: "<input type=\"text\" #input\r\n\r\n\r\n       [attr.data-stars]=\"values ? values.length : stars\"\r\n       [attr.data-round-func]=\"roundFunc\"\r\n       [attr.data-values]=\"values?.join(',')\"\r\n       [attr.data-message]=\"message\"\r\n       [attr.data-star-color]=\"starColor\"\r\n       [attr.data-stared-color]=\"staredColor\"\r\n       [attr.data-static]=\"static\"\r\n\r\n       [attr.data-cls-rating]=\"clsRating\"\r\n       [attr.data-cls-stars]=\"clsStars\"\r\n       [attr.data-cls-result]=\"clsResult\">\r\n",
            encapsulation: ViewEncapsulation.None,
            providers: [DefaultValueAccessor.get(RatingComponent_1), TypeAlias.get(RatingComponent_1)],
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [".rating.primary .stars li{color:#3969a0}.rating.primary .stars li.on{color:#0366d6}.rating.secondary .stars li{color:#767676}.rating.secondary .stars li.on{color:#607d8b}.rating.success .stars li{color:#607947}.rating.success .stars li.on{color:#60a917}.rating.alert .stars li{color:#906d6b}.rating.alert .stars li.on{color:#ce352c}.rating.warning .stars li{color:#d19c75}.rating.warning .stars li.on{color:#ff9447}.rating.yellow .stars li{color:#e0d3a3}.rating.yellow .stars li.on{color:#ffe484}.rating.info .stars li{color:#8baebf}.rating.info .stars li.on{color:#5ebdec}.rating.dark .stars li,.rating.dark .stars li.on{color:#505050}.rating.light .stars li,.rating.light .stars li.on{color:#f8f8f8}.disabled .rating .stars li.on{color:#e4e4e4!important}"]
        })
    ], RatingComponent);
    return RatingComponent;
}(ControlBase));
export { RatingComponent };
