// ng-metro4/esm2015/lib/information/hint/hint.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
import { AttributeHelper } from '../../helper/attribute-helper';
import { asapScheduler } from 'rxjs';
let HintDirective = class HintDirective {
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.hintPosition = 'top';
        this.hintOffset = 6;
    }
    createElement() {
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-text', this.hintText);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-position', this.hintPosition);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-hide', this.hintHide);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-cls-hint', this.clsHint);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-offset', this.hintOffset);
        asapScheduler.schedule(() => {
            if (!this.hintObj) {
                this.hintObj = $(this.element.nativeElement).hint().data('hint');
            }
            else {
                this.hintObj.options.hintText = this.hintText;
                this.hintObj.options.hintPosition = this.hintPosition;
                this.hintObj.options.hintHide = this.hintHide;
                this.hintObj.options.clsHint = this.clsHint;
                this.hintObj.options.hintOffset = this.hintOffset;
            }
        });
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
};
HintDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
__decorate([
    Input('m4-hint'),
    __metadata("design:type", String)
], HintDirective.prototype, "hintText", void 0);
__decorate([
    Input('hint-position'),
    __metadata("design:type", String)
], HintDirective.prototype, "hintPosition", void 0);
__decorate([
    Input('hint-hide'),
    __metadata("design:type", Number)
], HintDirective.prototype, "hintHide", void 0);
__decorate([
    Input('cls-hint'),
    __metadata("design:type", String)
], HintDirective.prototype, "clsHint", void 0);
__decorate([
    Input('hint-offset'),
    __metadata("design:type", Object)
], HintDirective.prototype, "hintOffset", void 0);
HintDirective = __decorate([
    Directive({
        selector: '[m4-hint]'
    }),
    __metadata("design:paramtypes", [ElementRef, Renderer2])
], HintDirective);
export { HintDirective };
