// ng-metro4/esm5/lib/form/checkbox/checkbox.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { ObjectHelper } from '../../helper/object-helper';
var CheckboxComponent = /** @class */ (function (_super) {
    __extends(CheckboxComponent, _super);
    function CheckboxComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.readonly = false;
        return _this;
    }
    CheckboxComponent_1 = CheckboxComponent;
    CheckboxComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.checkbox = _this.clonedElement.checkbox().data('checkbox');
            _this.clonedElement.one('blur', function () {
                _this.touchCallback();
            });
            _this.clonedElement.on('change', function () {
                _this.changeValue(_this.clonedElement.prop('checked'));
            });
            complete();
        });
    };
    CheckboxComponent.prototype.disable = function (disabled) {
        if (disabled) {
            this.clonedElement.attr('disabled', 'disabled');
            this.checkbox.disable();
        }
        else {
            this.clonedElement.removeAttr('disabled');
            this.checkbox.enable();
        }
    };
    CheckboxComponent.prototype.newValue = function () {
        if (!this.checkbox) {
            return;
        }
        this.clonedElement.prop('checked', this.innerValue);
    };
    CheckboxComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        if (this.clonedElement) {
            var target_1 = this.clonedElement.parent();
            oldClasses.forEach(function (cls) {
                target_1.removeClass(cls);
            });
            newClasses.forEach(function (cls) {
                target_1.addClass(cls);
            });
        }
    };
    var CheckboxComponent_1;
    __decorate([
        Input('value'),
        __metadata("design:type", Object)
    ], CheckboxComponent.prototype, "value", void 0);
    __decorate([
        Input('readonly'),
        __metadata("design:type", Object)
    ], CheckboxComponent.prototype, "readonly", void 0);
    __decorate([
        Input('style'),
        __metadata("design:type", Number)
    ], CheckboxComponent.prototype, "style", void 0);
    __decorate([
        Input('caption'),
        __metadata("design:type", String)
    ], CheckboxComponent.prototype, "caption", void 0);
    __decorate([
        Input('caption-position'),
        __metadata("design:type", String)
    ], CheckboxComponent.prototype, "captionPosition", void 0);
    __decorate([
        Input('indeterminate'),
        __metadata("design:type", Boolean)
    ], CheckboxComponent.prototype, "indeterminate", void 0);
    __decorate([
        Input('cls-checkbox'),
        __metadata("design:type", String)
    ], CheckboxComponent.prototype, "clsCheckbox", void 0);
    __decorate([
        Input('cls-caption'),
        __metadata("design:type", String)
    ], CheckboxComponent.prototype, "clsCaption", void 0);
    __decorate([
        Input('cls-check'),
        __metadata("design:type", String)
    ], CheckboxComponent.prototype, "clsCheck", void 0);
    __decorate([
        ViewChild('input', { static: true }),
        __metadata("design:type", ElementRef)
    ], CheckboxComponent.prototype, "input", void 0);
    CheckboxComponent = CheckboxComponent_1 = __decorate([
        Component({
            selector: 'm4-checkbox',
            template: "<input type=\"checkbox\" #input\r\n\r\n\r\n       [readonly]=\"readonly\"\r\n\r\n       [attr.data-style]=\"style\"\r\n       [attr.data-caption]=\"caption\"\r\n       [attr.data-caption-position]=\"captionPosition\"\r\n       [attr.data-indeterminate]=\"indeterminate\"\r\n       [attr.data-cls-radio]=\"clsCheckbox\"\r\n       [attr.data-cls-caption]=\"clsCaption\"\r\n       [attr.data-cls-check]=\"clsCheck\">\r\n",
            providers: [DefaultValueAccessor.get(CheckboxComponent_1), TypeAlias.get(CheckboxComponent_1)],
            changeDetection: ChangeDetectionStrategy.OnPush,
            encapsulation: ViewEncapsulation.None,
            styles: [".checkbox.primary input[data-indeterminate=true]~.check,.checkbox.primary input[type=checkbox]:checked~.check{background-color:#0366d6;border-color:#0366d6}.checkbox.primary .check{border-color:#0366d6}.checkbox.secondary input[data-indeterminate=true]~.check,.checkbox.secondary input[type=checkbox]:checked~.check{background-color:#607d8b;border-color:#607d8b}.checkbox.secondary .check{border-color:#607d8b}.checkbox.success input[data-indeterminate=true]~.check,.checkbox.success input[type=checkbox]:checked~.check{background-color:#60a917;border-color:#60a917}.checkbox.success .check{border-color:#60a917}.checkbox.alert input[data-indeterminate=true]~.check,.checkbox.alert input[type=checkbox]:checked~.check{background-color:#ce352c;border-color:#ce352c}.checkbox.alert .check{border-color:#ce352c}.checkbox.warning input[data-indeterminate=true]~.check,.checkbox.warning input[type=checkbox]:checked~.check{background-color:#ff9447;border-color:#ff9447}.checkbox.warning .check{border-color:#ff9447}.checkbox.yellow input[data-indeterminate=true]~.check,.checkbox.yellow input[type=checkbox]:checked~.check{background-color:#ffe484;border-color:#ffe484}.checkbox.yellow .check{border-color:#ffe484}.checkbox.info input[data-indeterminate=true]~.check,.checkbox.info input[type=checkbox]:checked~.check{background-color:#5ebdec;border-color:#5ebdec}.checkbox.info .check{border-color:#5ebdec}.checkbox.dark input[data-indeterminate=true]~.check,.checkbox.dark input[type=checkbox]:checked~.check{background-color:#505050;border-color:#505050}.checkbox.dark .check{border-color:#505050}.checkbox.light input[data-indeterminate=true]~.check,.checkbox.light input[type=checkbox]:checked~.check{background-color:#f8f8f8;border-color:#f8f8f8}.checkbox.light .check{border-color:#f8f8f8}"]
        })
    ], CheckboxComponent);
    return CheckboxComponent;
}(ControlBase));
export { CheckboxComponent };
