// ng-metro4/esm2015/lib/base/loading/loading.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, EventEmitter, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
import { Subscription } from 'rxjs';
import { AttributeHelper } from '../../helper/attribute-helper';
let LoadingDirective = class LoadingDirective {
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.stateChange = new EventEmitter();
    }
    createElement() {
        if (this.registeredTeardownLogic) {
            this.registeredTeardownLogic.unsubscribe();
        }
        if (this.subscription) {
            this.setDisabled(!this.subscription.closed);
            this.registeredTeardownLogic = this.subscription.add(() => {
                this.setDisabled(!this.subscription.closed);
            });
        }
        else {
            this.setDisabled(false);
        }
    }
    setDisabled(loading) {
        this.stateChange.emit(loading);
        AttributeHelper.setAttribute(this.renderer, this.element, 'disabled', this.disabled || loading);
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
};
LoadingDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
__decorate([
    Input('m4-loading'),
    __metadata("design:type", Subscription)
], LoadingDirective.prototype, "subscription", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], LoadingDirective.prototype, "disabled", void 0);
LoadingDirective = __decorate([
    Directive({
        selector: '[m4-loading]'
    }),
    __metadata("design:paramtypes", [ElementRef, Renderer2])
], LoadingDirective);
export { LoadingDirective };
