// ng-metro4/esm2015/lib/base/loading/loading-display.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnDestroy, TemplateRef, ViewContainerRef } from '@angular/core';
import { LoadingDirective } from './loading.directive';
let LoadingDisplayDirective = class LoadingDisplayDirective {
    constructor(element, templateRef, viewContainer, loadingDirective) {
        this.element = element;
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
        this.loadingDirective = loadingDirective;
        this.subscription = this.loadingDirective.stateChange.subscribe((loading) => {
            this.loadingState = loading;
            this.updateView();
        });
    }
    set loading(val) {
        this.conditionValue = val;
        this.updateView();
    }
    updateView() {
        this.viewContainer.clear();
        if (this.loadingState || this.conditionValue) {
            this.viewContainer.createEmbeddedView(this.templateRef);
        }
    }
    ngOnDestroy() {
        if (this.subscription) {
            this.subscription.unsubscribe();
        }
    }
};
LoadingDisplayDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: TemplateRef },
    { type: ViewContainerRef },
    { type: LoadingDirective }
];
__decorate([
    Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], LoadingDisplayDirective.prototype, "loading", null);
LoadingDisplayDirective = __decorate([
    Directive({
        selector: '[loading]'
    }),
    __metadata("design:paramtypes", [ElementRef,
        TemplateRef,
        ViewContainerRef,
        LoadingDirective])
], LoadingDisplayDirective);
export { LoadingDisplayDirective };
