// ng-metro4/esm5/lib/controls/activity/activity.component.js
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
var ActivityComponent = /** @class */ (function () {
    function ActivityComponent() {
    }
    ActivityComponent.prototype.createControl = function () {
        var _this = this;
        asapScheduler.schedule(function () {
            var originalElement = $(_this.activity.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.clonedElement.activity();
        });
    };
    ActivityComponent.prototype.ngOnInit = function () {
        this.createControl();
    };
    ActivityComponent.prototype.ngOnChanges = function (changes) {
        this.createControl();
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], ActivityComponent.prototype, "type", void 0);
    __decorate([
        Input('activity-style'),
        __metadata("design:type", String)
    ], ActivityComponent.prototype, "activityStyle", void 0);
    __decorate([
        ViewChild('activity', { static: true }),
        __metadata("design:type", ElementRef)
    ], ActivityComponent.prototype, "activity", void 0);
    ActivityComponent = __decorate([
        Component({
            selector: 'm4-activity',
            template: "<div #activity\r\n     [attr.data-type]=\"type\"\r\n     [attr.data-style]=\"activityStyle\"\r\n></div>\r\n",
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        }),
        __metadata("design:paramtypes", [])
    ], ActivityComponent);
    return ActivityComponent;
}());
export { ActivityComponent };
