// ng-metro4/esm5/lib/form/tag-input/tag-input.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ControlBase } from '../control-base';
import { TypeAlias } from '../../helper/type-alias';
import { ObjectHelper } from '../../helper/object-helper';
var TagInputComponent = /** @class */ (function (_super) {
    __extends(TagInputComponent, _super);
    function TagInputComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.clearButton = true;
        _this.readonly = false;
        _this.backspace = true;
        return _this;
    }
    TagInputComponent_1 = TagInputComponent;
    TagInputComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.tagInput = _this.clonedElement.taginput().data('taginput');
            _this.clonedElement.next('.input-wrapper').one('blur', function () {
                _this.touchCallback();
            });
            _this.tagInput.options.onTag = function (tag, val, vals) {
                _this.changeValue(vals.slice(0));
            };
            complete();
        });
    };
    TagInputComponent.prototype.disable = function (disabled) {
        if (this.tagInput) {
            var target = this.tagInput.element.parent('div.tag-input');
            if (target != null) {
                if (disabled) {
                    target.addClass('disabled');
                }
                else {
                    target.removeClass('disabled');
                }
            }
        }
    };
    TagInputComponent.prototype.newValue = function () {
        if (!this.tagInput) {
            return;
        }
        this.tagInput.clear();
        if (this.innerValue && this.innerValue.length > 0) {
            this.tagInput.val(this.innerValue);
        }
    };
    TagInputComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        if (this.clonedElement) {
            var target_1 = this.clonedElement.parent();
            oldClasses.forEach(function (cls) {
                target_1.removeClass(cls);
            });
            newClasses.forEach(function (cls) {
                target_1.addClass(cls);
            });
        }
    };
    var TagInputComponent_1;
    __decorate([
        Input('random-color'),
        __metadata("design:type", Boolean)
    ], TagInputComponent.prototype, "randomColor", void 0);
    __decorate([
        Input('max-tags'),
        __metadata("design:type", Number)
    ], TagInputComponent.prototype, "maxTags", void 0);
    __decorate([
        Input('tag-trigger'),
        __metadata("design:type", Array)
    ], TagInputComponent.prototype, "tagTrigger", void 0);
    __decorate([
        Input('clear-button'),
        __metadata("design:type", Object)
    ], TagInputComponent.prototype, "clearButton", void 0);
    __decorate([
        Input('clear-button-icon'),
        __metadata("design:type", String)
    ], TagInputComponent.prototype, "clearButtonIcon", void 0);
    __decorate([
        Input('readonly'),
        __metadata("design:type", Object)
    ], TagInputComponent.prototype, "readonly", void 0);
    __decorate([
        Input('backspace'),
        __metadata("design:type", Object)
    ], TagInputComponent.prototype, "backspace", void 0);
    __decorate([
        Input('cls-tag'),
        __metadata("design:type", String)
    ], TagInputComponent.prototype, "clsTag", void 0);
    __decorate([
        Input('cls-tag-title'),
        __metadata("design:type", String)
    ], TagInputComponent.prototype, "clsTagTitle", void 0);
    __decorate([
        Input('cls-tag-remover'),
        __metadata("design:type", String)
    ], TagInputComponent.prototype, "clsTagRemover", void 0);
    __decorate([
        Input('cls-clear-button'),
        __metadata("design:type", String)
    ], TagInputComponent.prototype, "clsClearButton", void 0);
    __decorate([
        Input('cls-component'),
        __metadata("design:type", String)
    ], TagInputComponent.prototype, "clsComponent", void 0);
    __decorate([
        Input('cls-input'),
        __metadata("design:type", String)
    ], TagInputComponent.prototype, "clsInput", void 0);
    __decorate([
        ViewChild('input', { static: true }),
        __metadata("design:type", ElementRef)
    ], TagInputComponent.prototype, "input", void 0);
    TagInputComponent = TagInputComponent_1 = __decorate([
        Component({
            selector: 'm4-tag-input',
            template: "<input type=\"text\" #input\r\n\r\n\r\n       [readonly]=\"readonly\"\r\n\r\n       [attr.data-random-color]=\"randomColor\"\r\n       [attr.data-max-tags]=\"maxTags\"\r\n       [attr.data-tag-trigger]=\"tagTrigger?.join(',')\"\r\n       [attr.data-backspace]=\"backspace\"\r\n       [attr.data-clear-button]=\"clearButton\"\r\n       [attr.data-clear-button-icon]=\"clearButtonIcon\"\r\n\r\n       [attr.data-cls-tag]=\"clsTag\"\r\n       [attr.data-cls-tag-title]=\"clsTagTitle\"\r\n       [attr.data-cls-tag-remover]=\"clsTagRemover\"\r\n       [attr.data-cls-component]=\"clsComponent\"\r\n       [attr.data-cls-input]=\"clsInput\"\r\n       [attr.data-cls-clear-button]=\"clsClearButton\">\r\n",
            providers: [DefaultValueAccessor.get(TagInputComponent_1), TypeAlias.get(TagInputComponent_1)],
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        })
    ], TagInputComponent);
    return TagInputComponent;
}(ControlBase));
export { TagInputComponent };
