// ng-metro4/esm5/lib/controls/draggable/draggable.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, Renderer2, SimpleChanges } from '@angular/core';
import { asapScheduler } from 'rxjs';
var DraggableDirective = /** @class */ (function () {
    function DraggableDirective(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.dragStart = new EventEmitter();
        this.dragStop = new EventEmitter();
        this.dragMove = new EventEmitter();
    }
    DraggableDirective.prototype.createElement = function () {
        var _this = this;
        asapScheduler.schedule(function () {
            if (_this.draggableData) {
                if (_this.dragArea) {
                    _this.draggableData.elem.remove();
                }
                _this.draggableData.destroy();
            }
            var draggableOptions = {
                onDragStart: function (position) {
                    _this.dragStart.emit(position);
                },
                onDragStop: function (position) {
                    _this.dragStop.emit(position);
                },
                onDragMove: function (position) {
                    _this.dragMove.emit(position);
                }
            };
            if (_this.dragElement) {
                draggableOptions.dragElement = _this.dragElement;
            }
            if (_this.dragArea) {
                draggableOptions.dragArea = _this.dragArea;
            }
            _this.draggableData = $(_this.element.nativeElement).draggable(draggableOptions).data('draggable');
        });
    };
    DraggableDirective.prototype.setPosition = function (position) {
        if (this.draggableData) {
            $(this.draggableData.elem).css({
                left: position.x + 'px',
                top: position.y + 'px'
            });
        }
    };
    DraggableDirective.prototype.ngOnInit = function () {
        this.createElement();
    };
    DraggableDirective.prototype.ngOnChanges = function (changes) {
        this.createElement();
    };
    DraggableDirective.prototype.ngOnDestroy = function () {
        $(this.element.nativeElement).remove();
    };
    DraggableDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    __decorate([
        Input('m4-draggable'),
        __metadata("design:type", String)
    ], DraggableDirective.prototype, "dragElement", void 0);
    __decorate([
        Input('drag-area'),
        __metadata("design:type", String)
    ], DraggableDirective.prototype, "dragArea", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], DraggableDirective.prototype, "dragStart", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], DraggableDirective.prototype, "dragStop", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], DraggableDirective.prototype, "dragMove", void 0);
    DraggableDirective = __decorate([
        Directive({
            selector: '[m4-draggable]'
        }),
        __metadata("design:paramtypes", [ElementRef, Renderer2])
    ], DraggableDirective);
    return DraggableDirective;
}());
export { DraggableDirective };
