// ng-metro4/esm5/lib/controls/popover/popover.component.js
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ContentChild, ElementRef, ViewChild } from '@angular/core';
import { PopoverDirective } from './popover.directive';
import { asapScheduler } from 'rxjs';
var PopoverComponent = /** @class */ (function () {
    function PopoverComponent() {
    }
    PopoverComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        asapScheduler.schedule(function () {
            _this.popover.popoverObj.options.onPopoverShow = function () {
                if (_this.popover.popoverObj.popover) {
                    $(_this.content.nativeElement).children().appendTo(_this.popover.popoverObj.popover.find('.popover-content'));
                    _this.popover.popoverObj.size = Metro.utils.hiddenElementSize(_this.popover.popoverObj.popover);
                    _this.popover.popoverObj.setPosition();
                }
            };
            _this.popover.popoverObj.options.onPopoverHide = function () {
                asapScheduler.schedule(function () {
                    if (_this.popover.popoverObj && _this.popover.popoverObj.popover) {
                        _this.popover.popoverObj.popover.find('.popover-content').children().appendTo($(_this.content.nativeElement));
                    }
                }, 300);
            };
        }, 1);
    };
    __decorate([
        ContentChild(PopoverDirective, { static: true }),
        __metadata("design:type", PopoverDirective)
    ], PopoverComponent.prototype, "popover", void 0);
    __decorate([
        ViewChild('popoverContent', { static: true }),
        __metadata("design:type", ElementRef)
    ], PopoverComponent.prototype, "content", void 0);
    PopoverComponent = __decorate([
        Component({
            selector: 'm4-popover',
            template: "<ng-content></ng-content>\r\n\r\n<div class=\"m4-popover-content\" #popoverContent>\r\n  <ng-content select=\"[content]\"></ng-content>\r\n</div>\r\n",
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [".m4-popover-content{display:none}"]
        })
    ], PopoverComponent);
    return PopoverComponent;
}());
export { PopoverComponent };
