// ng-metro4/esm5/lib/controls/popover/popover.directive.js
import { __decorate, __metadata, __param } from "tslib";
import { Directive, ElementRef, Inject, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import { AttributeHelper } from '../../helper/attribute-helper';
import { asapScheduler } from 'rxjs';
var PopoverDirective = /** @class */ (function () {
    function PopoverDirective(element, renderer, document) {
        this.element = element;
        this.renderer = renderer;
        this.document = document;
        this.popoverPosition = 'top';
    }
    PopoverDirective.prototype.createElement = function () {
        var _this = this;
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-text', this.popoverText);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-position', this.popoverPosition);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-trigger', this.popoverTrigger);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-popover-hide', this.popoverHide);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-cls-popover', this.clsPopover);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-close-button', this.closeBtn);
        asapScheduler.schedule(function () {
            if (!_this.popoverObj) {
                _this.popoverObj = $(_this.element.nativeElement).popover().data('popover');
            }
            else {
                _this.popoverObj.options.popoverText = _this.popoverText;
                _this.popoverObj.options.popoverPosition = _this.popoverPosition;
                _this.popoverObj.options.popoverTrigger = _this.popoverTrigger;
                _this.popoverObj.options.popoverHide = _this.popoverHide;
                _this.popoverObj.options.closeButton = _this.closeBtn;
                _this.popoverObj.options.clsPopover = _this.clsPopover;
            }
        });
    };
    PopoverDirective.prototype.ngOnInit = function () {
        this.createElement();
    };
    PopoverDirective.prototype.ngOnChanges = function (changes) {
        this.createElement();
    };
    PopoverDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 },
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
    ]; };
    __decorate([
        Input('m4-popover'),
        __metadata("design:type", String)
    ], PopoverDirective.prototype, "popoverText", void 0);
    __decorate([
        Input('popover-position'),
        __metadata("design:type", String)
    ], PopoverDirective.prototype, "popoverPosition", void 0);
    __decorate([
        Input('popover-trigger'),
        __metadata("design:type", String)
    ], PopoverDirective.prototype, "popoverTrigger", void 0);
    __decorate([
        Input('popover-hide'),
        __metadata("design:type", Number)
    ], PopoverDirective.prototype, "popoverHide", void 0);
    __decorate([
        Input('popover-close-button'),
        __metadata("design:type", Boolean)
    ], PopoverDirective.prototype, "closeBtn", void 0);
    __decorate([
        Input('cls-popover'),
        __metadata("design:type", Number)
    ], PopoverDirective.prototype, "clsPopover", void 0);
    PopoverDirective = __decorate([
        Directive({
            selector: '[m4-popover]'
        }),
        __param(2, Inject(DOCUMENT)),
        __metadata("design:paramtypes", [ElementRef, Renderer2, Object])
    ], PopoverDirective);
    return PopoverDirective;
}());
export { PopoverDirective };
