// ng-metro4/esm2015/lib/form/material-input/material-input.component.js
var MaterialInputComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ControlBase } from '../control-base';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
let MaterialInputComponent = MaterialInputComponent_1 = class MaterialInputComponent extends ControlBase {
    constructor() {
        super(...arguments);
        this.type = 'text';
        this.placeholder = '';
        this.readonly = false;
    }
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.parent().remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.materialInput = this.clonedElement.materialinput().data('materialinput');
            this.clonedElement.one('blur', () => {
                this.touchCallback();
            });
            this.clonedElement.on('keydown change', () => {
                asapScheduler.schedule(() => {
                    let newValue = this.clonedElement.val();
                    if (this.type === 'number') {
                        newValue = +newValue;
                    }
                    this.changeValue(newValue);
                }, 1);
            });
            complete();
        });
    }
    disable(disabled) {
        if (disabled) {
            this.clonedElement.attr('disabled', 'disabled');
            this.materialInput.disable();
        }
        else {
            this.clonedElement.removeAttr('disabled');
            this.materialInput.enable();
        }
    }
    newValue() {
        if (!this.materialInput) {
            return;
        }
        this.clonedElement.val(this.innerValue);
    }
    newClassValue(newClasses, oldClasses) {
        if (this.clonedElement) {
            const target = this.clonedElement.parent();
            oldClasses.forEach((cls) => {
                target.removeClass(cls);
            });
            newClasses.forEach((cls) => {
                target.addClass(cls);
            });
        }
    }
};
__decorate([
    Input('type'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "type", void 0);
__decorate([
    Input('placeholder'),
    __metadata("design:type", Object)
], MaterialInputComponent.prototype, "placeholder", void 0);
__decorate([
    Input('readonly'),
    __metadata("design:type", Object)
], MaterialInputComponent.prototype, "readonly", void 0);
__decorate([
    Input('label'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "label", void 0);
__decorate([
    Input('informer'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "informer", void 0);
__decorate([
    Input('icon'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "icon", void 0);
__decorate([
    Input('cls-line'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "clsLine", void 0);
__decorate([
    Input('cls-label'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "clsLabel", void 0);
__decorate([
    Input('cls-informer'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "clsInformer", void 0);
__decorate([
    Input('cls-icon'),
    __metadata("design:type", String)
], MaterialInputComponent.prototype, "clsIcon", void 0);
__decorate([
    ViewChild('input', { static: true }),
    __metadata("design:type", ElementRef)
], MaterialInputComponent.prototype, "input", void 0);
MaterialInputComponent = MaterialInputComponent_1 = __decorate([
    Component({
        selector: 'm4-material-input',
        template: "<input [type]=\"type\" #input\r\n\r\n\r\n       [placeholder]=\"placeholder\"\r\n       [readonly]=\"readonly\"\r\n\r\n       [attr.data-icon]=\"icon\"\r\n       [attr.data-label]=\"label\"\r\n       [attr.data-informer]=\"informer\"\r\n       [attr.data-cls-line]=\"clsLine\"\r\n       [attr.data-cls-label]=\"clsLabel\"\r\n       [attr.data-cls-informer]=\"clsInformer\"\r\n       [attr.data-cls-icon]=\"clsIcon\">\r\n",
        providers: [DefaultValueAccessor.get(MaterialInputComponent_1), TypeAlias.get(MaterialInputComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    })
], MaterialInputComponent);
export { MaterialInputComponent };
