// ng-metro4/esm2015/lib/form/textarea/textarea.component.js
var TextareaComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
let TextareaComponent = TextareaComponent_1 = class TextareaComponent extends ControlBase {
    constructor() {
        super(...arguments);
        this.placeholder = '';
        this.readonly = false;
    }
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.textarea.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.parent().remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.textareaObj = this.clonedElement.textarea().data('textarea');
            this.clonedElement.one('blur', () => {
                this.touchCallback();
            });
            this.clonedElement.on('keydown change', () => {
                asapScheduler.schedule(() => {
                    this.changeValue(this.clonedElement.val());
                });
            });
            complete();
        });
    }
    disable(disabled) {
        if (disabled) {
            this.textareaObj.disable();
        }
        else {
            this.textareaObj.enable();
        }
    }
    newValue() {
        if (!this.textareaObj) {
            return;
        }
        this.clonedElement.val(this.innerValue);
        this.textareaObj.resize();
    }
    newClassValue(newClasses, oldClasses) {
        if (this.clonedElement) {
            const target = this.clonedElement.parent();
            oldClasses.forEach((cls) => {
                target.removeClass(cls);
            });
            newClasses.forEach((cls) => {
                target.addClass(cls);
            });
        }
    }
};
__decorate([
    Input('placeholder'),
    __metadata("design:type", Object)
], TextareaComponent.prototype, "placeholder", void 0);
__decorate([
    Input('readonly'),
    __metadata("design:type", Object)
], TextareaComponent.prototype, "readonly", void 0);
__decorate([
    Input('default-value'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "defaultValue", void 0);
__decorate([
    Input('prepend'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "prepend", void 0);
__decorate([
    Input('append'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "append", void 0);
__decorate([
    Input('clear-button'),
    __metadata("design:type", Boolean)
], TextareaComponent.prototype, "clearButton", void 0);
__decorate([
    Input('clear-button-icon'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "clearButtonIcon", void 0);
__decorate([
    Input('auto-size'),
    __metadata("design:type", Boolean)
], TextareaComponent.prototype, "autoSize", void 0);
__decorate([
    Input('cls-component'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "clsComponent", void 0);
__decorate([
    Input('cls-textarea'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "clsTextarea", void 0);
__decorate([
    Input('cls-prepend'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "clsPrepend", void 0);
__decorate([
    Input('cls-append'),
    __metadata("design:type", String)
], TextareaComponent.prototype, "clsAppend", void 0);
__decorate([
    ViewChild('textarea', { static: true }),
    __metadata("design:type", ElementRef)
], TextareaComponent.prototype, "textarea", void 0);
TextareaComponent = TextareaComponent_1 = __decorate([
    Component({
        selector: 'm4-textarea',
        template: "<textarea #textarea\r\n\r\n\r\n          [placeholder]=\"placeholder\"\r\n          [readonly]=\"readonly\"\r\n\r\n          [attr.data-default-value]=\"defaultValue\"\r\n          [attr.data-prepend]=\"prepend\"\r\n          [attr.data-append]=\"append\"\r\n          [attr.data-clear-button]=\"clearButton\"\r\n          [attr.data-clear-button-icon]=\"clearButtonIcon\"\r\n          [attr.data-auto-size]=\"autoSize\"\r\n\r\n          [attr.data-cls-component]=\"clsComponent\"\r\n          [attr.data-cls-textarea]=\"clsTextarea\"\r\n          [attr.data-cls-prepend]=\"clsPrepend\"\r\n          [attr.data-cls-append]=\"clsAppend\">\r\n</textarea>\r\n",
        providers: [DefaultValueAccessor.get(TextareaComponent_1), TypeAlias.get(TextareaComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    })
], TextareaComponent);
export { TextareaComponent };
