// ng-metro4/esm5/lib/information/hint/hint.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
import { AttributeHelper } from '../../helper/attribute-helper';
import { asapScheduler } from 'rxjs';
var HintDirective = /** @class */ (function () {
    function HintDirective(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.hintPosition = 'top';
        this.hintOffset = 6;
    }
    HintDirective.prototype.createElement = function () {
        var _this = this;
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-text', this.hintText);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-position', this.hintPosition);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-hide', this.hintHide);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-cls-hint', this.clsHint);
        AttributeHelper.setAttribute(this.renderer, this.element, 'data-hint-offset', this.hintOffset);
        asapScheduler.schedule(function () {
            if (!_this.hintObj) {
                _this.hintObj = $(_this.element.nativeElement).hint().data('hint');
            }
            else {
                _this.hintObj.options.hintText = _this.hintText;
                _this.hintObj.options.hintPosition = _this.hintPosition;
                _this.hintObj.options.hintHide = _this.hintHide;
                _this.hintObj.options.clsHint = _this.clsHint;
                _this.hintObj.options.hintOffset = _this.hintOffset;
            }
        });
    };
    HintDirective.prototype.ngOnInit = function () {
        this.createElement();
    };
    HintDirective.prototype.ngOnChanges = function (changes) {
        this.createElement();
    };
    HintDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    __decorate([
        Input('m4-hint'),
        __metadata("design:type", String)
    ], HintDirective.prototype, "hintText", void 0);
    __decorate([
        Input('hint-position'),
        __metadata("design:type", String)
    ], HintDirective.prototype, "hintPosition", void 0);
    __decorate([
        Input('hint-hide'),
        __metadata("design:type", Number)
    ], HintDirective.prototype, "hintHide", void 0);
    __decorate([
        Input('cls-hint'),
        __metadata("design:type", String)
    ], HintDirective.prototype, "clsHint", void 0);
    __decorate([
        Input('hint-offset'),
        __metadata("design:type", Object)
    ], HintDirective.prototype, "hintOffset", void 0);
    HintDirective = __decorate([
        Directive({
            selector: '[m4-hint]'
        }),
        __metadata("design:paramtypes", [ElementRef, Renderer2])
    ], HintDirective);
    return HintDirective;
}());
export { HintDirective };
