// ng-metro4/esm2015/lib/controls/hotkey/hotkey.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, EventEmitter, Input, Output } from '@angular/core';
let HotkeyDirective = class HotkeyDirective {
    constructor() {
        this.hotkeyClick = new EventEmitter();
    }
    createElement() {
        this.unregister();
        this.previousKey = this.hotkey;
        $('body').hotkey(this.hotkey, () => {
            this.hotkeyClick.emit();
        });
    }
    ngOnInit() {
        this.createElement();
    }
    ngOnChanges(changes) {
        this.createElement();
    }
    unregister() {
        if (this.previousKey) {
            $('body').off('keyup', null, { ns: 'hotkey-method-' + this.previousKey });
        }
    }
    ngOnDestroy() {
        this.unregister();
    }
};
__decorate([
    Input('m4-hotkey'),
    __metadata("design:type", String)
], HotkeyDirective.prototype, "hotkey", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], HotkeyDirective.prototype, "hotkeyClick", void 0);
HotkeyDirective = __decorate([
    Directive({
        selector: '[m4-hotkey]'
    }),
    __metadata("design:paramtypes", [])
], HotkeyDirective);
export { HotkeyDirective };
