// ng-metro4/esm5/lib/form/date-picker/date-picker.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { ControlBase } from '../control-base';
import * as moment from 'moment';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { ObjectHelper } from '../../helper/object-helper';
var _moment = moment;
var DatePickerComponent = /** @class */ (function (_super) {
    __extends(DatePickerComponent, _super);
    function DatePickerComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    DatePickerComponent_1 = DatePickerComponent;
    DatePickerComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            var originalElement = $(_this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (_this.clonedElement) {
                _this.clonedElement.parent().remove();
            }
            _this.clonedElement = originalElement.clone();
            ObjectHelper.show(_this.clonedElement);
            originalElement.parent().append(_this.clonedElement);
            _this.datePicker = _this.clonedElement.datepicker().data('datepicker');
            _this.clonedElement.parent().find('.date-wrapper').one('click', function () {
                _this.touchCallback();
            });
            _this.datePicker.options.onSet = function (val, elem_val) {
                _this.changeValue(_moment(elem_val, 'YYYY-MM-DD'));
            };
            complete();
        });
    };
    DatePickerComponent.prototype.disable = function (disabled) {
        if (disabled) {
            this.clonedElement.parent().addClass('disabled');
        }
        else {
            this.clonedElement.parent().removeClass('disabled');
        }
    };
    DatePickerComponent.prototype.newValue = function () {
        if (!this.datePicker) {
            return;
        }
        this.datePicker.val(this.innerValue ? this.innerValue.format('YYYY-MM-DD') : '');
    };
    DatePickerComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        if (this.clonedElement) {
            var target_1 = this.clonedElement.parent();
            oldClasses.forEach(function (cls) {
                target_1.removeClass(cls);
            });
            newClasses.forEach(function (cls) {
                target_1.addClass(cls);
            });
        }
    };
    var DatePickerComponent_1;
    __decorate([
        Input('month'),
        __metadata("design:type", Boolean)
    ], DatePickerComponent.prototype, "month", void 0);
    __decorate([
        Input('day'),
        __metadata("design:type", Boolean)
    ], DatePickerComponent.prototype, "day", void 0);
    __decorate([
        Input('year'),
        __metadata("design:type", Boolean)
    ], DatePickerComponent.prototype, "year", void 0);
    __decorate([
        Input('locale'),
        __metadata("design:type", String)
    ], DatePickerComponent.prototype, "locale", void 0);
    __decorate([
        Input('min-year'),
        __metadata("design:type", Number)
    ], DatePickerComponent.prototype, "minYear", void 0);
    __decorate([
        Input('max-year'),
        __metadata("design:type", Number)
    ], DatePickerComponent.prototype, "maxYear", void 0);
    __decorate([
        Input('distance'),
        __metadata("design:type", Number)
    ], DatePickerComponent.prototype, "distance", void 0);
    __decorate([
        Input('cls-picker'),
        __metadata("design:type", String)
    ], DatePickerComponent.prototype, "clsPicker", void 0);
    __decorate([
        Input('cls-start'),
        __metadata("design:type", String)
    ], DatePickerComponent.prototype, "clsStart", void 0);
    __decorate([
        Input('cls-month'),
        __metadata("design:type", String)
    ], DatePickerComponent.prototype, "clsMonth", void 0);
    __decorate([
        Input('cls-day'),
        __metadata("design:type", String)
    ], DatePickerComponent.prototype, "clsDay", void 0);
    __decorate([
        Input('cls-year'),
        __metadata("design:type", String)
    ], DatePickerComponent.prototype, "clsYear", void 0);
    __decorate([
        ViewChild('input', { static: true }),
        __metadata("design:type", ElementRef)
    ], DatePickerComponent.prototype, "input", void 0);
    DatePickerComponent = DatePickerComponent_1 = __decorate([
        Component({
            selector: 'm4-date-picker',
            template: "<input type=\"text\" #input\r\n\r\n\r\n       [attr.data-month]=\"month\"\r\n       [attr.data-day]=\"day\"\r\n       [attr.data-year]=\"year\"\r\n       [attr.data-locale]=\"locale\"\r\n       [attr.data-min-year]=\"minYear\"\r\n       [attr.data-max-year]=\"maxYear\"\r\n       [attr.data-distance]=\"distance\"\r\n\r\n       [attr.data-cls-picker]=\"clsPicker\"\r\n       [attr.data-cls-start]=\"clsStart\"\r\n       [attr.data-cls-month]=\"clsMonth\"\r\n       [attr.data-cls-day]=\"clsDay\"\r\n       [attr.data-cls-year]=\"clsYear\">\r\n",
            providers: [DefaultValueAccessor.get(DatePickerComponent_1), TypeAlias.get(DatePickerComponent_1)],
            changeDetection: ChangeDetectionStrategy.OnPush,
            encapsulation: ViewEncapsulation.None,
            styles: [".date-picker.primary .date-wrapper{border-color:#0366d6}.date-picker.secondary .date-wrapper{border-color:#607d8b}.date-picker.success .date-wrapper{border-color:#60a917}.date-picker.alert .date-wrapper{border-color:#ce352c}.date-picker.warning .date-wrapper{border-color:#ff9447}.date-picker.yellow .date-wrapper{border-color:#ffe484}.date-picker.info .date-wrapper{border-color:#5ebdec}.date-picker.dark .date-wrapper{border-color:#505050}.date-picker.light .date-wrapper{border-color:#f8f8f8}"]
        })
    ], DatePickerComponent);
    return DatePickerComponent;
}(ControlBase));
export { DatePickerComponent };
