// ng-metro4/esm2015/lib/form/calendar-picker/calendar-picker.component.js
var CalendarPickerComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ControlBase } from '../control-base';
import * as moment from 'moment';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
const _moment = moment;
let CalendarPickerComponent = CalendarPickerComponent_1 = class CalendarPickerComponent extends ControlBase {
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.parent().remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.calendarPicker = this.clonedElement.calendarpicker().data('calendarpicker');
            this.clonedElement.one('blur', () => {
                this.touchCallback();
            });
            this.calendarPicker.options.onChange = () => {
                asapScheduler.schedule(() => {
                    const val = this.calendarPicker.value;
                    this.changeValue(val ? _moment(val.toLocaleDateString('en'), 'MM/DD/YYYY') : null);
                });
            };
            complete();
        });
    }
    disable(disabled) {
        asapScheduler.schedule(() => {
            if (disabled) {
                this.calendarPicker.disable();
            }
            else {
                this.calendarPicker.enable();
            }
        });
    }
    newValue() {
        if (!this.calendarPicker) {
            return;
        }
        asapScheduler.schedule(() => {
            this.calendarPicker.val(this.innerValue ? this.innerValue.format('MM/DD/YYYY') : '01/01/0000');
        });
    }
    convertMomentArray(arr) {
        return arr ? arr.map(v => v.format('MM/DD/YYYY')).join(',') : null;
    }
    newClassValue(newClasses, oldClasses) {
        asapScheduler.schedule(() => {
            if (this.clonedElement) {
                const target = this.clonedElement.parent();
                oldClasses.forEach((cls) => {
                    target.removeClass(cls);
                });
                newClasses.forEach((cls) => {
                    target.addClass(cls);
                });
            }
        });
    }
};
__decorate([
    Input('calendar-wide'),
    __metadata("design:type", Boolean)
], CalendarPickerComponent.prototype, "calendarWide", void 0);
__decorate([
    Input('calendar-wide-point'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "calendarWidePoint", void 0);
__decorate([
    Input('dialog-mode'),
    __metadata("design:type", Boolean)
], CalendarPickerComponent.prototype, "dialogMode", void 0);
__decorate([
    Input('dialog-point'),
    __metadata("design:type", Number)
], CalendarPickerComponent.prototype, "dialogPoint", void 0);
__decorate([
    Input('dialog-overlay'),
    __metadata("design:type", Boolean)
], CalendarPickerComponent.prototype, "dialogOverlay", void 0);
__decorate([
    Input('overlay-color'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "overlayColor", void 0);
__decorate([
    Input('overlay-alpha'),
    __metadata("design:type", Number)
], CalendarPickerComponent.prototype, "overlayAlpha", void 0);
__decorate([
    Input('locale'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "locale", void 0);
__decorate([
    Input('size'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "size", void 0);
__decorate([
    Input('format'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "format", void 0);
__decorate([
    Input('clear-button'),
    __metadata("design:type", Boolean)
], CalendarPickerComponent.prototype, "clearButton", void 0);
__decorate([
    Input('clear-button-icon'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clearButtonIcon", void 0);
__decorate([
    Input('calendar-button-icon'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "calendarButtonIcon", void 0);
__decorate([
    Input('years-before'),
    __metadata("design:type", Number)
], CalendarPickerComponent.prototype, "yearsBefore", void 0);
__decorate([
    Input('years-after'),
    __metadata("design:type", Number)
], CalendarPickerComponent.prototype, "yearsAfter", void 0);
__decorate([
    Input('week-start'),
    __metadata("design:type", Number)
], CalendarPickerComponent.prototype, "weekStart", void 0);
__decorate([
    Input('outside'),
    __metadata("design:type", Boolean)
], CalendarPickerComponent.prototype, "outside", void 0);
__decorate([
    Input('ripple'),
    __metadata("design:type", Boolean)
], CalendarPickerComponent.prototype, "ripple", void 0);
__decorate([
    Input('ripple-color'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "rippleColor", void 0);
__decorate([
    Input('exclude'),
    __metadata("design:type", Array)
], CalendarPickerComponent.prototype, "exclude", void 0);
__decorate([
    Input('include'),
    __metadata("design:type", Array)
], CalendarPickerComponent.prototype, "include", void 0);
__decorate([
    Input('min-date'),
    __metadata("design:type", Object)
], CalendarPickerComponent.prototype, "minDate", void 0);
__decorate([
    Input('max-date'),
    __metadata("design:type", Object)
], CalendarPickerComponent.prototype, "maxDate", void 0);
__decorate([
    Input('show-header'),
    __metadata("design:type", Boolean)
], CalendarPickerComponent.prototype, "showHeader", void 0);
__decorate([
    Input('cls-picker'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsPicker", void 0);
__decorate([
    Input('cls-today'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsToday", void 0);
__decorate([
    Input('cls-selected'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsSelected", void 0);
__decorate([
    Input('cls-exclude'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsExclude", void 0);
__decorate([
    Input('cls-calendar'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsCalendar", void 0);
__decorate([
    Input('cls-calendar-header'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsCalendarHeader", void 0);
__decorate([
    Input('cls-calendar-content'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsCalendarContent", void 0);
__decorate([
    Input('cls-calendar-months'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsCalendarMonths", void 0);
__decorate([
    Input('cls-calendar-years'),
    __metadata("design:type", String)
], CalendarPickerComponent.prototype, "clsCalendarYears", void 0);
__decorate([
    ViewChild('input', { static: true }),
    __metadata("design:type", ElementRef)
], CalendarPickerComponent.prototype, "input", void 0);
CalendarPickerComponent = CalendarPickerComponent_1 = __decorate([
    Component({
        selector: 'm4-calendar-picker',
        template: "<input type=\"text\" #input\r\n\r\n       data-input-format=\"%m/%d/%Y\"\r\n\r\n       [attr.data-calendar-wide]=\"calendarWide\"\r\n       [attr.data-calendar-wide-point]=\"calendarWidePoint\"\r\n       [attr.data-dialog-mode]=\"dialogMode\"\r\n       [attr.data-dialog-point]=\"dialogPoint\"\r\n       [attr.data-dialog-overlay]=\"dialogOverlay\"\r\n       [attr.data-overlay-color]=\"overlayColor\"\r\n       [attr.data-overlay-alpha]=\"overlayAlpha\"\r\n       [attr.data-locale]=\"locale\"\r\n       [attr.data-size]=\"size\"\r\n       [attr.data-format]=\"format\"\r\n       [attr.data-clear-button]=\"clearButton\"\r\n       [attr.data-clear-button-icon]=\"clearButtonIcon\"\r\n       [attr.data-calendar-button-icon]=\"calendarButtonIcon\"\r\n       [attr.data-years-before]=\"yearsBefore\"\r\n       [attr.data-years-after]=\"yearsAfter\"\r\n       [attr.data-week-start]=\"weekStart\"\r\n       [attr.data-outside]=\"outside\"\r\n       [attr.data-ripple]=\"ripple\"\r\n       [attr.data-ripple-color]=\"rippleColor\"\r\n       [attr.data-exclude]=\"convertMomentArray(exclude)\"\r\n       [attr.data-special]=\"convertMomentArray(include)\"\r\n       [attr.data-min-date]=\"minDate?.format('MM/DD/YYYY')\"\r\n       [attr.data-max-date]=\"maxDate?.format('MM/DD/YYYY')\"\r\n       [attr.data-show-header]=\"showHeader\"\r\n\r\n       [attr.data-cls-picker]=\"clsPicker\"\r\n       [attr.data-cls-today]=\"clsToday\"\r\n       [attr.data-cls-selected]=\"clsSelected\"\r\n       [attr.data-cls-exclude]=\"clsExclude\"\r\n       [attr.data-cls-calendar]=\"clsCalendar\"\r\n       [attr.data-cls-calendar-header]=\"clsCalendarHeader\"\r\n       [attr.data-cls-calendar-content]=\"clsCalendarContent\"\r\n       [attr.data-cls-calendar-months]=\"clsCalendarMonths\"\r\n       [attr.data-cls-calendar-years]=\"clsCalendarYears\">\r\n",
        providers: [DefaultValueAccessor.get(CalendarPickerComponent_1), TypeAlias.get(CalendarPickerComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    })
], CalendarPickerComponent);
export { CalendarPickerComponent };
