// ng-metro4/esm2015/lib/form/tag-input/tag-input.component.js
var TagInputComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ControlBase } from '../control-base';
import { TypeAlias } from '../../helper/type-alias';
import { ObjectHelper } from '../../helper/object-helper';
let TagInputComponent = TagInputComponent_1 = class TagInputComponent extends ControlBase {
    constructor() {
        super(...arguments);
        this.clearButton = true;
        this.readonly = false;
        this.backspace = true;
    }
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.parent().remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            this.tagInput = this.clonedElement.taginput().data('taginput');
            this.clonedElement.next('.input-wrapper').one('blur', () => {
                this.touchCallback();
            });
            this.tagInput.options.onTag = (tag, val, vals) => {
                this.changeValue(vals.slice(0));
            };
            complete();
        });
    }
    disable(disabled) {
        if (this.tagInput) {
            const target = this.tagInput.element.parent('div.tag-input');
            if (target != null) {
                if (disabled) {
                    target.addClass('disabled');
                }
                else {
                    target.removeClass('disabled');
                }
            }
        }
    }
    newValue() {
        if (!this.tagInput) {
            return;
        }
        this.tagInput.clear();
        if (this.innerValue && this.innerValue.length > 0) {
            this.tagInput.val(this.innerValue);
        }
    }
    newClassValue(newClasses, oldClasses) {
        if (this.clonedElement) {
            const target = this.clonedElement.parent();
            oldClasses.forEach((cls) => {
                target.removeClass(cls);
            });
            newClasses.forEach((cls) => {
                target.addClass(cls);
            });
        }
    }
};
__decorate([
    Input('random-color'),
    __metadata("design:type", Boolean)
], TagInputComponent.prototype, "randomColor", void 0);
__decorate([
    Input('max-tags'),
    __metadata("design:type", Number)
], TagInputComponent.prototype, "maxTags", void 0);
__decorate([
    Input('tag-trigger'),
    __metadata("design:type", Array)
], TagInputComponent.prototype, "tagTrigger", void 0);
__decorate([
    Input('clear-button'),
    __metadata("design:type", Object)
], TagInputComponent.prototype, "clearButton", void 0);
__decorate([
    Input('clear-button-icon'),
    __metadata("design:type", String)
], TagInputComponent.prototype, "clearButtonIcon", void 0);
__decorate([
    Input('readonly'),
    __metadata("design:type", Object)
], TagInputComponent.prototype, "readonly", void 0);
__decorate([
    Input('backspace'),
    __metadata("design:type", Object)
], TagInputComponent.prototype, "backspace", void 0);
__decorate([
    Input('cls-tag'),
    __metadata("design:type", String)
], TagInputComponent.prototype, "clsTag", void 0);
__decorate([
    Input('cls-tag-title'),
    __metadata("design:type", String)
], TagInputComponent.prototype, "clsTagTitle", void 0);
__decorate([
    Input('cls-tag-remover'),
    __metadata("design:type", String)
], TagInputComponent.prototype, "clsTagRemover", void 0);
__decorate([
    Input('cls-clear-button'),
    __metadata("design:type", String)
], TagInputComponent.prototype, "clsClearButton", void 0);
__decorate([
    Input('cls-component'),
    __metadata("design:type", String)
], TagInputComponent.prototype, "clsComponent", void 0);
__decorate([
    Input('cls-input'),
    __metadata("design:type", String)
], TagInputComponent.prototype, "clsInput", void 0);
__decorate([
    ViewChild('input', { static: true }),
    __metadata("design:type", ElementRef)
], TagInputComponent.prototype, "input", void 0);
TagInputComponent = TagInputComponent_1 = __decorate([
    Component({
        selector: 'm4-tag-input',
        template: "<input type=\"text\" #input\r\n\r\n\r\n       [readonly]=\"readonly\"\r\n\r\n       [attr.data-random-color]=\"randomColor\"\r\n       [attr.data-max-tags]=\"maxTags\"\r\n       [attr.data-tag-trigger]=\"tagTrigger?.join(',')\"\r\n       [attr.data-backspace]=\"backspace\"\r\n       [attr.data-clear-button]=\"clearButton\"\r\n       [attr.data-clear-button-icon]=\"clearButtonIcon\"\r\n\r\n       [attr.data-cls-tag]=\"clsTag\"\r\n       [attr.data-cls-tag-title]=\"clsTagTitle\"\r\n       [attr.data-cls-tag-remover]=\"clsTagRemover\"\r\n       [attr.data-cls-component]=\"clsComponent\"\r\n       [attr.data-cls-input]=\"clsInput\"\r\n       [attr.data-cls-clear-button]=\"clsClearButton\">\r\n",
        providers: [DefaultValueAccessor.get(TagInputComponent_1), TypeAlias.get(TagInputComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    })
], TagInputComponent);
export { TagInputComponent };
