// ng-metro4/esm5/lib/controls/hotkey/hotkey.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, EventEmitter, Input, Output } from '@angular/core';
var HotkeyDirective = /** @class */ (function () {
    function HotkeyDirective() {
        this.hotkeyClick = new EventEmitter();
    }
    HotkeyDirective.prototype.createElement = function () {
        var _this = this;
        this.unregister();
        this.previousKey = this.hotkey;
        $('body').hotkey(this.hotkey, function () {
            _this.hotkeyClick.emit();
        });
    };
    HotkeyDirective.prototype.ngOnInit = function () {
        this.createElement();
    };
    HotkeyDirective.prototype.ngOnChanges = function (changes) {
        this.createElement();
    };
    HotkeyDirective.prototype.unregister = function () {
        if (this.previousKey) {
            $('body').off('keyup', null, { ns: 'hotkey-method-' + this.previousKey });
        }
    };
    HotkeyDirective.prototype.ngOnDestroy = function () {
        this.unregister();
    };
    __decorate([
        Input('m4-hotkey'),
        __metadata("design:type", String)
    ], HotkeyDirective.prototype, "hotkey", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], HotkeyDirective.prototype, "hotkeyClick", void 0);
    HotkeyDirective = __decorate([
        Directive({
            selector: '[m4-hotkey]'
        }),
        __metadata("design:paramtypes", [])
    ], HotkeyDirective);
    return HotkeyDirective;
}());
export { HotkeyDirective };
