// ng-metro4/esm5/lib/form/radio-group/radio-group.component.js
import { __decorate, __extends, __metadata } from "tslib";
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ContentChildren, ElementRef, forwardRef, Input, QueryList, ViewChildren } from '@angular/core';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { ControlBase } from '../control-base';
import { RadioComponent } from '../radio/radio.component';
import { StringHelper } from '../../helper/string-helper';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
var RadioGroupComponent = /** @class */ (function (_super) {
    __extends(RadioGroupComponent, _super);
    function RadioGroupComponent(element, cdRef) {
        var _this = _super.call(this, element, cdRef) || this;
        _this.name = StringHelper.guid();
        return _this;
    }
    RadioGroupComponent_1 = RadioGroupComponent;
    RadioGroupComponent.prototype.createControl = function () {
        var _this = this;
        return new Promise(function (complete) {
            asapScheduler.schedule(function () {
                _this.radios = !!_this.options ? _this.radiosView : _this.radiosContent;
                var radioCreations = _this.radios.map(function (item) {
                    return new Promise(function (radioComplete) {
                        item.name = _this.name;
                        item.registerOnChange(function (v) {
                            _this.changeValue(v);
                            _this.newValue();
                        });
                        item.registerOnTouched(function () {
                            _this.touchCallback();
                        });
                        asapScheduler.schedule(function () {
                            item.createControl().then(function () {
                                radioComplete();
                            });
                        }, 1);
                    });
                });
                Promise.all(radioCreations).then(function () {
                    _this.callNewValue();
                    complete();
                });
            });
        });
    };
    RadioGroupComponent.prototype.disable = function (disabled) {
        this.radios.forEach(function (item) {
            asapScheduler.schedule(function () {
                item.disable(disabled);
            }, 1);
        });
    };
    RadioGroupComponent.prototype.newValue = function () {
        var _this = this;
        if (!this.radios) {
            return;
        }
        this.radios.forEach(function (item) {
            item.writeValue(_this.innerValue);
        });
    };
    RadioGroupComponent.prototype.newClassValue = function (newClasses, oldClasses) {
        this.radios.forEach(function (item) {
            asapScheduler.schedule(function () {
                item.newClassValue(newClasses, oldClasses);
            }, 1);
        });
    };
    var RadioGroupComponent_1;
    RadioGroupComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: ChangeDetectorRef }
    ]; };
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], RadioGroupComponent.prototype, "options", void 0);
    __decorate([
        ContentChildren(forwardRef(function () { return RadioComponent; }), { descendants: true }),
        __metadata("design:type", QueryList)
    ], RadioGroupComponent.prototype, "radiosContent", void 0);
    __decorate([
        ViewChildren(forwardRef(function () { return RadioComponent; })),
        __metadata("design:type", QueryList)
    ], RadioGroupComponent.prototype, "radiosView", void 0);
    RadioGroupComponent = RadioGroupComponent_1 = __decorate([
        Component({
            selector: 'm4-radio-group',
            template: "<ng-container *ngIf=\"options; else noOptions\">\r\n  <m4-radio *ngFor=\"let option of options | objectKeys\" [value]=\"options[option]\"\r\n            [caption]=\"option\" [name]=\"name\"></m4-radio>\r\n</ng-container>\r\n<ng-template #noOptions>\r\n  <ng-content></ng-content>\r\n</ng-template>\r\n",
            providers: [DefaultValueAccessor.get(RadioGroupComponent_1), TypeAlias.get(RadioGroupComponent_1)],
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [""]
        }),
        __metadata("design:paramtypes", [ElementRef, ChangeDetectorRef])
    ], RadioGroupComponent);
    return RadioGroupComponent;
}(ControlBase));
export { RadioGroupComponent };
