// ng-metro4/esm2015/lib/form/select/select.component.js
var SelectComponent_1;
import { __decorate, __metadata } from "tslib";
import { ChangeDetectionStrategy, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { ControlBase } from '../control-base';
import { DefaultValueAccessor } from '../../helper/default-value-accessor';
import { TypeAlias } from '../../helper/type-alias';
import { asapScheduler } from 'rxjs';
import { ObjectHelper } from '../../helper/object-helper';
import { StringHelper } from '../../helper/string-helper';
let SelectComponent = SelectComponent_1 = class SelectComponent extends ControlBase {
    constructor() {
        super(...arguments);
        this.multiple = false;
        this.duration = 0;
    }
    createControl() {
        return new Promise((complete) => {
            const originalElement = $(this.input.nativeElement);
            ObjectHelper.hide(originalElement);
            if (this.clonedElement) {
                this.clonedElement.children().appendTo(originalElement);
                this.clonedElement.parent().remove();
            }
            this.clonedElement = originalElement.clone();
            ObjectHelper.show(this.clonedElement);
            originalElement.parent().append(this.clonedElement);
            originalElement.children().appendTo(this.clonedElement);
            this.select = this.clonedElement.select().data('select');
            this.clonedElement.parent().on('mousedown', (ev) => {
                if (ev.button === 0) {
                    this.touchCallback();
                    this.clonedElement.parent().off('mousedown');
                }
            });
            this.select.options.onChange = (val) => {
                if (this.options instanceof Array) {
                    const allOptions = [];
                    this.options.forEach((option) => {
                        if (!!option.options) {
                            option.options.forEach((subOption) => {
                                allOptions.push(subOption);
                            });
                        }
                        else {
                            allOptions.push(option);
                        }
                    });
                    val = val.map(key => allOptions.find(option => StringHelper.createHash(option.value) === key)).filter(v => !!v).map(v => v.value);
                }
                if (this.multiple) {
                    this.changeValue(val.slice(0));
                }
                else {
                    this.changeValue(val[0]);
                }
            };
            if (this.options && !(this.options instanceof Array)) {
                this.select.data(this.options);
            }
            complete();
        });
    }
    disable(disabled) {
        if (disabled) {
            this.select.disable();
        }
        else {
            this.select.enable();
        }
    }
    newValue() {
        if (!this.select) {
            return;
        }
        let selectValue = this.multiple ? this.innerValue : [this.innerValue];
        if (this.options instanceof Array) {
            selectValue = selectValue.map(v => StringHelper.createHash(v));
        }
        if (this.multiple) {
            this.select.reset();
        }
        this.select.val(selectValue);
    }
    newClassValue(newClasses, oldClasses) {
        asapScheduler.schedule(() => {
            if (this.clonedElement) {
                const target = this.clonedElement.parent();
                oldClasses.forEach((cls) => {
                    target.removeClass(cls);
                });
                newClasses.forEach((cls) => {
                    target.addClass(cls);
                });
            }
        }, 1);
    }
};
__decorate([
    Input('options'),
    __metadata("design:type", Object)
], SelectComponent.prototype, "options", void 0);
__decorate([
    Input('multiple'),
    __metadata("design:type", Object)
], SelectComponent.prototype, "multiple", void 0);
__decorate([
    Input('duration'),
    __metadata("design:type", Object)
], SelectComponent.prototype, "duration", void 0);
__decorate([
    Input('prepend'),
    __metadata("design:type", String)
], SelectComponent.prototype, "prepend", void 0);
__decorate([
    Input('append'),
    __metadata("design:type", String)
], SelectComponent.prototype, "append", void 0);
__decorate([
    Input('drop-height'),
    __metadata("design:type", Number)
], SelectComponent.prototype, "dropHeight", void 0);
__decorate([
    Input('filter'),
    __metadata("design:type", Boolean)
], SelectComponent.prototype, "filter", void 0);
__decorate([
    Input('filter-placeholder'),
    __metadata("design:type", String)
], SelectComponent.prototype, "filterPlaceholder", void 0);
__decorate([
    Input('cls-select'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsSelect", void 0);
__decorate([
    Input('cls-prepend'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsPrepend", void 0);
__decorate([
    Input('cls-append'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsAppend", void 0);
__decorate([
    Input('cls-option'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsOption", void 0);
__decorate([
    Input('cls-option-group'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsOptionGroup", void 0);
__decorate([
    Input('cls-drop-list'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsDropList", void 0);
__decorate([
    Input('cls-selected-item'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsSelectedItem", void 0);
__decorate([
    Input('cls-selected-item-remover'),
    __metadata("design:type", String)
], SelectComponent.prototype, "clsSelectedItemRemover", void 0);
__decorate([
    ViewChild('input', { static: true }),
    __metadata("design:type", ElementRef)
], SelectComponent.prototype, "input", void 0);
SelectComponent = SelectComponent_1 = __decorate([
    Component({
        selector: 'm4-select',
        template: "<select #input [multiple]=\"multiple\"\r\n\r\n\r\n        [attr.data-prepend]=\"prepend\"\r\n        [attr.data-duration]=\"duration\"\r\n        [attr.data-append]=\"append\"\r\n        [attr.data-drop-height]=\"dropHeight\"\r\n        [attr.data-filter]=\"filter\"\r\n        [attr.data-filter-placeholder]=\"filterPlaceholder\"\r\n\r\n        [attr.data-cls-select]=\"clsSelect\"\r\n        [attr.data-cls-prepend]=\"clsPrepend\"\r\n        [attr.data-cls-append]=\"clsAppend\"\r\n        [attr.data-cls-option]=\"clsOption\"\r\n        [attr.data-cls-option-group]=\"clsOptionGroup\"\r\n        [attr.data-cls-drop-list]=\"clsDropList\"\r\n        [attr.data-cls-selected-item]=\"clsSelectedItem\"\r\n        [attr.data-cls-selected-item-remover]=\"clsSelectedItemRemover\">\r\n\r\n  <ng-content *ngIf=\"!options\"></ng-content>\r\n\r\n  <ng-container *ngIf=\"options | isArray\">\r\n    <ng-container *ngFor=\"let option of options\">\r\n      <option *ngIf=\"!option.options\" [value]=\"option.value | createHash\" [attr.data-template]=\"option.dataTemplate\">\r\n        {{option.title}}\r\n      </option>\r\n\r\n      <optgroup *ngIf=\"option.options\" [label]=\"option.groupName\">\r\n        <option *ngFor=\"let childOption of option.options\" [value]=\"childOption.value | createHash\" [attr.data-template]=\"childOption.dataTemplate\">\r\n          {{childOption.title}}\r\n        </option>\r\n      </optgroup>\r\n    </ng-container>\r\n  </ng-container>\r\n</select>\r\n",
        providers: [DefaultValueAccessor.get(SelectComponent_1), TypeAlias.get(SelectComponent_1)],
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [""]
    })
], SelectComponent);
export { SelectComponent };
